<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="generator" content="MATLAB R2016a"><meta http-equiv="X-UA-Compatible" content="IE=edge,IE=9,chrome=1"><title>Flux Balance Analysis (FBA)</title><style type="text/css">
* {margin: 0; padding: 0;}
body {text-align: start; line-height: 17.234px; min-height: 0px; white-space: normal; color: rgb(0, 0, 0); font-family: Consolas, Inconsolata, Menlo, monospace; font-style: normal; font-size: 14px; font-weight: normal; text-decoration: none; white-space: normal; }
h1, h2 {font-weight: normal;}
.content { padding: 30px; }

.S0 { margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S1 { line-height: 26.4px; min-height: 24px; white-space: pre-wrap; color: rgb(213, 80, 0); font-family: Helvetica, Arial, sans-serif; font-size: 22px; white-space: pre-wrap; margin-left: 4px; margin-top: 3px; margin-bottom: 15px; margin-right: 10px;  }
.S2 { min-height: 0px; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S3 { line-height: 20.576px; min-height: 20px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-size: 16px; font-weight: bold; white-space: pre-wrap; margin-left: 4px; margin-top: 15px; margin-bottom: 9px; margin-right: 10px;  }
.S4 { line-height: 21px; min-height: 17px; white-space: pre-wrap; font-family: Helvetica, Arial, sans-serif; white-space: pre-wrap; margin-left: 4px; margin-top: 2px; margin-bottom: 9px; margin-right: 10px;  }
.S5 { text-align: center; line-height: 21px; min-height: 17px; white-space: pre-wrap; font-family: Helvetica, Arial, sans-serif; white-space: pre-wrap; margin-left: 4px; margin-top: 2px; margin-bottom: 9px; margin-right: 10px;  }
.S6 { font-family: Helvetica, Arial, sans-serif; margin-left: 0px; margin-top: 10px; margin-bottom: 20px; margin-right: 0px;  }
.S7 { text-align: left; line-height: 21px; white-space: pre-wrap; white-space: pre-wrap; margin-left: 56px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S8 { font-weight: bold; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S9 { line-height: 20.576px; min-height: 20px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-size: 16px; font-weight: bold; white-space: pre-wrap; margin-left: 4px; margin-top: 3px; margin-bottom: 9px; margin-right: 10px;  }
.S10 { margin-left: 3px; margin-top: 10px; margin-bottom: 10px; margin-right: 3px;  }
.S11 { line-height: 15.5927px; min-height: 18px; white-space: nowrap; font-size: 12.6667px; white-space: nowrap; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S12 { min-height: 0px; white-space: pre; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S13 { line-height: 21px; min-height: 17px; white-space: pre-wrap; font-family: Helvetica, Arial, sans-serif; white-space: pre-wrap; margin-left: 4px; margin-top: 10px; margin-bottom: 9px; margin-right: 10px;  }
.S14 { min-height: 0px; font-family: monospace; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S15 { min-height: 0px; white-space: pre; color: rgb(34, 139, 34); white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S16 { min-height: 0px; color: rgb(0, 95, 206); margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S17 { min-height: 0px; white-space: pre; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S18 { min-height: 0px; white-space: pre; color: rgb(160, 32, 240); white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S19 { min-height: 0px; white-space: pre; color: rgb(0, 0, 255); white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S20 { min-height: 0px; text-decoration: underline; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S21 { min-height: 0px; font-weight: bold; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S22 { font-family: monospace; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S23 { min-height: 0px; white-space: pre; color: rgb(0, 0, 255); white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S24 { margin-left: 3px; margin-top: 10px; margin-bottom: 4px; margin-right: 3px;  }
.S25 { min-height: 0px; font-style: italic; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }

.LineNodeBlock {margin: 10px 0 10px 0;}
.LineNodeBlock+.paragraphNode {margin-top: 10px;}
.lineNode {padding-left: 10px; background-color: #F7F7F7; border-left: 1px solid #E9E9E9; border-right: 1px solid #E9E9E9;}
.inlineWrapper:first-child .lineNode,.inlineWrapper.outputs+.inlineWrapper .lineNode {padding-top: 5px; border-top: 1px solid #E9E9E9;}
.inlineWrapper:last-child .lineNode,.inlineWrapper.outputs .lineNode {padding-bottom: 5px; border-bottom: 1px solid #E9E9E9;}
.lineNode .textBox {white-space: pre;}
</style></head><body><div class = "content"><div class = 'SectionBlock containment active'><h1 class = "S1"><span class = "S2">Flux Balance Analysis (FBA)</span></h1><h2 class = "S3"><span class = "S2">Author(s): </span><span class = "S2">Vanja Vlasov, </span><span class = "S2">Marouen Ben Guebila, </span><span class = "S2">Systems Biochemistry Group, LCSB, University of Luxembourg, </span></h2><h2 class = "S3"><span class = "S2">Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg</span><span class = "S2"> </span></h2><h2 class = "S3"><span class = "S2"> Reviewer(s): Ines Thiele,</span><span class = "S2"> Catherine Clancy, Systems Biochemistry Group, LCSB, University of Luxembourg</span></h2><h2 class = "S3"><span class = "S2">Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg</span><span class = "S2"> </span></h2><h2 class = "S3"><span class = "S2">INTRODUCTION</span></h2><p class = "S4"><span class = "S2">Flux balance analysis (FBA) evaluates the metabolic flux distribution</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAAAjUlEQVRIS+3VIQ7CUBBF0VN2gGAPCGxXUQUGj0UhWQK2ZSWoqi6EJYAAiSQkJUFAmvLln/FzJ7l5eVNInCJxXwaAKbY44v7N1y8HM+ywxwVz3MYAljjjgPIfwPtYjXUAwkFKDiY49Ulc4DomyitsUPVLHVo0eHyCMiiUwc4NB3J4LJGDQQORg5ei5D54At1+ICnPQ6rOAAAAAElFTkSuQmCC" width="8" height="20" /></span><span class = "S2">, and is one of the most used modelling approaches for metabolic systems. </span></p><p class = "S4"><span class = "S2">The applications of FBA for molecular systems biology include prediction of the growth rates, uptake rates, knockout lethality and product secretion. In FBA, the solution space is constrained by the assumption of a steady-state, under which each internal metabolite is consumed at the same rate as it is produced.</span></p><p class = "S4"><span class = "S2">For the quantitative estimation of the metabolic fluxes, linear programming (LP) can be used to solve the stoichiometric matrix for a given objective function under different constraints. The constraints of the problem depict the space of all eligible possibilities from which an optimal solution can be selected; </span></p><p class = "S5"><span style="vertical-align:-30"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMIAAACUCAYAAADBNvbbAAAWW0lEQVR4Xu2dBfQ9RRXHv3ZwxG5RAbsLC+soit2KioEdhAWKYBOigmLhUVHEVkzsRBELOwBFbMHAOnahng/cwfuf/+x7b/e93Z19/zvncPi/38bc+c7cmVt771kULRAIBHSWwCAQCAQUjBCLIBAAgTgRYh0EAsEIsQYCgTMQiBOh3pVwPknXlHSapLMbmf+2//Obufuv/XeqpO/XO5T6KQtGqHeObirpc0beyyT9SdJe9vv1kr4m6TGSriZpF0mvqHco9VMWjFDvHD3Ddvv97P8XlPRdI/eKkv5op8InJO0m6bh6h1I/ZcEIdc7ROSQdIenekpI4dEdJH5T0FkkPkvQfY4RDJe0q6a91DmUaVAUj1DtPZzP9IFH4Akl7SNpJ0uGO7M3tdKh3JBOgbAxG4Ih/u6TbSnqIpDdMAKexSTyvpC9Iupakq0j63tgErVv/YzDCNpKONSD9Mb9u2K5yPCz+EyT9UNI1JP1tlS+Pd41jPsUk+C0D//2S7m7ybsxHMwKIQ4dJer6kPQOo1SMwxonAKG4taStJH5J0yuqHtVZvZI5QnO8laTtJn1yr0VUymLEYoZLhT4IMbza9jqRfTILqiREZjFD/hCXHGqbTu2WWpPqpnwiF8xjhkpJw3ydb9oUkYa77l6STC2O8qKTNTJn71RwMCBO4rCmApVuxlGzhLCSEHFxO0lkl/VTSHyaC8bJkohM8T9LOkg5Z9mXxfBmBEiOwANmFHidpW5PlYQgmAZNnau+Q9HBJfzaz3ksk3cpd/7Kku0r6ZdY1ixvn0HPt2vXcLgcTXV/SfSxs4NOSbiPpYZJek73nzuZgmurcXl3SXSyeCC/x78yU7E2jiEXHWBgF1ravzBksG9HlJXHv4yWBUR6DdF9JN5b0cUkfnip4q6Y7ZwSO3vdmnTxH0rMs1uXzkrZ3118oiQUPUxALg1LHok0NoGEGThQYgJ3ct7dJ2tGsRueW9CVjqvye+5ml6RKSLmYX6Y/F9LNVg9Lz+xjDy035BS/+zQaA4YCWFvwjMubH0oZ4tL9tPjmZ4AujJHy4nvtp2OS+bZvbr80n8fuexzuJ1+eMgNiD+MPk3MmNYAdJ77YFTZwLu5QH/JHm7URk2lIS8S9YhTzYeErPLwkRBz8Cz5fMpwSR+biZn0u6g6TvWBTmSyU91mh7VOGkqBl4Tr+vGoEPlvRG+/dlHEMvG0DnGYJgPU4GolRTY2N7pv3AJxExSjPCsIlxYbeilUSQZNfmOmY9mMQ3/zyWjm+6i37nLzGCv44D6YaSfuuev4I77qckN+MVTjg8UdLBbkxsLifa790lHbQkNxOJimj7WTNVJx2P16Jjfd02PDYdTtZNvjUpy7MWMqC1uZ7Ltm0YoeR5niIjcNISIsHCKy1OML2dpAub+IPOsEx7gKQ3Zydyeh8BfYR3f9GiVpfpZ22eXYQRSkqaZ4S219swQunEmCIjPEHSi23V3FLS0T2vIIwciK/s9hgfvMKcQlwQXX/UMx2TeX0wQv9ThXKcnGBDxQr5zcJvVCl4Dx0wt8L1j0TFPQQj9D85KWCOnp4s6UX9dynMqBgXMEhgfv6U9YmhAasf1javNwxAUt1dBCP0Pz9ejBxKuffiZ7JCEdxIyPsUTc69z1IwQu8Qb2BYyD+q6at3RCD8Oyjn+BL49/HZ6dBX35N879iM4B1qCUC/m5Wue/kX/wWfKtbc/InQ5CNAj3iopAMym3/XcfmIVRxxmG75nBPdIFoBgUUYIfcD8Jp5ViPvoc6tSpjv8IAyOSXlcd51HHY8R3utJJjBO4xqm2hPL3SDp7fdw9gflfRKSQeukHj0ARZ/wgnnI985R1uQEVCwPmCufh5hF2OSEoiXsjAMFjiN8AoWY7J9Ex9DBgacRjTCMPaxyT+X2cuPdLTcw/pDecP7TBySv+6VPY583ruvPc/C4tuGn1Q8u2w27MQ4uGiIKegKeOHZMJ4taW8LrFslQydGwGfAdwx/rxij0UnLTwTy5pA+pNSwfqSPaUrXCYMA7KManiflCJNTaoRRcELg/k+7mL8PxxqiA4uI+/J2f0mIUbU2Im2fZoGGnkZCUO7p8hetkn6+C7+B+RE2lUjdzvjNC8Pu/OJ4sIgAp+XF7QqLM4/MXSVs6B2EpnDyRJuDQDBCLJFAIHKfxhoIBM5AIE6EWAmBQDBCrIFAIE6EWAOBwJkIhGgUiyEQCNEo1kAgEKJRrIFAIESjWAOBgEcgdIRYD4FA6AixBgKB0BFiDQQCoSPEGggEhtYRiLgkP6qPguRvkWow1mI1CPSpLJNI+D2WScFnw+O7BfJ81v4NQTWTFIT0j0CfjAD1KQ8nJY9SGhPi5PlqirSEQ6Q26R/F6GHyCPTNCOkj8nNmtdJebSkJPzN5BGMAa4FA34wASNQHRjRK1SBJPvWuQnLatQA0BjFNBIZgBIp+8PE63zzzeSJZ1ygwkjJRTBO5oHqtEBiCEajQQp5+Up9TE4DvaN+6VijGYCaPwBCMQPZnSkCRae1jLs3L5MGLAawPAkMwwtaSTrJCGOgJkVVhfdbP2oxkCEa4qiUEI08/jrVogUB1CAzBCJhO/1ndyIOgQMAhMAQjBOCBQPUIBCNUP0VB4BAIBCMMgXL0UT0CwQjVT1EQOAQCwQhDoBx9VI9AMEL1UxQEDoFAMMIQKEcf1SMQjFD9FAWBQyAQjDAEytFH9QgEI1Q/RUHgEAgEIwyBcvRRPQLBCNVPURA4BALBCEOgHH1Uj0AwQvVTFAQOgUAwwhAoRx/VIxCMUP0UBYFDIBCMMATK0Uf1CAQjVD9FkyDwXFZi4O+ToLZAZDDC8jPHIrimpAvZJ6mkq/mRfZ99dknbSTpK0j+W76qaN5DE+U6SLi7pOpIeKOm1kh4p6b/VUNmCkGCEFmAVbt1W0rst0XF++XWSyOp3K0mXl/S75bqq6unLSHqapHu7sU86qXMwQvf1dVvL08QbjrCFcbKdDGT3O9he/dk1Tm95N8tQwlDJZPi97nCO++RQjEA/L5H0rDWpi7C5pC9IupqkD1qC439nU/l4Y4bn2rjHnel+ej9A0lMtfWfKbdtPTz2/dShGQIY8yHaNdSgQgkh0jM3NkxvS219E0qmWABnxad3auSV9SdK1JL1UEow/2TYEI1xX0tck/XqNGOEBltaeid9Z0iGFFcBC+aqkHSR9Z7IrpJlwRKET7PJdJb1/ymOcxQhns4V7AUl/kvRT+z9Wkr8uOGisKd+ye2EELAxJaZyyFYWJf5+N6+eSYPbfZJikHRNleR1OwXzK02bA2kAsYn1MtjUxAmkaPyBpK1vIHH+p7W5izrxBUzqKyjilNvXT4QqSvu8G9mFJMEeuJ7CZnCbpPJIuZRvL7SX9xZRrb2pElNrD3kElIcywtTbWzWGSHmI5ba8uibHeRdJtJLF5kvSZqkiTsJaVGAGTH0c5aRpvLunHkrCHoxTtO0MUyCeN04BU8PuYUsnO8TirkcD7qKM21VSQ4EY5LBZuaphLH2ULP8cCU+P+7o+cIijaYJLajpLeZD+a9I5aGAM/wldso4RW6MZXwph8Y8O7tqRf1kJ4Ex0lRsD09w5J+WRwLzsfsiCcvmhDkdpVEpN/5RZi1aLvH+u+85rlyJ+WGASeIuk/DUTtJWk/u8Yuyq6ZGr4GHHG02p1T20g61mjdxTa44yQ9w/5GPQzMy7QXGiZjzdNC/ZYYAScJdnHEGgbzN/cmjvUbS3r2Qm8/46bECFMXh0pDxrGEQnwxd3HvbPf3z3mRChGC6kG+IVocKWknSYe3wPieki7d0quLDvNOO/FbdHX6rU+Q9GL3EOXBOPXSBuCZmo3zHg0nZdt+e7u/xAjeIvJFc5//wCjgSLxktpPNI26dGYGxo0/5nZ2/EWpwaAGYzWwnRYQoWZsIx/h4B+cUJ8jD5k1E4foNJX255XNnlfRJY37GcaIkTjevH3nTKoxw9xmnZMvu+7m9xAi5IpiONzylp3QgY90ZAUi8XyFBdKVMoebvLCJ2fOJ0EBnQu5LCzFx8wk4YxKs2DccW+pdX4Oc9fz2z4n1z3o3ZdcoD/8L9DYsRYpFvSc/kpHyLpAdNkREYEBOF1Shvj5CEUtgmsGpTYARw8iEX/G5SeNGvWLT5Tsmu+aoJ+FrSqcUYmxxpyXfEPegQbXTKlny5mttn+RFQAt9jlgHfG/Lgni2YYV0YAUsX3lMWa1PlH0JIkv7UJBsn+RodDB8DIkU6hUt6wyIzfT97RxvfzPkkvcHKei3SR7onhVXwuxRf5E2r3HMLScRbVd1KjIDNmx2f2HKuo+hQF9k3CgQeveDI1oURkHsRPdggmhxk3tv6NkmYRHMLUjJGJOMBxghCFXDQPXNBTPPbuuoIWH8wgy7avOzfND6PwSTEIgZfYoRHm5cQU2lqmApfIwlFmoY5lNrJi7R1YYQtJSFPYwL2MrLHwNvXm5TEZIygzvQNJB1o70ynwyKY5vdwCt2xpfMKBRcROHn+F+nX64+lsAp0IOKqiErFR4IPIZmEF3n/aPeUGIG4ckykeRCVt3iU5D6sSRyDP5GEtSk1zwg42P7YMFp2WqwQWDGSlWo0YAodp3CRJosQj3j7elMMEpYaTgAaFiIWCwonAXq1N2pmv93ixko0P9bFXd3ZInP9mFDQkTBomG7bKuq94TPLj5A7fOB2FjiTjT/ho46qFGCWPIuEVxCmTEvKIf9OH29gZXmMuegRHfxOU6u/ITkaoe9mcyxC7IZNJ0dulYMR2uzKvS2GOS/2sn9pjljgKcoWf0hubOG0/G7mc8nX2FhjK4pGSYZlMrGEsHuhKLIQkPk40vPYc+9AYTCEGiBK0bCGoHSnRiQqO4NXDDnWietPDZEBR1VNzTuRwOZeFlaAskscEQ4mdkzP8CX6PSNs7z7uqWmsJVr8ZufFnvNL2k0S313AIHewaOP8HVsUAvOqsSg1OdQw/REHhIjkG7IoE56LNzDKp82ezv3emsA1TK7YkmmAiCcUm3lq3jZdYrSxFwk4oRwSQEaUpY8bymmbt7iTMomlBzFjSg1P+nMKzjvmFN8HVqimyGQw5NkUhsG4m8THwTEpMUKKmIQYPkjHisRiJhpyVsFwwrPREwCCncE33smpgSXqV4UoTe7l6GR3IZy5xsLknI6YARkDtaMRAzkZibRkt+QzRTYDH0hXmlBwItKUzzqn2pgrnGWsH8bbNKdN40sbH4aDt9YAwhAf5tQwzqChLgQIByFSYZYFblCKgxEGhTs6sy/2EDPbBhb2Cl4wQq/wxssdAoiE6BEo1bNM0KOAFowwCuybZKc4JDG/Y3r/em0IBCPUNiNBzygIBCOMAnt0WhsCwQi1zUjQMwoCwQijwB6d1oZAMEJtMxL0jIJAMMIosEentSEQjFDbjAQ9oyAQjDAK7NFpbQgEI9Q2I0HPKAgEI4wCe3RaGwLBCLXNSNAzCgLBCKPAHp3WhkAwQm0zEvSMgkAwwiiwR6e1IRCMUNuMBD2jIBCMMArs0WltCAQjdJsRUpPwATpZHUh4S2aGtukTu/UcT/WCQDBCN1jzUlBk7SBZVV5QsNvb46nBEQhG6A75OSyBLqkqJ1EVpvtQ1//JYITuc+yLYVSTqKr7cDbtJ4MRus8/KStTtj7qR1eT0Lb7kDbdJ4MRus99KgpSKhXb/a3x5CgIBCN0gx39gIo3WIrI60qenqaSst162Pgp0iySbtJXOV3Vu4d8DyIlRoU25cd6py8YoRvE5OghWTGN6vMkv+2jkV/0JtYHadd9uv1F+iOXLCZeRLcdJL3epW5Pz2P+JUEzBT0O6akM7FZWJotUj9TIIBFyU9WhpnFtbrl1qWJKNnVKX5Gs2m9AbBZUc2Jds1HlOXgbMQtGWGQ5bXwPFWHea39edY5/fBToHxRupI6Eb20YgVOLclSkaU+tVMjcl51q8/5ZyJH0mcVOMi+y2/k61F3qXzRVes2rj6YaFtDWSm8LRujGCKkK0CpS2LNotrbFv3uheCMUUkiEehPUpyA1fZtGpm6UepgKeilM4rON+9oUy5xuZE1nU6BSDvpTqVFcBsajoMhpbQZh91KOl2KGtLxYTar/TAmu1vMSjNB+NqgnR3kr0sK/zEpstZV3yQNKKarb2fN+x0wUfUrSG62498/ak7nBE77aDztlXgOOmhcUQinVTJ7VNeIKehLFYCj6UWroUJyeVFBa1uFIlaVXWip6mM7j4osYMi8UL1m4BSMsDNWZN3rAqZqTyiW1eVNTFUwWDeIMVYqoObCqhlx9vL2sFArCDv7gQiWkWf3n5cLSvdRLQNegAtI3FqgXsegYWatUXkIs9aV50/O+tG/reQlGWHQa/n/fA22n5i+INElpbvOmEiM83Wo4L7trluggLgo/BydPvkhQMBG3SlUyZ43Jn4z+PqohfaQH65avwbZ3VrWIwivH2fg6VfMMRmizfM+wRhxmVpwTbQf9V7tXnH43zHSQUyLRAQjVoFFjjtK9n+nIZCVy/O6de8EpIM8ph3jTxgRMFSV2YRg4tVQfj99HSHrTCk83X400r/NN0CPp5mmd5iUYod0qRiYmpTnmwOdL2rPd4xvdfTlTkjEr5hYibsbC8jwrOMgEU7iwS0ORPNLqKu9l7+Q9LH5Mv8tUrqGMFiZRKmk+MbMQJVrRd+gH5v5xlwFYbW+MFGDiS9vmFVnRIaj116oFI7SCa4M6yrnVot2bNr6b+nOUrcWmz6IqtWVK0SZLF7I7YhC+kJMK1pdlxrGIBQxREoV9Xq05T4fXD6AfBmZT4FSiVjWWotR8tdaFxxKMsDBUp9+4qyQWFJOYrBbI3+xSbcSKeb0iD+M0Qp6niHdqXb95YJ4PtWqY6Cd7SDrGzLH7zSNmieuU3eWk88zdxY/g9QOcaC8ymrB0sSEda9+E8O6SVWzuEIIR5kJ05g2IF1iIvNUCJQ2RBRGnjacURROHV1Njt0t2diqbwgCUtKUKZdfgPhgKaw5MfKqkE9zOujgKG96JqDhrDaVxgB0nHjs3YgubSF6ieBYNXj/A007lne1sPsAGMZV5Sacd/hD6S2WAOWF57g+SDi9Z5IIRFl8C3kqCzZydFTMnDp62JVKbzKfzqFmmEHsqJE8fiCfXt4Uxr8+m6+Dx7QYH4Kx3EqSIYaDNxsEG8GanH+DDoZQvC/zzVvqYPjktjjY/D3ocYSMwCidGaiWn4kxu7grQuj7nGQGLCAvpXZLwdrZ1qFFo/L4dgFomBCIxAuLDjZZQWhPZm5nHmwXXpnURjXzoROoLLzyWN+/XSde8/uY95+n6RiJmnAiLTyGizFHOukMV+X06hgqwIyJWtGEgFEPMrG12Uj+6Hc0ri8zOTr5sgx5Eli5riB26jdkZMQilODU2ETYjmtcf+L29WdnSvVxn46FgfDDCsrNuzxNPc1k7ivtwfK2IzOJrWBC0rozUJ22LvJtgREJTTi446xgbzkLErr/MeBn+B0SqjQIlu3DzIkTHPYFAbQiw1lGUEQthhA18MsEItU1X0NMHAqzzAyU9qSntTjBCH7DHO2tCgC/iOAn4LiOZXjeiLxihpikLWvpAAKvRThZmfkpTB8EIfUAf75wcAsEIk5uyILgPBIIR+kA13jk5BIIRJjdlQXAfCAQj9IFqvHNyCAQjTG7KguA+EPgfO7x80W82sd8AAAAASUVORK5CYII=" width="97" height="74" /></span></p><p class = "S4"><span class = "S2">Equation 1: Formula of standard FBA.</span></p><p class = "S4"><span class = "S2">where </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAF8AAAAqCAYAAADVo4eHAAAGAUlEQVRoQ+2aZ4hlRRCFvzVnVAwoZkyYc0RRMa+YEcz6w4yoiCKKmEBRQcUIphUTq2BCRETEnLNgRgUx54SKmW+pXtre9+7c+97M3IG5/Wdn5lan09WnTlXvFLrWGgJTWpu5m5gO/BadoAO/A79FBFqcuvP8DvwWEWhx6sns+fMBiwf23wK/9DmH2YElgbnC5pvROq/JCP7cwAnAhQWIJwFXAH9nf98LuLuwOwc4r7Ab6DwmG/h6+2PAUsDuwKfx73WB3oPxuwfg4RwN7AK8AxwEXBZ2xwFXVyC+KPA98C/gnN6aH0r7yQb+5cBUYN2CZnYFHghw9gHeB94AVo2f/TQn8BqwBvAesBbwZwbo+sCRwI7AX8CawKHA9WGzNfBkfgATCXy9xabXzVHjHms3izdV9FsR+DAAubmwmw14AtgSeAX4CbgfuCSzE6tp0f8rYPXw7mRibNgIeA44PPawRADu2CcX47We4RrwvM6nAS60SesFQFX/5N2zgBCdDgBuywZYHvi4GNCbczzQb+40xw1xY6SureJgtwCenSiefzBQeuBYgp/A/Tlo56NiMj357fjbJ8A6hWf76SLgFMDv0o9j5S19l7I2iFt8FnBs0ND/lFIbtOOcVwHHxKr1oqPiupabqToMg9mvDU5rE+D5sJd+St5fLL6vVOHZ0tJTMcZO8XNaw/zAC3EoKwMfRJx4On72hudKqhXa0QsE33YIcGuoggY4DmS6UFx7PdZmgN0zgqO/zxPg6/G9aEXFsjDwcNyKtIiNgZeAtSNIK0XPjo/LBnUdkQXemYsfb8/Xu94MfldVlBp6IFQbdMo9126qkyQzxeLKoAjB3wyYN+hDBSNN2gT/LuBlQOoyQbMpS68JFeQebSkGrBDyVsk6UyTUBX8ZYAdgO+C7uD73AY832LimJwKXhtftMRqJSsP5Nc8TJylhm8z7U0CtGlZJqeTMm2rJQxGnXIIaAzwUb7s0dVi+55HANxs8FTg3rqKdBf/auHq559TBQbpxIdsDj9TpMEY2xhuTJD1cPZ4CYQ7+bpn2T4HU5ahgVGd5WySox/3l8tRgKwXdCRyYHfKMvlXgO6DZoBzodTT9NnnwlNWycp28WdeDE6eqt1UWn40RsHWGlf9fjQzX2+y+bDn46wGvx99zJdRkz2r/BYAfey2qH/h5cOoVmOQ0VYEZ27Y16SMPaKqaZ+qgVGHjpnauKIhVDS+HPxSlBJOp1HLwUyD1m2uX4w3W2huo/xly/T09v8zk9IDPi4m8DUq3R0NG1VlHDn4d+5FsmiZZjmdSd3rEHpXWLcUkeRIld38d3/O1jyn4uSJwoReMhELN7xaYXgSUXxuGShgp5lQNrc43/jRpevP0qO+oPMrWL4MdF/DlcwOhCsCWF5aabLKXrRuwxiEAcui7ww44Bv1bBT95p9zWJLDUxSGpHat9w5QW6s7X1C6BX5WAeXNULqPO+abFllNto8ZtGQKm2PKs6b2c+ltTdIa03zsSpNujpFG+XiXnKDPS0inzzHjgJZWcm4N/I+Aihj7hbHV5/UP9ayo+Xs25LXip0mxlhp3zekmL0rHZuLLaQL9KlJ0tKfgg0zT2zFhAFfj9aMda+5nAxQPKPL3mngDA/MFEZKDFNzy1XKvbtUz00vd++063wr6+bv0R8TGXpI2WVIJfykGzv7eyERcM2vDhI09OGk0aWW4qrtnXW6Du/qLmQ4p9VDtWDuu23PMtK5ifpJcoHereUEHlntP4qTSSz9ezYFZ3Qb2k3n7AHTGAydD+scnNAanId06v7LB8vSkg9yYaqLvmZDeIzk8ACr7PiWaeKjrfZvVm6z4eQq9mvmNWnNoZwPlNF53b99PZKTCVY1vZE7DRigN6nG+b+2b1/br7MWgLSJM3AMdWad1UTGL+4aO4//ZrYmVdx8cU6dL/ATEUDlVJjuXU5SIu/B6vN6kGUhegiWonfS4de/N/GXzZYKGrhSIcCnjnGybDbLDezrQXAh34LfpFB34HfosItDh15/kd+C0i0OLUned34LeIQItTd57fgd8iAi1O/R/M7V86pTYNIgAAAABJRU5ErkJggg==" width="47.5" height="21" /></span><span class = "S2"> is a parameter vector that linearly combines one or more reaction fluxes to form what is termed the objective function,  and where a </span><span style="vertical-align:-5"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAE4AAAAoCAYAAABQB8xaAAAE/UlEQVRoQ+2aech1UxSHn8+QKVMhCpnnKSFTmRJJoc+cWWQeMsscmYfMZcw/xlBIyhTJLLMilJIyj5G5R+u87fZ77rnnnHuves939l/vvWevvdf6nTX81rrvLPrVCoFZraR6IXrgWjpBD1wPXEsEWoo18biFgF9a3tM5sSrgtgTWBFYA9gZWAtYB3usYCksD2wGrA78B3wDPAB9W2VkF3KHAtsC+ccAnAdyvHQFO248Crg97rgS+BS6Kz5cDZwJ/ltk7LFTnBV4D1gOuA44H/ukIcEcCN4YtmwEvxd/LAE8CawGCd1qZzcOAWxHQ01yzgQc7AtoawAdhyzHADZldmwMvxHdbAc/ldg8Dbnfg/hBaOQFxJuOnzbcChwA/Rfr5LDNofuDliDQB3DoP2SrgfGb8mwdMlBaGP2YyYqH7ksC7wFLAY8CuA/LYCcDVITOtKFYBJ/14JWL9UuD0DoCmCVZQc5jrcOCWAXal0TYtnKuAWxd4Ow7dEXgCWC2qrKVbTndPosT/jav06ABgVWB/4O+aClwSCd/tRycFIhdP7X8E2A34q9hUBdwRwE3Al8AGwE6RG/ILTgKuqqn0qNvmA7YAzo6843nTjBpyiYXA9DMMuFWAj2JfbeAE9CFglygOhqyleU/geWAH4M5EwUkXDimCd19TAsp5wAUNaNK1wLFxjtXzxQFAp8DpPFbi74Z53BLRIZhAXVafTYH3k0tSl984+N6oHpXKzwNsBOj5B2YHa4j86lHg64aXpsBV6d0KOLnLs4lCJtSnMwX1xofju3ECZ9XbGfDFFC+uuPo24Gbg9QYeluM6UeCsoBfHjVKS40oUtYsoQscc+FbDN59vXxy4IvhV+kxvPyXI91cj3qF4ClzaMeRHpx5nE7A+8HNVqBoieptJ2FWWv8yBEmO7iTT+9Ra9z3L/e0MjUzZfiO4D3DuCd5Wp0LY4yPemKndZVXUa8mncaH+qt+XLqUmR7ywWDgNshg0lGXlV0h2E53KRJ/PwvAy4a4xTmYOB20OJKjqiA1gUXecDFqGpVQbcsNyVdhQelIbpXhEKjqSKUt7E8fR2w+fE4E2p7BuR9/TmqerW5PDYmwJSBdwewH0hU/DYSuCKHGBc61l5yKUNsDTg3BbK1xFZHjBULRL5GsUL0z60qpUsWIOpaO28eucetyDwarRZZcRWmmLzK2u3+d0eSOdzElQV+6EOMjX3LBBk99SE9BaieuFZwOM1zyq2peEq5bFKpysFd1qYujEHLm0z8v5skUj6urqKSkiLKiMTd/C5IXAy4FBwEsv2aj/gnORwc+w2aTtU42KBcVSkLbaNnjnVTkWX5ACg6Jq+yM/MgUvj2sb+jKhovhWrm57m977lfDJ6EHBHKOPwc5JrYcC8Yzh93gI4dXNk/lREl12QL0PqY362xXKV8df/HuTA+dkEf3eJ1SZKud2bJc+UEzQvMh/8OEnUkrPnBhaL3wnaXOkLuLCEOWirY/OPBx06qMlXoWUB84ueZVvzfYVmkle9zB85Dhsz72oDSFMZ9V809DY/V9la6nFNLyz2bxJFo0vj9Uosho3O6wJZTEvNgQV5ris7I/eNA7iiRbMdsYNwcmqOqDtYnGOBK7jfA1GV3olp8YwEpK7S4/C4lCzK9OVFnV/jAE6QrL5zzUn/WzIu4DrvYbmBPXAtX3kPXA9cSwRaivUe1wPXEoGWYv8C3Q4JODhQVyQAAAAASUVORK5CYII=" width="39" height="20" /></span><span class = "S2">, or  </span><span style="vertical-align:-5"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAE4AAAAoCAYAAABQB8xaAAAE9klEQVRoQ+2aWai2UxTHf58hU6ZClAtT5ikhU5kSuUGGD30XHyLzEGVIpgjJkPHCeCFzKAmZyUyZhVAoJfMQmftp7dO2z/M+z/vu85yL83r31XfOs6f132v913+t881jMqoQmFe1arKICXCVTjABbgJcJQKVy0bxuGWAnyvPGbtlbcDtAGwArAEcCKwFbAy8M2YorArsCqwH/Ap8DTwJfNBmZxtwhwG7AAfHBh8HcL+MCXDafjRwddhzKfANcEH8fAlwBvBHk71dobo48CqwKXAVcALw95gAdxRwbdiyLfBi/Hs14DFgQ0DwTm2yuQu4NQE9zbEvcO+YgLY+8F7YcixwTWHXdsBz8bsdgWdKu7uA2w+4OxatnYE4l/HT5huAQ4Efg34+LQxaEngpIk0AdypDtg04vxn/8oBEaWL4fS4jFndfGXgbWAV4ENh7AI+dCFwea6YlxTbglB8vR6xfDJw2BqBpghlUDnMcAVw/wK482qaFcxtwmwBvxqZ7AI8A60aWNXWr6e7ILjFTXKWC58OYJyKMZrpn0/qLgvD9dkyWIMq5uf0PAPsAf6ZJbcAdCVwHfAlsDuwZ3FAecDJwWQ8W5oTtdmcDtwMf9rB3voWJQPrpAm6d7OyhgRPQ+4C9IjkYsqbmA4Bngd2BW7Lb9JE4VgeeDqGdG/ooIFXojX1oyCuB4+IAs+cLAx4mB07n8WG/7fK4laJCkEAdZp9tgHezQ3KX3yr0Xh/OsRmg+E7G5XueBNwDfD6Dg3Lg2u5dBZza5anschKqvJMPvfH++EWfwKUzVgx6kAbSA6ZvdwFXAK8MUvYtwM4qcGbQC+NwJcnxDerZKsLLO+TAN2bgBV1LtwQOybgpzTcS5KtbuzbIvufA5RVDuUXucRYBRsJPbaG6WHjb9jGpib/kQIWx1UQe/2okvc90/9sIxgw71f3VXp6Rhpy7c57xOjarTQ7qvb/agLMb8klMsD7V28ph1yTxnRe3GWAxfGMo8jbSHRakfN6ykZAUpelB0/dRa2g996ZY3CZHfByTouNc4Jz8Qk1ypIu78oqiDNP5gKFgS6oPGaFetDtzVgPa0sltwGcjvkQOSBtw+wNyqSPp2KmjmoBLHGBc61llyOUF8Hmht0a8e+v0pUPd25UovctkoL4zcdVKk7wObSslk2qQijYCvmrzOC/t5WypNAlbZYrFr01Ni9/dCgOWALzY9xVILgcoN/4TErGPOu5m4P2KfZuW5OFq4nmtmJSDOy1MnVt6XF5mlPWZhkn6uvpDIYZTljGzqb22AE4BbAqOOsrKwZf2Dg/PQvklMLaKtMWycUGRXKySTEKpavqiNKYELo9rX/n0kCG+yp3haf7+zAb9tDC8olbTJeAke5OMdfJsNk1tmT8e0WUVJI8qb+RnSyxHk37990MJnD9L8NaI5ZAo1XavN3xznaHkQfLBD6O6G7AosEL0/CuWVy0xW5/foBy01bb5R4N2HVTka4S141LhWRLjdy1XU+XbYvePHIfPsqdUIdSxyPsvH/eWn9tsbfS42kttHUljnNrrrVh0tc6HBTJ1S822STwPu3ZOzusDuFSiWY5YQdg5lSOmypM5iUzHpfsALmk/2z1mpbeiWzyOeE3Z1AdwuVg8KHTRWIPWJEdqDTb7LvJ/+r8lfXhcLdhzet0EuMrnmwA3Aa4SgcplE4+bAFeJQOWyfwBFUQs4ailzJgAAAABJRU5ErkJggg==" width="39" height="20" /></span><span class = "S2">, represents some fixed output, or input, of the ith molecular species. </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAH4AAAAqCAYAAAB1PO9pAAAITUlEQVR4Xu2ad6x8RRXHP9gbIirEhhELYkGwF4gFFY2oIIpdQRMrdmMFFNSoRFQC6B8qYAVbVAJqYosNKfZulBZ7V1BjF/Ih55DjvLu7d+8+Hm/ZO8kveb+9M2dmzvf0M1swjpXkwBYreevx0ozAr6gQjMCPwK8oB1b02qPGj8CvKAdW9NqjxsPWwFbAv4HfAP+ZIAtXBbaFiwLi3wN/XWaZWWXgtwGOBh5ZAPwL8GjgE+W3KwPPBt7QAL0P8NFlBX9Vgb858BPgncCrgf8CLwCeE0C+HHgdcE3gS8A1gAcBfwcOAp4S8+4MfG0C+ArMVYDz4ruC9qfGovib1uOChsaOwM+Av3XQ9ts5wD8n7OuZtV6e9Upx9j+2c+cBfgfgxsEkL/PTOLQ0dwtGaio3+7gacCrwBeC5DdMPBV4RF1A47hlA3zYY6acUGv9+FfDK5sJPAB4F7Ak8BvgY8BlgV+AkYG/gpsAp4TpaGtcDfgV8EtCq/KPQvytwGnBUEVI/Xxd4MbALcH/ghvHvjFj7QuBN9Zx9gN8OeDdw7w5EPwecCTwVuEcwdBHgldAtQ7jUgiv0ICZjujRj0tK9AgzB/H4zKZnuzx8C7hVAKSg51Cj/f2vg44D0tBg51HKZ/HDgPsDbgJcCCsRtgAOAj4RL0eKovS0NBef9Ab501N4E/efA3YBfNGe/fgiYgnFsWKynxz3W7DEL+JvEwdzjt8DDgG+GCVMblOb8JiN/1wOorim3AvYPqZ2XRGrR/3oufD3wkgmCKj/eATw5aP0Y8F6azgrs6cDtigbXvS8XwCrAV4z44LuhqZp/3cZO4UbU7DXaGBsl+B8MxTsZEHTdy6877ppYvQu4dliLawUmc2m8TDgO2C9A14x40DrUiM+H2VK6K4P64KDZPTg0os/8rjnzAF+B1dR2nfmxwPtiI7XucUAFVo1O4I8HHt+4C63Gt8OMqygqR3UP/n0W8FDgxBmW0vVaB4earJuY5E4fEdrtXIE3nrgf8Gng7rH+Yv5N03gDj+/FBQxwZFI1aRJRuj8LKNGtv5wFpPQVGk2mQ8k2yv5BmO7LzyIQ3z1T9YOzlj0DeGtMemNYmQrsXQJYp0wSKu96RGigPvXccoZktne7bwjNvnG/A4HXxt5anicBBmuC1DXU7vTTuhWB70o3xVHePRNQcE8IYsYf/rZmj2nAO/mHQWCaVunDvhIR8iym53f3NRXSt+lClH61aCOGZvY7ZaPWDFbt7NJ483mD3B8VGt4hmZuuRAup5qeVeXCZk1ZDIW8tSpI1GPwy8NUw9QZ0XQGf8407dMFaZGMxhUM3o1UzezG++D9XOA347YGzy+XWmIv4JvBvjwP2Bc5A0EOZN98xDtd37XrMex7w5kKopmW6H5mdwdsTgVsCtwceCDwk1qlhBnky3NTLgk4C6r0SgARFy/i0cAs3i6DYmoECf/Um0EzQ9el3CvPeFfDlFe4AfD3csoG4I32+waGKeZ26xzTgvYTE0hQr1YLkYRYZ6R5kzKTAZhH6fdZ678OAF8Vk07hD4u/qwyfRkhe3AM5vJqSVNHJPd6LAfCOsmpbTkf7YeoAZ0R7An+PbjUKQKui5TYKvoqUQ+S0F2TTRCN6R2YuRvYGzdYiLXcqsqF6fUqtTWgB94B/6cHfCnNQo00QZ9csFaC2ytLqbmpZV4AXYAFaz3gqE7imBzHMkoNY8NLEOK4NmCqZyWoYKivw0TbOIU4dxg26gTdmcY33AolEGeXkPU2HPlDFAxhreYec2E5gFfKsZbuxhJTS0Vp0+VFOf5nARABdZqx/WVNegqwW+anaN+J8FvGWBze0RqOVt1W4BkmuW6mZ0O2v2mAW8lCyimBLUAo6gKVHzRNN5qho8GflOKj32YYDNFStwLxvAQO9umfbwqKSliWyBrxFxPXs1533Ouqnm9AHeA2fN2qJFDnPYNdFij9tV5vWYPnPKpFRz2kLjFqtmBnUGT8YyOSrwbQq1csDLFPNuiwgGEDmMGLPAMBOhmJDMsypmdL/o0GLM63bM5QXfPLdtYFTg23RuJYEXIFME0w/70g61bfcpPewuUDPNMFaozY9FBWC91rcaX+vol2ngLShYLcqotGVoRov+XgsXfRlf08SuRklfOpfUvAr8tOLOZc7H2wwwr7WI0TVqgWMI8LUHcEz0ti/JyLa9g8Gq3TL78DZrDO5qVaveL6tvSaNWMy+tGsS6CHwb3KW0++LkixN2cI0tS/37EOAlW8umDwA+tS636UekLdnWnFsKac67XFHtX9Qytm1SFaXtZfQ70aUwqwU+e83fmhKxm39a0NDPz9MZa6+XNW1/t5Gg9v9rA3hgQeUDsY85bltEylxd4bcsW0d1Uwq9jStjHDuMXZW8DbjOsC1a4GvwYk1as9+OWue2++RjjCFDJlrnNyV0CIKvSKwrWwrt8wjDdT5S6KpwTTpTNdddr19srJgldBWparm50lfjN6rJNITXa9a0wGd/PSeqie8NULQG1pXz0aHaYHuy7wOISQe2UZFtxCGXmjePV6A+HLVs6/P6ejtZNjq8q+Z8WuPIRxpapxwb7aqG8Ggm8BbzNVs+HbKqVXP2uvj5wJHrAHrSVKhsWyoE2f3qe0FBUCDnEUADuNcA3qMOnyz55m6aBanFLDt37+l70M00r9V4A598SOk59Vs2F3y857Mh+72WWduXOJvpTvOcxUeK/rOxYaMkO2SzaNiTv0G8pJk1d1N+71uy3ZSHHw81nAMj8MN5t9QrR+CXGr7hhx+BH867pV45Ar/U8A0//Aj8cN4t9coR+KWGb/jhR+CH826pV47ALzV8ww8/Aj+cd0u98kInNvY62j/isgAAAABJRU5ErkJggg==" width="63" height="21" /></span><span class = "S2"> is a stoichiometric matrix for </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABwAAAAkCAYAAACaJFpUAAACSElEQVRIS+3WTciOaRQH8N9LIhNlQRYWzCjyMb7FhhnJjhFSGBSREhtF+chHiaI0FmxQQhqKpMxCFprkI99NEjVWUjbSWChCR+fW5e55533e0rvQ86zu677Ouf7/8z//c91Pmy7+tXUxnhbgN1e8JWlL0k4r0DJNpyXrKKE9SXtjIIZjOm7gfB72M5agBy7gagEyBMsxDOdwtk6gEeBBrKsFjsUDbMD+Yu9lknqFhfiznbwvrxsB9szqLmEE/s1D1+IAZmMcdiEAo6pZqcAiDMK+RJiE2yWJ9iT9AbcScAvuJfsJeIptCRhS78RfmIKbWIxTCTIGD5sBHI87GXgUK1Gx7ZUHRy+r3xxc5PPn7jiWpjLRiv+aAVyPP4rAZTiR66FZZbW9Fbtz0TfV+BFBdBU+dgTYDVcwICUN5nPxIRMX4Bj6ZBUj8Tb3ord383l+OvUrHzXqYYzDi4wKU0RvnuU64o9gRa6n4e/ixDU4nOufklCHgDNxOaPCbRuLjJDselZ+Db/gfUEm5i4qe4JReFcbk4b/afZiUwZW81fljS5cF248XRzYD4+zFTEy2+tgsa5LWjowWEZ/qgoivpIsnBeWr6SOvcnp3niekTdQjM1q3K/A64ClA2PQDxUsIzaut99Ql7MkE32fiB3oj3kl6Trg74X9o7pHBWAk/5OSbcaeGpnSTLEVN1So8Ob/xqLqX1Twa63pZfVT80Ivz4oL/WS+OJNSvm7GNI163cy77hicM/m8vYTWF78ZKTsV05K0U3I1E/z9S/oJWihvJesRNhMAAAAASUVORK5CYII=" width="14" height="18" /></span><span class = "S2"> molecular species and </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABUAAAAkCAYAAABmMXGeAAABrUlEQVRIS+3VT4iOURQG8N/ETmEiO/mTaZgUFmyYlA0rpFlJWc3CWCo2StlISimsxGIWivwpSayEEomyMEkRCbESJZF0dF69bt/n/aZmdu9dvefe5zz3nOc+9759pmH0TQOnlnTqVW01bTWdYgW6WWoGFmAAazEPB/ELS7Abg3iCE/her6sT6TZcLYrfi9PYgUvF2ijONJFGlXNxI6sM/CKsxHUEyWdcSKJqw7+83dqP+XPZZpAfwR0M4x4W41Wy7MT5pkpjvR+PsBQXsRF7cDmTR3I+wuV43gvpOjyoAQ/jUMbRxUmM4WXK8q0X0v04msC3WIGvGc/CQwzhGAL7z+ik6UzcxvpEbsatWlZs8CzjLbjZC+lCvEngXWzCz1riLozjS+r5rhfS7biSwPBl9R1TdVd02vBPWtl+/RA+5iF8qlUyO29RuGIfjuNA6n2qwpWkkXQ/D+FsGj2uZjXCPhMZbMCc9PMqfOhGugaPc3ErrhV6LcOLYi4In/7PUpWpo/XVeF8QxGMS3owRmDj9eFQaLVViynh+vg2v8aMTuP2bNkk4+fVW08lr1pTxG5HzSyVHoMLwAAAAAElFTkSuQmCC" width="10.5" height="18" /></span><span class = "S2"> reactions, and </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABUAAAAkCAYAAABmMXGeAAAB8UlEQVRIS+3VS6hPURTH8c8NA5SRjAw8yitkQAmlSGRi4JmpieQxMEApZYKU5JGJgZlXYSYlKRl4DNxbDAyYS1FIoWhp3Tr3/z9nn7/77w7UPbOzz17f/Vtr/dY+A8bgGRgDpnFoz1Wdim+97C7VdA0WYhZ2YQ4W43UbuATdg3XYnZB3Cf3eDzRiJ+ElluIiDuF3v9DZCIXxbMWdNmB8b/PpNtxO0NzKAUV2CRrfLmEf3mY9f/arNCz0HItwBkd7AbalvwRDCdqEB5iXbpifnr2Bh52HldLfiyv4gGXYjKs1ag/jXHW9CRrrd7ElGxVlOIsdeIKNuFYBjWhiE3R6Ts6MDPyClXhTAZ3GkXxfkX7++9oEXYvHFcB6POpIPbK49y/Q6PSpDAhbHayZpJiu87knaj44fGid0ompcnXB9BEXQxFTFo1cgE8laNxK73NDzHuo7Hzi9hqubzQuLp5fJWhjrTKoOmmxNCL1pkZdwIGc81D0o0PmKjzNtZM40Wb+KXiRo9llaoTVnuWFHeAN6LpfOxtVHc39uFxRMS1HMjx5Pwfha029u3y6HbdyY1wix9JKy3EzFcb68Wpj2tIP5TtxvUZBHBbefVWnrrrWNFETMBOTU9FHfG6DlSzVa2zjvrbfyagOGIeOqmzFoP+npn8AEhhYJaZqT+0AAAAASUVORK5CYII=" width="10.5" height="18" /></span><span class = "S2"> is a vector of known metabolic exchanges. The output of FBA is a particular flux distribution, </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABMAAAAkCAYAAABrLwHZAAABX0lEQVRIS+3UoUudURjH8Y9oEExGUQTFMNwQg6aBICjaHIjBNmRNEJNhYFOEFf8ARRCLGCxGm1oGM4rCwKAmi8GwMKbywLmXlxfv8V4wjXvaeV+e7/k9v+d3Tot3XC3vyNKENe5m07OmZzUcKEejFd3ow2dMYwp/CvVtWEYXtnBV+VeGTeC4dPAnXBS+fcBl2p9iHP9iXyu0Reg89guwdpxgFNcYxmMOFgXnGMR3bJTU9uAWO/iG5xwsFG9jAUf4gqcCsAKbxWEtz4oCwuTN1Er4VhzCEhZTi9XvuYs+hwPcI0x/SCcN4DdGkhVVATlYROMswaIwPAovf6ZD1stxy8EqCqImJvcLPzCDaPtvI7BinqL4YxpKTPjutUuQU9aZ1PSnIE9iDBHUV1cOVvFnKFV+xW7uYaoXtoq1t164HKwjZWwPK6XQNtxmFPTi5i1F9dyAehl1hbYJazy0/5FnLz2AOSUYu6BwAAAAAElFTkSuQmCC" width="9.5" height="18" /></span><span class = "S2"> , which maximises or minimises the objective function and stands between upper and lower bounds, </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABUAAAAkCAYAAABmMXGeAAABt0lEQVRIS+3VzYtOcRQH8M/Iym4ksTCbSY28xIIFpcFmpFBkwc6G1GwoCzV5+QPkJathJUWKBRaSGRFiYaemSYwoZaNYaLx3dB79ut15nmdqZuW5q/v7/s753nO+53u6XWbh6ZoFTh3SmVe1o2lH0xlWYCpLzUcP+rART3G58u2lWIlJ3MO3xn0d6W5crxBsx60Kdgn78RHxgc/NSONuDu6jH1+wHO8K0gV4iYUYxgH8bkUaHUS1u/A4yX8UpHtwLc8Rc6PsYipNuzGWlZzEiSJpLh5gQ2K9eN0O6To8y8AtGCmS1mf1AY1jBb63QzqIczmE1fiQSaH1w6LKUzhedWRd+4HdxA48wmY09NyG2wXJAO62Q1rqeQSnM2lRUXFAYaWyi3/cdZWuxfOMCP3C+A2LjWJVuqLaRVPSUs/YqE84hn2IJXiSrjiMM7iKs4n/Ja5WGueLuSl3sBMHcT7Xdkkx+bDUMhxNGb42M/8FHKqIH8OK1vfiSnEX2xZyTLSy1BDCKo1hbMWLPK8p3t9jE161M/2IiS35ibflTmfyYszDG/yqEtZpWhczbazzi562ZC0T/nNN/wCn9VElAUnGFwAAAABJRU5ErkJggg==" width="10.5" height="18" /></span><span class = "S2"> and </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAkCAYAAAC5fwuBAAABUElEQVRIS+XUPyjncRzH8YeblAy3sTmT/ImFus1gVmLBYpFIsRnJZLmUupvIqoymSxYTkmJjYCG54QySOnWntz5fyf38fp9Mymd+P7+f9/v5eb2/Vd54qt7I+ejgZzSgHs0Yxzf8CKHl5PRg84X1r9ipBBbMAmbwC024zgE/YQvdWMMw/uaAdbhMV49iuWijUgCez9mBw1xwFnM4T2ZvcsBqbKMTK4hW/+WAX3CaCgeTnKfXKTfjANZTZTzD8fM3LQd+x0S6tRV3OWAN9pKQJUy93NvXbmzDUSruxUYuOIJVhP52nOWA0UVI6ccJYr77HDDWaR+NmEeE4L9TasYu7KbKiFyEvDaF+7ZcAKaxmNYo8nmVdnAymX5kS91YvF+Y7EsfifiNVYpcAR7gAi1JUMUAFBsRHf3EEH7nWI2aMPonrVMpqR/+T15SSs6v452BD9siPSVyOh/7AAAAAElFTkSuQmCC" width="7" height="18" /></span><span class = "S2">, respectively.</span></p><p class = "S4"><span class = "S2">There are multiple different variants of FBA which will be discussed here:</span></p><ol class = "S6"><li class = "S7"><span class = "S8">Standard FBA</span></li><li class = "S7"><span class = "S8">Sparse FBA</span></li><li class = "S7"><span class = "S8">Metabolite dilution FBA (mdFBA)</span></li><li class = "S7"><span class = "S8">Geometric FBA</span></li><li class = "S7"><span class = "S8">Parsimonious enzyme usage Flux Balance Analysis (pFBA)</span></li><li class = "S7"><span class = "S8">Dynamic FBA</span></li><li class = "S7"><span class = "S8">Relax FBA</span></li><li class = "S7"><span class = "S8">Flux enrichment analysis (FEA)</span></li></ol></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S2">EQUIPMENT SETUP</span></h2><h2 class = "S3"><span class = "S2">Initialise The Cobra Toolbox and set the solver.</span></h2><p class = "S4"><span class = "S2">If necessary, initialise the cobra toolbox:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"> initCobraToolbox</span></p></div></div><p class = "S13"><span class = "S2">For solving LP problems in a FBA analysis, certain solvers are required and can be set using the </span><span class = "S14">changeCobraSolver</span><span class = "S2"> function:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% solverOK = changeCobraSolver(solverName, solverType, printLevel, unchecked)</span></p></div></div><p class = "S13"><span class = "S2">The present tutoria</span><span class = "S2">l can ru</span><span class = "S2">n </span><span class = "S2">with </span><a href = 'https://opencobra.github.io/cobratoolbox/deprecated/docs/cobra/solvers/changeCobraSolver.html'><span class = "S0">glpk package</span></a><span class = "S2">, which does not require additional installation and configuration. Although, for the analysis of large models is recommended to use the </span><a href = 'https://github.com/opencobra/cobratoolbox/blob/master/docs/source/installation/solvers.md'><span class = "S0">GUROBI</span></a><span class = "S2"> package.</span></p><p class = "S4"><span class = "S2">Setup the appropriate solver for the machine you are using by removing the "%" (comment) sign for only the desired solver.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> changeCobraSolver(</span><span class = "S18">'glpk'</span><span class = "S17">,</span><span class = "S18">'all'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% changeCobraSolver('tomlab_cplex','all');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% changeCobraASolver('ibm_cplex','all');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% changeCobraSolver ('gurobi', 'all');</span></p></div></div><h2 class = "S3"><span class = "S2">Model Setup</span></h2><p class = "S4"><span class = "S2">Thi</span><span class = "S2">s tutorial will use the generic model of the human cellular metabolism</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAABGklEQVRIS+3UvytFcRjH8dfFZGDyDxhEKBaLYlDkf2AxUbesNqVsJsVEyUIpo7LIj52yGOwGdhakU1+6nc73e8/tDup2nvou3/N93ud5Ps+np6bNqLWZr4MBA1jCFLpwhnN85zUr0mAS9wXingboV+O3PCD72y2OwvnEPC5D0hyuUoARrGI9V8Ea9lHHXgowgVe85ADDeCoDiPlqEReYwV2qghhgFwsYRabLX5Rx4jgeMYTnMmNsfNOHB6zgpqi8VAU9uA6qn8R6iwGy+4Og/E4uuRfvv3cxwAa6sZ1LnkY20sMUYBnHwXEfyFrJoh+DGMNbDDAb+o61vIXNVseYXFplfFABkgro5LXepPPWFkplpMpITQ31/xvpBzY+JylMcpDRAAAAAElFTkSuQmCC" width="8" height="20" /></span><span class = "S2">, Recon 2.0. Other COBRA models, including </span><span class = "S2">Recon 3, may also be run with this tutorial. For information on metabolites structures and reactions, and to download the latest COBRA model releases, visit the Virtual Metabolic Human database (VMH, </span><a href = 'http://vmh.life%29./'><span class = "S0">http://vmh.life).</span></a></p><p class = "S4"><span class = "S2">Before proceeding with the simulations, load the model into the workspace:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S19">global </span><span class = "S12">CBTDIR</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelFileName = </span><span class = "S18">'Recon2.0model.mat'</span><span class = "S12">;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelDirectory = getDistributedModelFolder(modelFileName); </span><span class = "S15">%Look up the folder for the distributed Models.</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelFileName= [modelDirectory filesep modelFileName]; </span><span class = "S15">% Get the full path. Necessary to be sure, that the right model is loaded</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">model = readCbModel(modelFileName);</span></p></div></div><p class = "S13"><span class = "S2">In this tutorial we assume, that the cellular objectives include energy production or optimisation of uptake rates and by-product secretion for various physiological functions of the human body.</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S2">PROCEDURE</span></h2><h2 class = "S3"><span class = "S20">1. Standard FBA</span></h2><p class = "S4"><span class = "S2">Standard FBA predicts an optimal solution for a cellular objective within a given set of constraints on a metabolic network (see Equation 1). Constraints on the network are set by assigning limits on the uptake, consumption or production of metabolites in reactions.</span></p><p class = "S4"><span class = "S21">Timing:</span></p><p class = "S4"><span class = "S2">The time to determine a FBA solution depends on the size of the genome-scale model and is commonly less than a second for a medium sized model.</span></p><p class = "S4"><span class = "S21">Calculating maximal ATP energy production under aerobic conditions:</span></p><p class = "S4"><span class = "S2">For each new simulation, the original model will be copied to a new variable. This preserves the constraints of the original model to perform further simulations with new constraints. Additionally, this method of renaming the model avoids confusion while performing multiple simulations at the same time.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelaerobic = model;</span></p></div></div><p class = "S13"><span class = "S2">The ATP demand reacti</span><span class = "S2">on, i.e., </span><a href = 'http://vmh.life/#human/all/DM_atp_c_'><span class = "S22">DM_atp_c_</span></a><span class = "S14"> </span><span class = "S2">within the model is a reaction that involves hydrolysis of ATP to ADP, Pi and proton in the cytosol. </span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> printRxnFormula(model, </span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div></div><p class = "S13"><span class = "S2">We will set this reaction</span><span class = "S2"> as our obj</span><span class = "S2">ec</span><span class = "S2">tive with the </span><span class = "S14">'changeObjective'</span><span class = "S2"> command. Maximising the flux through the ATP demand reaction will result in the network producing a maximal amount of ATP (up to the limit of the reaction).</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelaerobic = changeObjective (modelaerobic, </span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div></div><p class = "S13"><span class = "S2">The glucose and oxygen, in this case, are provid</span><span class = "S2">ed in high </span><span class = "S2">am</span><span class = "S2">ounts for</span><span class = "S2"> ca</span><span class = "S2">lc</span><span class = "S2">ulating the flux through ATP demand. </span></p><p class = "S4"><span class = "S2">The </span><span class = "S14">'changeRxnBounds' </span><span class = "S2">function changes the flux constraints of the lower ('</span><span class = "S14">l</span><span class = "S2">'), upper ('</span><span class = "S14">u</span><span class = "S2">'), or both the bounds ('</span><span class = "S14">b</span><span class = "S2">'), of the specified reaction. Here, we will change the maximal uptake of glucose to 20 </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAmCAYAAAAvDACoAAAB/0lEQVRIS+3VS6hNcRTH8c8VESOFGCkMvB8DFBMxlRQjZUaEkSITKQPC2CN5zESKkRHKRJ4DBkopJkoo8iiF0Lqto312++y9u6db1PlP9j5r///f/1q/9ThDRnENjSLbAF6p7kCWgSzt2+6fqJbx+I3vPfweW7D/7Ly38fwgDueBTbhWumAiHmE+3mEh3seeJnh4dAerE7gUT0vwOXiRtpcJ/9YGPh1v8uDr9O5LCb4ZV9N2AvvbyrIOt3LzJWzFrxL8JHZVydYkywEczYPbcL4EnoSHGVF8mo2QZnjVwdvovTzhwerSuwneRu+IKqKL1aV3E7xJ78l4jFm9yrROlia9d+BMIQddetd5Pga3sSYPR2M8K4CmZsOEKRrnK+ZlDpd08tDL8wj5Q6G+F+Bz/o5E3yxcHObo4ENYgYtYFCXbC7640Inh2Vx8RLR6SLESW1LzgG/HOVzBdVyuk2UnThdkuID7OJsyROgT8Cr3RCTxvrHgSGWdRzTRlRHqBuwtXHIcR/Ap9T2GfQX51hbmTCW8o/cyPMEMRIIDGIkrr5n4kTMoxvLfVaX5KtxFPO9VwFqbquCd+t6DGEojXmX4uPQ6ZsZunBoxuWJwTcGDbOmokCix8ohtfV/Z80jmc0zDetxoTarYWKV5tHZUx9t+wHVN1C93+HzTP1Ffl/y/8D8RNmUnFlG+GAAAAABJRU5ErkJggg==" width="11.5" height="19" /></span><span class = "S2">mol/min/gDW and of oxygen to 1000 </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAmCAYAAAAvDACoAAAB/0lEQVRIS+3VS6hNcRTH8c8VESOFGCkMvB8DFBMxlRQjZUaEkSITKQPC2CN5zESKkRHKRJ4DBkopJkoo8iiF0Lqto312++y9u6db1PlP9j5r///f/1q/9ThDRnENjSLbAF6p7kCWgSzt2+6fqJbx+I3vPfweW7D/7Ly38fwgDueBTbhWumAiHmE+3mEh3seeJnh4dAerE7gUT0vwOXiRtpcJ/9YGPh1v8uDr9O5LCb4ZV9N2AvvbyrIOt3LzJWzFrxL8JHZVydYkywEczYPbcL4EnoSHGVF8mo2QZnjVwdvovTzhwerSuwneRu+IKqKL1aV3E7xJ78l4jFm9yrROlia9d+BMIQddetd5Pga3sSYPR2M8K4CmZsOEKRrnK+ZlDpd08tDL8wj5Q6G+F+Bz/o5E3yxcHObo4ENYgYtYFCXbC7640Inh2Vx8RLR6SLESW1LzgG/HOVzBdVyuk2UnThdkuID7OJsyROgT8Cr3RCTxvrHgSGWdRzTRlRHqBuwtXHIcR/Ap9T2GfQX51hbmTCW8o/cyPMEMRIIDGIkrr5n4kTMoxvLfVaX5KtxFPO9VwFqbquCd+t6DGEojXmX4uPQ6ZsZunBoxuWJwTcGDbOmokCix8ohtfV/Z80jmc0zDetxoTarYWKV5tHZUx9t+wHVN1C93+HzTP1Ffl/y/8D8RNmUnFlG+GAAAAABJRU5ErkJggg==" width="11.5" height="19" /></span><span class = "S2">mol/min/gDW. The uptake of oxygen is effectively unconstrainted (i.e. infinity). </span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelaerobic = changeRxnBounds (modelaerobic, </span><span class = "S18">'EX_glc(e)'</span><span class = "S17">, -20, </span><span class = "S18">'l'</span><span class = "S12">);  </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelaerobic = changeRxnBounds (modelaerobic, </span><span class = "S18">'EX_o2(e)'</span><span class = "S17">, -1000, </span><span class = "S18">'l'</span><span class = "S12">); </span></p></div></div><p class = "S13"><span class = "S2">The function </span><span class = "S14">optimizeCbModel</span><span class = "S2"> calculates one of the op</span><span class = "S2">timal solu</span><span class = "S2">tions for a (maximum or minimum) objective reaction wit</span><span class = "S2">hin</span><span class = "S2"> t</span><span class = "S2">he defined solution space. In the above example, the maximal flux through the </span><a href = 'http://vmh.life/#human/all/DM_atp_c_'><span class = "S22">DM_atp_c_</span></a><span class = "S2"> is desired. </span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">FBAaerobic = optimizeCbModel (modelaerobic, </span><span class = "S18">'max'</span><span class = "S12">)</span></p></div></div><ul class = "S6"><li class = "S7"><span class = "S8">Anticipated results</span></li></ul><p class = "S4"><span class = "S2">When oxygen and all carbon sources (internal and external) are provided the flux through ATP demand reaction can reach its maximum rate of 1000 </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAmCAYAAAAvDACoAAAB/0lEQVRIS+3VS6hNcRTH8c8VESOFGCkMvB8DFBMxlRQjZUaEkSITKQPC2CN5zESKkRHKRJ4DBkopJkoo8iiF0Lqto312++y9u6db1PlP9j5r///f/1q/9ThDRnENjSLbAF6p7kCWgSzt2+6fqJbx+I3vPfweW7D/7Ly38fwgDueBTbhWumAiHmE+3mEh3seeJnh4dAerE7gUT0vwOXiRtpcJ/9YGPh1v8uDr9O5LCb4ZV9N2AvvbyrIOt3LzJWzFrxL8JHZVydYkywEczYPbcL4EnoSHGVF8mo2QZnjVwdvovTzhwerSuwneRu+IKqKL1aV3E7xJ78l4jFm9yrROlia9d+BMIQddetd5Pga3sSYPR2M8K4CmZsOEKRrnK+ZlDpd08tDL8wj5Q6G+F+Bz/o5E3yxcHObo4ENYgYtYFCXbC7640Inh2Vx8RLR6SLESW1LzgG/HOVzBdVyuk2UnThdkuID7OJsyROgT8Cr3RCTxvrHgSGWdRzTRlRHqBuwtXHIcR/Ap9T2GfQX51hbmTCW8o/cyPMEMRIIDGIkrr5n4kTMoxvLfVaX5KtxFPO9VwFqbquCd+t6DGEojXmX4uPQ6ZsZunBoxuWJwTcGDbOmokCix8ohtfV/Z80jmc0zDetxoTarYWKV5tHZUx9t+wHVN1C93+HzTP1Ffl/y/8D8RNmUnFlG+GAAAAABJRU5ErkJggg==" width="11.5" height="19" /></span><span class = "S2">mol/min/gDW. </span></p><ul class = "S6"><li class = "S7"><span class = "S8">Troubleshooting</span></li></ul><p class = "S4"><span class = "S2">If there are multiple carbon sources available in the model, it may be necessary to specify more constraints in order to examine the effect of a single carbon source on ATP production.  </span></p><p class = "S4"><span class = "S2">To avoid this issue, all external carbon sources need to be closed with the exception of the single carbon source of interest.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%Closing the uptake of all energy and oxygen sources</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">[exchBool,uptBool] = findExcRxns(model);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">uptakes = model.rxns(uptBool);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% If you use Recon3.0 model, then:</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelalter = model;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelalter = changeRxnBounds(modelalter, uptakes, 0, 'b');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelalter = changeRxnBounds(modelalter, 'EX_HC00250[e]', -1000, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% The alternative way to do that, in case you were using another large model, </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% that does not contain defined Subsystem is</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% to find uptake exchange reactions with following codes:</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% [selExc, selUpt] = findExcRxns(model);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% uptakes1 = model.rxns(selUpt);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% Selecting from the exchange uptake reactions those </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% which contain at least 1 carbon in the metabolites included in the reaction:</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"> subuptakeModel = extractSubNetwork(model, uptakes);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"> hiCarbonRxns = findCarbonRxns(subuptakeModel,1);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% Closing the uptake of all the carbon sources</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"> modelalter = model;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> modelalter = changeRxnBounds(modelalter, hiCarbonRxns, 0, </span><span class = "S18">'b'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% Closing other oxygen and energy sources. Use the following lines for recon2, or uncomment the lines below for recon3</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> exoxygen = {</span><span class = "S18">'EX_adp'</span><span class = "S17">,   </span><span class = "S18">'EX_amp(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_atp(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_co2(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_coa(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_fad(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_fe2(e)'</span><span class = "S17">,</span><span class = "S23">...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">     </span><span class = "S18">'EX_fe3(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_gdp(e)'</span><span class = "S17">,   </span><span class = "S18">'EX_gmp(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_gtp(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_h(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_h2o(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_h2o2(e)'</span><span class = "S17">,</span><span class = "S23">...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">     </span><span class = "S18">'EX_nad(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_nadp(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_no(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_no2(e)'</span><span class = "S17">,    </span><span class = "S18">'EX_o2s(e)'</span><span class = "S12">};</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> </span><span class = "S15">%exoxygen = {'EX_adp',   'EX_amp[e]',    'EX_atp[e]',    'EX_co2[e]',    'EX_coa[e]',    'EX_fad[e]',    'EX_fe2[e]',...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> </span><span class = "S15">%    'EX_fe3[e]',    'EX_gdp[e]',   'EX_gmp[e]',    'EX_gtp[e]',    'EX_h[e]',    'EX_h2o[e]',    'EX_h2o2[e]',...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17"> </span><span class = "S15">%    'EX_nad[e]',    'EX_nadp[e]',    'EX_no[e]',    'EX_no2[e]',    'EX_o2s[e]'};</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelalter = changeRxnBounds (modelalter, exoxygen, 0, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div></div><p class = "S13"><span class = "S21">Calculating maximum ATP energy production under anaerobic and glucose only conditions:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelanaerobic = modelalter;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelanaerobic = changeRxnBounds(modelanaerobic, </span><span class = "S18">'EX_glc(e)'</span><span class = "S17">,-20,</span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelanaerobic = changeRxnBounds (modelanaerobic, </span><span class = "S18">'EX_o2(e)'</span><span class = "S17">, 0, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelanaerobic = changeObjective(modelanaerobic,</span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">FBAanaerob = optimizeCbModel(modelanaerobic,</span><span class = "S18">'max'</span><span class = "S12">)</span></p></div></div><ul class = "S6"><li class = "S7"><span class = "S8">Anticipated results</span></li></ul><p class = "S4"><span class = "S2">Compared to the aerobic condition, anaerobic condition with only glucose as an energy source has reduced flux through ATP demand (82 </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAmCAYAAAAvDACoAAAB/0lEQVRIS+3VS6hNcRTH8c8VESOFGCkMvB8DFBMxlRQjZUaEkSITKQPC2CN5zESKkRHKRJ4DBkopJkoo8iiF0Lqto312++y9u6db1PlP9j5r///f/1q/9ThDRnENjSLbAF6p7kCWgSzt2+6fqJbx+I3vPfweW7D/7Ly38fwgDueBTbhWumAiHmE+3mEh3seeJnh4dAerE7gUT0vwOXiRtpcJ/9YGPh1v8uDr9O5LCb4ZV9N2AvvbyrIOt3LzJWzFrxL8JHZVydYkywEczYPbcL4EnoSHGVF8mo2QZnjVwdvovTzhwerSuwneRu+IKqKL1aV3E7xJ78l4jFm9yrROlia9d+BMIQddetd5Pga3sSYPR2M8K4CmZsOEKRrnK+ZlDpd08tDL8wj5Q6G+F+Bz/o5E3yxcHObo4ENYgYtYFCXbC7640Inh2Vx8RLR6SLESW1LzgG/HOVzBdVyuk2UnThdkuID7OJsyROgT8Cr3RCTxvrHgSGWdRzTRlRHqBuwtXHIcR/Ap9T2GfQX51hbmTCW8o/cyPMEMRIIDGIkrr5n4kTMoxvLfVaX5KtxFPO9VwFqbquCd+t6DGEojXmX4uPQ6ZsZunBoxuWJwTcGDbOmokCix8ohtfV/Z80jmc0zDetxoTarYWKV5tHZUx9t+wHVN1C93+HzTP1Ffl/y/8D8RNmUnFlG+GAAAAABJRU5ErkJggg==" width="11.5" height="19" /></span><span class = "S2">mol/min/gDW), signifying the need to oxygen to run the oxidative phosphorylation. The results are dependant on the model you are using. For Recon 3.0, under anaerobic conditions with only glucose as an energy source, the flux for ATP demand is 40 </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAmCAYAAAAvDACoAAAB/0lEQVRIS+3VS6hNcRTH8c8VESOFGCkMvB8DFBMxlRQjZUaEkSITKQPC2CN5zESKkRHKRJ4DBkopJkoo8iiF0Lqto312++y9u6db1PlP9j5r///f/1q/9ThDRnENjSLbAF6p7kCWgSzt2+6fqJbx+I3vPfweW7D/7Ly38fwgDueBTbhWumAiHmE+3mEh3seeJnh4dAerE7gUT0vwOXiRtpcJ/9YGPh1v8uDr9O5LCb4ZV9N2AvvbyrIOt3LzJWzFrxL8JHZVydYkywEczYPbcL4EnoSHGVF8mo2QZnjVwdvovTzhwerSuwneRu+IKqKL1aV3E7xJ78l4jFm9yrROlia9d+BMIQddetd5Pga3sSYPR2M8K4CmZsOEKRrnK+ZlDpd08tDL8wj5Q6G+F+Bz/o5E3yxcHObo4ENYgYtYFCXbC7640Inh2Vx8RLR6SLESW1LzgG/HOVzBdVyuk2UnThdkuID7OJsyROgT8Cr3RCTxvrHgSGWdRzTRlRHqBuwtXHIcR/Ap9T2GfQX51hbmTCW8o/cyPMEMRIIDGIkrr5n4kTMoxvLfVaX5KtxFPO9VwFqbquCd+t6DGEojXmX4uPQ6ZsZunBoxuWJwTcGDbOmokCix8ohtfV/Z80jmc0zDetxoTarYWKV5tHZUx9t+wHVN1C93+HzTP1Ffl/y/8D8RNmUnFlG+GAAAAABJRU5ErkJggg==" width="11.5" height="19" /></span><span class = "S2">mol/min/gDW.</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S20">2. Sparse FBA</span></h2><p class = "S4"><span class = "S2">Sparse FBA calculates the optimal solution of an objective function and finds the smallest set of reactions that can carry flux to achieve the objective. Sparse FBA minimises the number of reactions by keeping same maximal objective;</span></p><p class = "S5"><span style="vertical-align:-40"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMIAAAC+CAYAAABnAdCLAAAanElEQVR4Xu2dC9R+RVXGH29psoTS0jTFG6lpmvcss1QUUghBTSgU0QQTRbO0FFKUS2kqKiZUoIiKAqYoqGQpeCEL0EwDFNFUEESJYmneMrT1gz24Of9z3nfmvN85Z8777VmLxf/9zpy5PDPPzJ49++x9PUUKBAIBXS8wCAQCAQURYhIEAiAQO0LMg0AgiBBzIBC4BoHYEWImBAITEeGnJZ0o6ZGSnizpzTESgcDUCEyxIzxA0tnW8bdJepKkH04NRNS/uRGYggj3lPQZg/1USbsGETb3JKyh91MQgX4/XNKdJL1f0qU1ABFt2NwITEWEzY169L46BIII1Q1JNGgKBJYR4daSLpf0f9a4m0vaWtIPJF3S0uCflbSVpO9K+vqSDt1Q0raS/qMj300l3U7SBfb8ZpJuL+n6ki6SdOUUgEWd64lAGxGYgL8maT9JDzZZHkIcaSrPhMRJkn5f0v9Iupek10p6qIPpHEm7SLqsAR2T+9GSDrZn95V0leWBRPeT9DuSniXpw5K2l/RUSUc3ytlZ0vvWc1iiV2Mj0CTCYyS9u9GIl0o6SNK3JH1c0o7u+SskMeEhBc/fYZM2ZTnNyMCOAgFYyX06QdKepjW6iaSzjFTNPHuYpunnJN3SHlLfPSRdPDZoUd/6IdAkAmIP4s9fSdrJdXd3Se8yEekXJJ3pJiTZ9pF0nIlMd5T0QdtJviHpbpL+W9INJG0jCRGHewQmdJv69O6SznN1f1XSoySdKwlx6ghJz7Dn+7bsFOs3StGjwRHoOiM83lZ3GtAmguwt6Vhr3eOMJL6x/v17S/q0e+hX/jYi+OecHx4o6Qr3/naSLrTfzzSRbXCgooL1RiCHCM2JDCKLJnrzOTfJn+hJhLab5yDCes/JSXqXQ4TmRF420Zc9L9kR2naMIMIkU2W9Kw0irPf4Ru8yEQgiZAIV2dYbgSDCeo9v9C4TgSBCJlCRbb0RmJoI/kItIe0P023P/WGZ+4tj1nuIondjIJBDhGXq0zatkr+hbj6/kalTMcvgnuCXzDYp9XfZcy7skn3SG+wy70djgBV1rC8CbUTgxve9kpjAJGx+jnIfz9zGzDDSc8wrWJm/afn5FPNFkp5rvzHDOMRMMG4saQdJpzhId7P6MMPg9hk7JP8cW6PTLT92UJR7qP2GEHzb8JX1HaLo2RgINIlwgKTDOirGVCJ9TNOWBTOI70k6o+P9Z5t5RNtjzCjYIbBr2r8lAxdrTzG7JvI10+9KQoyKFAj0QmCZGXavQuOlQGBuCAQR5jZi0d5BEAgiDAJrFDo3BIIIcxuxaO8gCAQRBoE1Cp0bAkGEuY1YtHcQBIIIg8Aahc4NgSDC3EYs2jsIAkGEQWCNQueGQBBhbiMW7R0EgSDCILBGoXNDIIgwtxGL9g6CQBBhEFij0LkhEESY24hFewdBYAwi8H0C/lFxHJwSf8P7XaRAoAoEhiQCjoRPNteO3hse3y0QICS+IahiCkQjQGBIIlA+jn/xcvdySYcb5Djy/SdJr3d/i9EIBCZFYGgiUD4esn+iESvtbyUdL+kjk/Y+Kg8EDIGhiUA1fykJ0Sh9pE8wkXfat8YpAEkMSCAwKQJjEIGgH7iZ55tnotzwIT4BRroi5UwKSFS+OREYgwhPkPQ6ScRV2MtcvL99c8Idva4VgTGI8JsWAup8Sf/g3LzUikm0axMiMAYR7izpC5I+b+cEf5+wCSGPLteIwBhE+EVzCEaQQC7WIgUC1SEwBhFQnf5vdT2PBgUCDoExiBCABwLVIxBEqH6IooFjIBBEGAPlqKN6BIII1Q9RNHAMBIIIY6AcdVSPQBCh+iGKBo6BQBBhDJSjjuoRCCJUP0TRwDEQCCKMgXLUUT0CQYTqhygaOAYCQYQxUI46qkcgiFD9EEUDx0AgiDAGylFH9QgEEaofomjgGAgEEcZAOeqoHoEgQvVDFA0cA4EgwhgoRx3VIxBEqH6IZtHAG5vXxO/NorUtjQwirD5yTIJ7Srq5fZJ6haQv2ffZN5T0CElnSPr+6lVVUwJOnHeSdCtJ95b0RElvkLSPpB9V08qChgQRCsBqyfpgSe8yR8fNx2+UhFe/h0q6g6T/Wq2qqt6+raQXSnq86/usnToHEfrPr0eanyZKwL8rE+MS2xnw7vcaK/pja+ze8jHmoYSu4snwgv5wTvvmWESgntdKOmhN4iJsLemfJd1d0vvMwXHTj+tzjAwHW7+nHelhan+ZpD81953Jt+0wNQ1c6lhEQIZ8la0a6xAgBJHoTBubP+5wb/8zki43B8iIT+uWbiLpLEn3knSEJIg/2zQGEe4j6V8lfWONiPB75taegX+mpCNbZgAT5ZOSdpd07mxnSHfDEYU+a493kXTqnPu4iAg3sIn7U5K+Jeki+z9aku9kdhptymcsL0RAw5AOjXPWojDw77F+fVUSZP/PBiZpxeSwvA67YHPI02LA3EAsYn7MNnURATeN75V0J5vIbH8pPc/EnGWdJnQUkXHa0tx3h+0kXeg6dpokyNE8J7CYXCXpJyXdxhaW35L0bTtce1UjotTzrQyiC6GGrTUxb46V9GTzaXsPSfT1tyVtL4nFE6fPREWahbasjQio/NjKcdP4EElfloQ+nEPRoQtEgeagsRvgCv4QO1SycuxnMRIojzhqc3UFCW6Ew2LipoS6dF+b+E0s0Cj9ufsjuwgHbTBJaU9Jb7UfXeeOWojBPQIhwVgoaSvt5q6EPvnEgvfLki6rpeFd7WgjAqq/k6yDKe4Z75OXlQ9ZEKbnJg5S+0ti8O9aIFbllj9Vvpua5sjvligE/kTSDzsadYCkw+wZqyirZkrcNXARR6r9cuoBks62tj7LFrjzJL3I/kY8DNTLpFcYJlONU1a9bUTgkgS9OGINnfmuK4lt/UGSXpJV+jWZEhHmLg61dZmLJQ7Et3QPD2ys/v49L1IhQhA9yCdEi1Mk7S3puAKMHyvp5wtvdTnD/J3t+AVVXZ31DyW92r1EeDB2vbQAeFKzcO7WsVOW1jtY/jYieI3Iv9j1+RetBWyJt26sZMsat85EoO+cp/zKzt8wNTimBZitbCVFhGjTNmGO8Y89LqfYQZ66bCBanj9Q0jmF711f0oeM/PSDuBfsbv585FWrEGHXBbtkYfXDZG8jQvMgmLY3bkov7dGMdScCkPh7hQTRXRoHav7OJGLFx04HkYFzVzowMxYftB0G8aokcbHF+csf4Je9f1/T4n16WcbGc8IDf839DY0RYpFP6ZzJTvk2SU+aIxHoEAOF1qiZniaJQ2GJYdVmIAI4eZMLfncdeDlfMWmbKyWr5t/M4K4l7Vr0sesiLd0dkYczRMmZspCXG5N90T0Ch8CTTTPga0MefEEBGdaFCGi6uD1lsnZF/sGEJJ2fumTjJF9zBuOOAZEi7cJt54ackd7Dyii5m7mZpDdbWK+cOlKeZFbB7zb7Iq9aJc9vSMLequrURgR03qz42JbznIMOcZF9IkDgRzN7ti5EQO5F9GCB6Log87etJ0hCJdrUICVlRFIeoIzAVIELuhdnYtrM1veMgPYHNWhu8rJ/V/88BrMQi+h8GxGebreEqEpTQlV4tCQO0iTUocROzknrQoQ7SkKeRgXsZWSPgdevdx0SkzKCONP3l/RKKzPtDjmYNvOwCz268PKKAy4icLr5z6nXnx/bzCo4A2FXhVUqdyTcISSVcE75k+VpIwJ25ahIm0ZUXuPRJvehTWIb/IoktE0peSJwwfbNjt6y0qKFQIuRtFSTAdNScTIX6dII8YrXr3fZIKGpYQcgoSFisnDgxECv9kTM7BPNbqytzc9wdlc7m2Wu7xMHdCQMEqrb0oP6YPgsukdoXvjAdiY4g819wgdcq5KBWbpZxLwCM2VSOhzy7/TxBlqWP7ArekQHv9LUet+QLhpp368v0QixGnbtHE2tHEQoWZUHmwxLCvayf9sYMcGTlS33IU1lC7vl5xp3Ls05NlXfWkWjJMMymGhCWL04KDIRkPnY0pu25/4Chc5gaoAoRUIbwqE7JSxRWRn8wZBtHbv+lBAZuKiqKflLJLB5nJkVcNjFjogLJlZMT/i29nsi7Og+7qmpr21t8YudF3u2kfRsSXx3AUEeZdbGzTJu12KYV41GqetCDdUfdkCISD4hizLgTfEGonzY9Onk99oEnqFyRZdMAkRuQtGZp+R1021Em3qSgBOHQwzIsLL0dkPNti2b3OkwiaYHMWNOiZv0l7Zc3jGm3H2gheqyTAZD3k1mGPS7S3wcHZM2IiSLSRrDB+lokZjMWEMuChiOeTbnBIBgZfCJMtk10ER9vcVKk7xsnawumDPXGJic3RE1IH0gdjRiIDsjlpaslnymyGLgDenaBhScsDTls865JsaKyzLmD/3tGtOu/qWFD8XB22sAYYwPc2roZ7ShLgQwB8FSYZEGbtQWBxFGhTsqsy/2EDNLDQsHBS+IMCi8UbhDAJGQcwSH6kUq6ElACyJMAvumrJQLSdTvqN4/VRsCQYTaRiTaMwkCQYRJYI9Ka0MgiFDbiER7JkEgiDAJ7FFpbQgEEWobkWjPJAgEESaBPSqtDYEgQm0jEu2ZBIEgwiSwR6W1IRBEqG1Eoj2TIBBEmAT2qLQ2BIIItY1ItGcSBIIIk8AeldaGQBChthGJ9kyCQBBhEtij0toQCCLUNiLRnkkQCCJMAntUWhsCQYR+I4JrEj5Ax6sDDm/xzFDqPrFfzfHWIAgEEfrB2gwFhdcOnFU1Awr2Kz3eGh2BIEJ/yG9kDnRxVTmLqDD9u7r+bwYR+o+xD4ZRjaOq/t3Z3G8GEfqPPy4rk7c+4kdX49C2f5c275tBhP5jn4KCtIWK7V9qvDkJAkGEfrBzPiDiDZoi/Lrip6crpGy/GrZ8CzeLuJv0UU43quwxy0GkRKlQEn5s8PYFEfpBjI8enBWTiD6P89shEv5Ff9XqwO26d7efUx++ZFHxIrrtLulNznV7eh/1Lw6aCehx5EBhYAlMTiAUXD0SIwNHyF1Rh7r6tbX51iWKKd7UCX2Fs2q/ALFYEM2Jec1C1fTB24lZECFnOm2Zh4gw77Y/b7SPf+4oOH8QuJE4Ej6VEIFdi3BUuGlPqS2QuQ87VVL+IuRw+sxkx5kX3u18HOo+8S+6Ir02o4+mGBa0rejcFkToR4QUBWgjXNgzae5sk/95LcEbaSGBRIg3QXwKXNOXJDx1c6iHVLSXwCTe27iPTbHK7obXdBYFIuVwfmpLBJeBeAQUuaqkE5aXcLwEMyQ1g9Wk+M/sPMXjEkQoHw3iyRHeCrfwr7MQW6XyLn5ACUW1g73vV8zUotMlvcWCe19c3szrvOGj/bBSNmPAEfOCQChtMZMXVY24wjmJYDAE/WhLnKHYPYmgtOqFI1GWjjJX9JDO4+KDGDIuBC/JTkGEbKiuzegBJ2pOCpdUUlJXFEwmDeIMUYqIObBRCbn6fCuszRSEFXyvlkhIi+pvhgtLeYmXwFmDCEj/lhEvIrePzFUiLyGW+tC86X0f2rd4XIIIucPw43xPtJWavyDSpENzSUltRPgzi+G86qrZ1g7sorjnYOdpThIOmIhbbVEyF/XJ74w+H9GQ/n4A7ZaPwXZgI2oRgVfOs/71iuYZRCiZvtdoI441Lc7nbQX9QVkRV+eGTK9yh0jOAJhqkIgxR+jej/QkWVtz/OrdvAUngDy7HOJNiQqYKEqswhA4pRQfj9/vkPTWDdzdfDTSZpxvjB5xN0/qNS5BhLJZjEyMS3PUgS+X9IKy17fIfXs7JKNWbGqIyIyG5S8s4CADTODCPomD5CkWV/kAK5NymPyofleJXEMYLVSiRNJ8bkNDlNrKeYd6IPeX+3TAYnujpAATH9q2GZGVMwSx/opSEKEIruvEUW5qLcpK2jI38ecIW4tOn0nVllYJRZs0XcjuiEHchXyhRfuySj9yNGCIkhzYl8Wa8+3w5wPaD4FZFNiViFWNpiglH601uy9BhGyors64vyQmFIOYtBbI36xSJWLFslqRh7k0Qp4niHdKfb95YJyPsQstzifPl3SmqWMPW9aYFZ4TdpedzpO7zz2CPx9wiXa4tQlNFwvS2fZNCGW3acWWdiGIsBSiazMgXqAh8loLDmmILIg4JTelHDS58OpKrHZJz05kUwhASFuiUPY17oNQaHMg8eWSPutW1nwUrpsTUXHRHEr9ADt2PFZuxBYWkWaI4kVt8OcDbtqJvPMIGw+wQUxlXNJux30I9aUwwOywvHelpOPaNHJBhPwp4LUk6MxZWVFzcsFTGiK1S326rDWrBGJPgeSpA/HkfjYxltXZ9Rw8/r3jAnBRmRgpohgoWThYAI535wPucAjlywT/uIU+pk52i4/aPQ/nOMxGIAo7Rkptl4oL2dwXoHV9zxMBjQgT6Z2SuO0svVAj0PgTegC1iglEIgLiw6+scGhNzd7KbryZcCWpj2jkTSdSXdzCo3nz9zrpmT+/+Zvz9HwLETN2hPwhRJQ5w2l3iCJ/SE9TAVZExIoSAnEwRM1aspL63u1pt7LI7Kzkqybag8jSZw6xQpeonRGDOBSnxCLCYkTy5wd+72hatpSX5yw8BIwPIqw66vY+9jTb2lY8xMXXBjWztRgmBKkvkYZsW07ZGCNimnJJy2UdfeOyELHr2wsK4/4BkWoLQ8k+bM5pdOQJBGpDgLnOQRmxECJc504miFDbcEV7hkCAef5KSX/U5XYniDAE7FFmTQjwRRw7Ad9lJNXrFu0LItQ0ZNGWIRBAa7S3mZlf2lVBEGEI6KPM2SEQRJjdkEWDh0AgiDAEqlHm7BAIIsxuyKLBQyAQRBgC1ShzdggEEWY3ZNHgIRAIIgyBapQ5OwSCCLMbsmjwEAgEEYZANcqcHQJBhNkNWTR4CASCCEOgGmXODoEgwuyGLBo8BAJBhCFQjTJnh0AQYXZDFg0eAoEgwhCoRpmzQyCIMPyQEdkFF/D4KeKDd1L6TJDfjAEf8fMf/oYuHL5JUUMTgSDC8HMCFyy4MSfhtx8HW/gfJRHKCce5+P3HVw/+kl4/fJOihiDC+HMAT82s9rhW5P/J/QgtwXkuHt9YkIhqQ3AL3JtHGhmB2BGGBRxfSPjfwblWEoeSwynCQOETFJ+pyTcpvlW/M2yTZls6YuRdTHQs8YmU1eEgQhZMK2XCQ7SPF0Y8Apzw8h0tH5WnhMOvEn+gKzVqhi8nj3arePvr7HYQYdwZgdtIYonh6Y6BvWDc6mdXG7sAHq/xZ3qRxX4DN2IysMsSO4LY0yunIMLKEBYVkFa14qiPRbWsT2bvXxVP2vibxekvi0kfH6qxI1QyNxCHCD21EdF2KunS4M3ALT6x6tgZ8Ip9moWJQs18xUbVHjvCRiG5vByw5uBM8A+c2n5o+SuRw1y/P8jcz+8h6SRJX5T0SYuyuSEBWoII4801rzbtFdVlvKZWUxOiEcHRce77EHPBf6gRANfut+gRgL21c0GE8cY8XawR1YXoLn0iz4/X2vFqQqtG1E/S91sCJqJYIAAh3q4Rh+5j0X64gCSOc4lr/TgjjDeunTURgRMtRzO8awVNm6QJBEF/iqmSfQOIJoTD3s81WkXwwxTthjgRG5piR1gdTlyME8IIeyLuAQjeTShVrxpFLCJ4H6tYTkBAHNfewfI+R9LOLTZIBMtAdiaABgfIuSRUyNy2szCg+SHm9MWmESIuc0qrRBAtxiKIUAzZtS8QTZPA4Bx+OQTzb2Ta91uONOGfJuloVw2rGeIRge6Qf5uJgBifaMQrJjge5ErJxy/bUDVifziy3uTSEFMSsIG8kNljkGKlUdjHJD18hdjSWQ1KmYIIRXBdm5nQr2gtSHtJeov9+7a2uvFzVQM6TwiM9dgZvDxM6KoXW70E4K7dRom5huoYUneR1wd0p2vJpKLfKBW8FUQoAMuycnhLIV6JNP8aVwRGdISbJaVgd+U1/PgNLFH361gdCdn6KUno2RG5cgN4s5MRT42DaUlCZ3/CCodT4hOk3XLfxi6Z2sGhmVtkdgxSjhhZ0ofOvEGEMhhZsbjVZOJ1bd07mFoP8WdV2yEfVpVbaR//DIM+zLuJOYzVam5qi0KZ8y79fVhPbRemEsQuI5AhhAU/VKLN5PMFEXJGZaI83G6+2uomMB2r15CJicMhm8mDeYH/aCfFD07xhHPbsZ2Vw3cQJQlV5T5mLVvyHnk9+U6VtFsHodgREDnTDtfsc2m92fljR8iGSogUX7PsY9kKpUnbXB2T8R4HdH8Qz+/NuDmTaQm1LlIf+zMCO0aJyLdSj4II+fD5VY0I74fnv9o7J2rUc02DtL2k062kIyye8BbRITNqgtCcO3LPFBSJGMZnpMf0OCP4Q/Iyccd/zQfBn96jvgwItswSRMiHLUWuX7aq5Ze4PCeiwllmtp20ULuaKhUSoH8vTX3PCH1vxJuaoEUH4IMkvcQ65Ilf2sfi/EGEfMg8EZof1eSXUpYTEegcExFQO/Lv8yWtMkm4qOPgW6pu5YIQLVmpkRsXjf4muIsI3MGcbX1F84ZKeMO/ROuCP4iQPzE9EbruCBA7MBt42QZt6d5ilcmE6pbPOTkbzCWBxxtdY7uUDB5fjBKTinqUfgYR8mFOti68wWGZwfJyNgfbD0g6ymxl8ktenJPzAJOfhB0OOvjSVXmj2lJaTvN8wPttIhZ2R+x0JH9BWVpf7/xBhHzowIqVmIMmCTEFDQjbN9akyLYHmmHdhlhEWj2JCNwZ8B3DhnyamN/tlXI2zwepMNzZ/LUReidJx9uDx5qJ9UqV9nk5iFCGGhc+L7QvpPybmAwwiMl/UVmpi3OfKOn+do9w5UYWPEJZ/nxwsNlkvcc+svHVoyHiOwO+S54kBRH6wY416a3sVSbnZf2KyXqLcwfmDaMdHLNalZfJnw92kcRlGmlbSewWuLgBu8kJHkTIG9DIVY6APx9wlsKs+kvlxYzzRhBhHJw3Yy3+fDC6OrQU8CBCKWKRPxcBfz6o3mtHECF3WCNfKQJd54PSckbJH0QYBeZNVwnz6mRTK9P5UgvZ0QELIowO+aaokHnFxdg2Zm6NsV7ph0CjAhVEGBXuqKxWBIIItY5MtGtUBIIIo8IdldWKQBCh1pGJdo2KwP8DzusUChE3g1QAAAAASUVORK5CYII=" width="97" height="95" /></span></p><p class = "S4"><span class = "S2">Equation 2: Formula of Sparse FBA.</span></p><p class = "S4"><span class = "S2">where the last constraint is optional and represents the requirement to satisfy an optimal objective value </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAqCAYAAADS4VmSAAACpUlEQVRYR+3XS6hOYRQG4OdEEkVRhgol5JqUSMRIQiRJcpkwMDERKZGUlLGU2wATBpIwkUQokcsAkUsoUi7FgFxb9f362v59/ss525mcb3LO2f/3r/fd71rrXet06OHT0cP4egn0KtCrQNUK9MVoPMH3ei1fNYExeIgZuPG/CMRbb8IVvMQbBJF+WIY9+FojU4UCA3EfIzEHU3E9KfAukflYJYGIPQSjkhIrcQG7Ui28z1NRhQIRfyGmJxVW4CSe4jZO41eVCkQKvuA1ZmELdifgaRiKD10l0Af9U5Bv+FGo8Il4gWFJ9impG8bhLn63S2As1mFzAfAw9uFR4fkIPMOkVJj/dGKzNTAA27EVUcmr8CpV+PEsailQ2d7RDIFBuIjIX1Tz8pTjWsyo8hPpj6uYWyclpXtPIwLx+VGsSW8ehvK3h1PUIHgnVXw8qllvU8tWIwLzcT5FWo+DdaJGMYbrhUJx4uetptDpdCULS72MmfiMqOBoreLJ73UrgdogiaBnsQQ/SxQIgwmCQTSsN6ZfU6ezFKxN+Y9AG7G/JGJeA6FQjUiXCOTF10jWGLXXElrUyIbcaBqxKFOgWNmdFdYO7ExA83CpEWj+eRmBCQXnKiMQvn8zyf4Y48s2nzJSZQTCbo9kX5qdWq0YJxaMU+nhZNxr5e3jbj0CxfzHvXNYXOiCmAsPEuBqHGsVvIxAMf+1uNtwIM3yBZn9Lk2jth38ugrk+Y8tJiQ+k1ltDSgqPuZ87H1tn3opyPO/KJlQAAxHqBOz/y0+tY2afbFIIM9/uFqM1+fdAdRsF+T5b6utWiVbVCDP/960gLQas6X7RQJl+W8paCuXcwLxe6zM0e9x4h+LSvNf9IEgEIYyOBnOIcTGW+lptBFVCl7mhJWD5gC9CvwBQ1mBKyLXZUsAAAAASUVORK5CYII=" width="16" height="21" /></span><span class = "S2"> derived from any solution to a FBA problem. This approach is used to check for minimal sets of reactions that either should be active or should not be active in a flux balance model that is representative of a biochemical network.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% [vSparse, sparseRxnBool, essentialRxnBool]  = sparseFBA(model, osenseStr,...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  checkMinimalSet, checkEssentialSet, zeroNormApprox)</span></p></div></div><p class = "S13"><span class = "S2">As an optional input, there are different appoximation types of zero-norm (only available when </span><span class = "S14">minNorm = 'zero'</span><span class = "S2">). Default is </span><span class = "S14">cappedL1</span><span class = "S2">.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% Other types of zero-norm:</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'cappedL1' : Capped-L1 norm</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'exp'      : Exponential function</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'log'      : Logarithmic function</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'SCAD'     : SCAD function</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'lp-'      : :math:`L_p` norm with :math:`p &lt; 0`</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'lp+'      : :math:`L_p` norm with :math:`0 &lt; p &lt; 1`</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'l1'       : L1 norm</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%  * 'all'      : try all approximations and return the best result</span></p></div></div><p class = "S13"><span class = "S21">Timing:</span></p><p class = "S4"><span class = "S2">The time to determine a </span><span class = "S14">sparseFBA()</span><span class = "S2"> solution depends on the size of the genome-scale model and is taking from </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAC8AAAAmCAYAAABUKMJkAAABwklEQVRYR+3XT6gPURTA8c8r5c/OgrKRZKWelSd/Q7F6K2+hFHmrt2FjQVHKRsmG8icUHit7JQtlKflTFlb0bGRBsrAVujVTY5r3vO69w2/qznJmzp3v+Z7TnHvHDPgaGzC7Av+/qlfMF/N/GtiL5Xi0kJhRa5utuIlNOI5rQ4DfjHvY2IA9huujDr8BT3EY47hSAQ8Cvil3HT4MFT5U4V2BX8T/ewV24yqm8GYRMQu98k/Mr8UhXGiQTODlqMIvwTacwIEW5GfsbPRrbA7Zza+qWuJGB9FthPuv8CuWuBGXDT4MjjDljnZYPo2H+JIBuLlEMvwk7mB1C2wWt/AcPzND18slwYe9zt2W7TnswqeegLOaP4PzLdDX1Z/lCb71mESS+ZprDQ7icgfoRdzH2x6SyAJfcy3FDpzFnnmq8RjfMyWSFb7JtB5HcK4D9FLVal8Tk+gNvuYKW4J9OFVVpb6/Hc8ywvd+GAmHh2mcxBa8iIQP0zsMw1DVmWqNj9iP9/jRtW6uY+DKqu87P/KXhJZVsyMc/ea7wj7qQfthLvhI4WlhBT7NX3x0MR/vLi2ymE/zFx9dzMe7S4ss5tP8xUf/BjcJViegBBPmAAAAAElFTkSuQmCC" width="23.5" height="19" /></span><span class = "S2"> second for a 1,000 reaction model, to </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAC8AAAAmCAYAAABUKMJkAAACk0lEQVRYR+3XXchNeRTH8c9T3lKUC6KYIjc0obyUl2kkF0SKKSWvczlFUl6ilGgKJZKkqfF6o0hJTUbTuJC8m0xxQbkhiuSCciO06r+1Hfuc5+xjPz3Pk/O/ObXPf6/1Xb//+q+1dodevDp6Mbs2fHedXlv57135yIBZmIsR+Ij/cB7P64nTE9JmbIIcXwdyGU4X/dfd8CPxJAf2PyYUgC7Chdrn3Qkfvo9hNbbhEN6gPxbibA72MSbibT6A7oQfjYBahVMFas/A1dzzqbjdU+CXJ8VD0fcF8CHsGfyS/utR8LtxCf82KLUrcqdSOfxA/JzydQnuVVzz8/CTau23mvM/IEpYqJetr5SpIJCDWJfuxo9412rO98F0bMDiGrAXqck8qgA4MzEY1xD1fw1OtFIqhyJS4kgB2J/p+Z3UFStk9yuO4iEiZb5QPRw1SpspWJvqcB4qVN6amsbLKmlztobn8nsynjbbYRekiIfVvHAcf+AGPnQRdJjti8uYiYb3qFb5fNfL+KKR/IRnXQicmQ7/e7AJ8/B3I59FaROt+veal+6myvIPXndhEL/hcL0L2uyFjbF0KQ4UgO7FSdyvOIj5+Asbsa8Z253V+RiSIve2Y3ad07iYBqpm/NXbE7l9Ezuxo07lGoAt2JXduc7g887GYGUyXguxP6XaqxYiGIcHnYCHiAEdv+vzF6SsvxgJ4otnczqV7P2YAqOplFkxWV5HVLaYcaIo9KupZoMQFTBWdNnP6VpG+SKorPtFdZiGWyXIR6URN8DrfYTkzV3BnPwE+q3wmfEhKe+LRtt68WRzS7Pxxmh8Lr+5KvhmASrd14avVM4SxtrKlxCr0q1t5SuVs4SxtvIlxKp0a1v5SuUsYewTh3RtJ+XZFGMAAAAASUVORK5CYII=" width="23.5" height="19" /></span><span class = "S2"> seconds for a model with more than 10,000 reactions.</span></p><p class = "S4"><span class = "S21">Calculating maximal ATP energy production under anaerobic and glucose only conditions:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelspar = modelalter;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% For Recon3.0 model</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelspar = changeRxnBounds (modelspar, 'EX_glc_D[e]', -20, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelspar = changeRxnBounds (modelspar, 'EX_o2[e]', 0, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelspar = changeRxnBounds(modelspar, </span><span class = "S18">'EX_glc(e)'</span><span class = "S17">,-20,</span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelspar = changeRxnBounds (modelspar, </span><span class = "S18">'EX_o2(e)'</span><span class = "S17">, 0, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelspar = changeObjective(modelspar, </span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">[vSparse, sparseRxnBool, essentialRxnBool] = sparseFBA(modelspar, </span><span class = "S18">'max'</span><span class = "S12">);</span></p></div></div><ul class = "S6"><li class = "S7"><span class = "S8">Anticipated results:</span></li></ul></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">Commonly, a sparse FBA solution will have much smaller number of active reactions compared to a standard FBA on the same model with same objective function. The outputs </span><span class = "S14">sparseRxnBool </span><span class = "S2">and </span><span class = "S14">essentialRxnBool</span><span class = "S2"> return vectors with 1 and 0's, with sparse and essential reactions respectively.</span></p><p class = "S4"><span class = "S2">Display the sparse flux</span><span class = "S2"> solution, but only the non-zero fluxes.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S19">for </span><span class = "S12">i=1:length(vSparse)</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">    </span><span class = "S19">if </span><span class = "S12">abs(vSparse(i)) &gt; 1e-3</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">        fprintf(</span><span class = "S18">'%1.3f \t %s\n'</span><span class = "S12">, vSparse(i), modelspar.rxns{i})</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">    </span><span class = "S23">end</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S23">end</span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S20">3. Metabolite dilution flux balance analysis (mdFBA)</span></h2><p class = "S4"><span class = "S2">This is a variant of FBA for predicting metabolic flux distributions by accounting for growth-associated dilution of all metabolites in a context-dependent manner</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAABIklEQVRIS+3UsStGYRTH8c8bSliFmUEZbP4EKcVgVWYLMlCKxYCFQbGziM0iZTSSLFKK1SKLgSJ6eNR13fe9t95Fume6t/uc7znn9/zOragzKnXm+8eAdsxiEG/Yxk58/iFblgY9uMEWHrAYM5axlBY9DQjvJ5jHWTzchVO0oRePSUga0IkBHKYqbWAE/XiqBciyxXdX55jLGyH9vTEmTaAPr0UBoeoYDhIJC1jFe5ERQuVuPp06iamYNITjIoB0p+PYxSamk10U3YVmBBHvMJo0VFFAK26xhvW8ERoyLNuB+3gTV3lGusQFZnCNsBfBnSvYy7vGFuxjOHHwKC5WgP2KahoEUBNe8Fzrr1VUxKqMEvC1LHVFCShFDAYqffAXNPgAMKMtKaTYhDsAAAAASUVORK5CYII=" width="8" height="20" /></span><span class = "S2">.</span></p><p class = "S4"><span class = "S2">A solution from the function </span><span class = "S14">mdFBA</span><span class = "S2"> supports that all metabolites used in any reaction of the solution can either be produced by the network or taken up from the surrounding medium.</span></p><p class = "S4"><span class = "S21">Timing:</span></p><p class = "S4"><span class = "S2">Since this is a MIXED Integer Problem it can take a long time to solve.</span></p><p class = "S4"><span class = "S21">Calculating ATP energy production under aerobic condition using mdFBA:</span></p><p class = "S4"><span class = "S2">In this function, there is an optional output </span><span class = "S14">newActives</span><span class = "S2">, th</span><span class = "S2">at represent reactions that are only active in this analysis.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% The valid solution can be produced with the Recon 3.0 model</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelmd = model;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelmd = changeRxnBounds(modelmd, 'EX_glc_D[e]',-20,'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelmd = changeRxnBounds (modelmd, 'EX_o2[e]', -1000, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelmd = changeObjective(modelmd, 'DM_atp_c_');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% [sol, newActives] = mdFBA(modelmd)</span></p></div></div><ul class = "S6"><li class = "S7"><span class = "S8">Troubleshooting:</span></li></ul><p class = "S4"><span class = "S2">When a model do</span><span class = "S2">es not have a feasible solution, add the input:  </span><span class = "S14">'getInvalidSolution', true</span><span class = "S2">.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% clear modelmd</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelnosol = modelalter;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelnosol = changeObjective(modelnosol, </span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">[sol, newActives] = mdFBA(modelnosol,  </span><span class = "S18">'getInvalidSolution'</span><span class = "S12">, true)</span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">Sometimes when an FBA analysis of a model with the same objective function and constraints is run many times, or using different LP logarithm, we may get different set of solutions for individual reactions. In other words, there are different sets of </span><span class = "S14">'FBAsolution.x'</span><span class = "S2"> values (fluxes of the reactions) and still get the same objective function value </span><span class = "S14">'f'</span><span class = "S2">. Therefore, the opitmal solution is not unique. This can create difficulty when investigating the changes to fluxes between two different conditions. In this case a unique solution is required to compare the changes to fluxes.</span></p><p class = "S4"><span class = "S2">This issue can be solved with wither of the following the methods</span><span class = "S14"> </span></p><ul class = "S6"><li class = "S7"><span class = "S22">geometricFBA</span><span class = "S0">, which provides a standard, central and reproducible solution, or</span></li><li class = "S7"><span class = "S22">pFBA</span><span class = "S0">, which provides a solution based on the minimal fluxes through the model, and classify each gene according to how it contributes to the optimal solution.</span></li></ul><h2 class = "S3"><span class = "S20">4. Geometric FBA</span></h2><p class = "S4"><span class = "S2">The geometric FBA solves the smallest frame that contains all sets of optimal FBA solutions and posts a set of multiple linear programming problems</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAAA7ElEQVRIS+3VMUpDQRDG8V/A2tJOELQRa+9hpX2IlXa2Fp7AzlSxExQUEq9gkSaFVjYKgifwBIkM7IMkiHljOvO22WJ3/jvzzTdsy5KrtWS8FQFEmRd4xmBeszoaHKCPU1xlAZv4LEEn6GYAaxjiFpdIA87xhceSRQqwjxvsYQtvmQzWS0BAov6dDCC6cod7PBTBUoBoWRuxjwtgF684K2LONGLaB9XrhwsGbMYP80Y6wgYmU5AQMV4fFWFf8FSd13FiAD5wjF7GSNXdlIg/lb+Nd3Rw/ZcMftW0jgYNYIEtGxGtytfWzML/n4VvjUopKTCKDqoAAAAASUVORK5CYII=" width="8" height="20" /></span><span class = "S2">.</span></p><p class = "S4"><span class = "S2">This FBA analysis applies iterations, where by each iteration reduces the permissible solution space. After a finite number of iterations, it resolves one single solution of the flux distribution.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% USAGE:</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% flux = geometricFBA(model, varargin)</span></p></div></div><p class = "S13"><span class = "S21">Timing:</span></p><p class = "S4"><span class = "S2">The time to determine </span><span class = "S2">a geometric FBA solution depends on the size of the genome-scale model and the number of iterations. For a model with more than 10,000 reactions and several iterations takes </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAD4AAAAmCAYAAACcRCiyAAAEMUlEQVRoQ+2YW6hVVRSGv+MNLVQUtRcfLBS8kYKKltHVHhRvJKUS6IM3vGH5ICGaBKFUGKIWhFd88AqVpPQgihViRF6SwqhEIeihgswCIy/FfxzzMPY6c62z195nu89m7/m215pjrPmPf4x/jLmbqNPVVKe4aQCvN+YbjDcYr04EugDPA9OBh+0I54FdwJUijjQBeBq4C3QFzgGfATfTbDtCqg8GjgLDUw45BziU8k5BOma2vwJvAi9YEPR7igWhlXm1gfcDvgMGAG8DpwAFYnvipI8AVxPP+gPfmu1uYAlw2/YsBHYAfwFjgB+TyKsN/DXgVeAp4Ht3uF7AQWCyPVsOvO/e69wKzjJAzA4F/nDvOwEnjfkvgGddUJq3VRO42P4NGGmsJ0kRmMspwP27d4A1kVKYCXxkz5+zbGrZVk3gA4EnjNlYCXe3+lTtJxkX0LfM6ElArCaXLwWJ5CLgv7ApBlwGSptQL0WIakW2BFZVpyOAn+0r6gCngYmW5kOAG5ET+MCpHOTj9yzgUlDVllLoU0vD1LZQEcj3SvA9YCkwzZQ7fGqQE7pYfXtse4D59mAc8HUW8L1uc9i3BfgQuGhKWSG8zW6VcRKul0yZFyc+JtUPKp0FXGZbgZVm/xjwZRZwDQPzLNoxgGodH5sTiVN7rT7Ai8AHCYfbgNWu9Dzw44BELK0sPfACncgStx5WF1OBDSno1HcViM9dDZYSCF+PMXuBX2Xi5IGr5b1sE1vMriTg3pEERWIzCZgNKCuSq6CGSkFvNr3N/z4bToKr0cA3NuCEVK84cI9jFLA/MmK2F/DwLdW61DuMsitM8JKpPgO4kxJoz/hc3zqL6eOaojRk6ANhUFCL6ek+JvXXFHWtDKZjpo8DZ+xFqFE/vOQRtzZVXd/pC4y3tA7tIHkwKf0ngG5R19sZcHAXG2JExAVA83tb7ewIMMv2jfU6FGP8DeD1CBCxvBmQkupicT96u0CqnnWJ0eS108ZsjaLKwCzgPmg/WNbeympnGhyUtlpyvBE4YXNzy8hXIYaTbh8EvrI69zO9QKulaukOr8xLLj/oqJcX3PhijEsQ/gEOAz8BfwPdACl7liZobMwbGPmUTZo4CZTu6r6dCaBP9zRlD7bKVOnCLz4yacDDtJOH2Lyq7hnRnCDN8DP3MFN1BV716a+dOpe/fRVMZfYvjERRZ1oLbEoCiQHXrK5xMe+SAp/NYeTVWWZiRvdziaWAvAtofH4F+DPF7wKre9mqXQms5oD1gN6p2+gfGWVwwYoBf9RSKU/aKmUvRVjJioNsxPS6yCaVmS5JLZeKDEfPWH+WAIalQEinDqSVUTF9PAeJJW3VaPyAaYiCrXRvxVAbnjsDD5kW/WuinHmt7gjAS4pWuUYN4OVGsNbsG4zXGmPlnrfBeLkRrDX7BuO1xli5561bxv8H2LjnJ4sJ+8cAAAAASUVORK5CYII=" width="31" height="19" /></span><span class = "S2">minutes.</span></p><p class = "S4"><span class = "S21">Calculating ATP energy production under anaerobic conditions using geometric FBA:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelgeo = modelalter;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% For Recon3.0 model</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelgeo = changeRxnBounds (modelgeo, 'EX_glc_D[e]', -20, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%modelgeo = changeRxnBounds (modelgeo, 'EX_o2[e]', 0, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelgeo = changeRxnBounds(modelgeo, </span><span class = "S18">'EX_glc(e)'</span><span class = "S17">,-20,</span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelgeo = changeRxnBounds (modelgeo, </span><span class = "S18">'EX_o2(e)'</span><span class = "S17">, 0, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelgeo = changeObjective(modelgeo, </span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% WARNING: Depending on the size of the model running this function might take very long; </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% FBAgeo = geometricFBA (modelgeo, 'flexRel', 1e-3);</span></p></div></div><p class = "S13"><span class = "S2">Display the unique fluxe</span><span class = "S2">s from reactions, that are non-zero in the geometric FBA solution.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% for i=1:length(FBAgeo)</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%     if abs(FBAgeo(i)) &gt; 1e-3</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%         fprintf('%1.3f \t %s\n', FBAgeo(i), modelgeo.rxns{i})</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%     end</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% end</span></p></div></div><p class = "S13"><span class = "S21">Troubleshooting:</span></p><p class = "S4"><span class = "S2">When the algorithm has convergence problems, change one of the optional inputs, </span><span class = "S14">flexRel</span><span class = "S2">, into e.g. </span><span class = "S14">1e-3</span><span class = "S2">. The default is 0 when there is flexibility to flux bounds.</span></p><p class = "S4"><span class = "S2">Enter the optional parameters as parameter name followed by parameter value, for example:  </span></p><p class = "S4"><span class = "S14">flux = geometricFBA(model, 'epsilon', 1e-9)</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S20">5. Parsimonious enzyme usage Flux Balance Analysis (pFBA)</span></h2><p class = "S4"><span class = "S2">The pFBA method was developed to achieve higher flux levels when more enzymes are required</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAABJUlEQVRIS+3VsStGURjH8c+bgRhsJsogSiYlk2yMNguLxWLyZ5hMKJR/wUopyUAxGpDJoJgMNqFTh67r3Pve12vSPXWWc87zvff5Pb/nnIY2R6PNeP8cMINBvMcZ0h3HCt4+tSvSoBc36MuJPI2T7FoRYAlD2EVnDAh/cpv9elhPAbpwhQWcNStzCjCH/UzgGjZwn4KlAKuYjTMbM4XTPKTMSCH3SWxhFI8Yw1MVEbNnurGHeSxjp1VAON8fNQge2PwNIFTmHNtR0C9G1WYKgEusR2+UAnrwklN7AkcYwUNZCsGqx9FAi7jDMK5jWQ+blXEAF7keOIgNFGA/RkqDDoRmCnuveC6zc1URCxk1IH2hNLtDvu3XItYiFr1MtZFaUuAPjPQB4RopKUv7WGgAAAAASUVORK5CYII=" width="8" height="20" /></span><span class = "S2">. </span></p><p class = "S4"><span class = "S2">After performing the FBA to find the optimal value for the objective function, pFBA gets the answer of an another linear program to determine the flux distribution that minimises the total flux through all metabolic reactions in the model.</span></p><p class = "S4"><span class = "S21">Timing:</span></p><p class = "S4"><span class = "S2">The time to determine a pFBA solution depends on the size of the genome-scale model and is taking from </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAC8AAAAmCAYAAABUKMJkAAABwklEQVRYR+3XT6gPURTA8c8r5c/OgrKRZKWelSd/Q7F6K2+hFHmrt2FjQVHKRsmG8icUHit7JQtlKflTFlb0bGRBsrAVujVTY5r3vO69w2/qznJmzp3v+Z7TnHvHDPgaGzC7Av+/qlfMF/N/GtiL5Xi0kJhRa5utuIlNOI5rQ4DfjHvY2IA9huujDr8BT3EY47hSAQ8Cvil3HT4MFT5U4V2BX8T/ewV24yqm8GYRMQu98k/Mr8UhXGiQTODlqMIvwTacwIEW5GfsbPRrbA7Zza+qWuJGB9FthPuv8CuWuBGXDT4MjjDljnZYPo2H+JIBuLlEMvwk7mB1C2wWt/AcPzND18slwYe9zt2W7TnswqeegLOaP4PzLdDX1Z/lCb71mESS+ZprDQ7icgfoRdzH2x6SyAJfcy3FDpzFnnmq8RjfMyWSFb7JtB5HcK4D9FLVal8Tk+gNvuYKW4J9OFVVpb6/Hc8ywvd+GAmHh2mcxBa8iIQP0zsMw1DVmWqNj9iP9/jRtW6uY+DKqu87P/KXhJZVsyMc/ea7wj7qQfthLvhI4WlhBT7NX3x0MR/vLi2ymE/zFx9dzMe7S4ss5tP8xUf/BjcJViegBBPmAAAAAElFTkSuQmCC" width="23.5" height="19" /></span><span class = "S2"> minute for a 1,000 reaction model, to 5 minutes for a model with more than 10,000 reactions.</span></p><p class = "S4"><span class = "S2">The function is:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% [GeneClasses RxnClasses modelIrrevFM] = pFBA(model, varargin)</span></p></div></div><p class = "S13"><span class = "S2">Where 'varagin' includes required inputs:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% * 'geneoption' - 0 = minimize the sum of all fluxes in the network,</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%                   1 = only minimize the sum of the flux through</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%                   gene-associated fluxes (default),</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%                   2 = only minimize the sum of the flux through</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%                   non-gene-associated fluxes</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% * 'map' - map structure from readCbMap.m (no map written if empty)</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% * 'mapoutname' - File Name for map</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% * 'skipclass' - 0 = classify genes and reactions (default).</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%                 1 = Don't classify genes and reactions. Only return</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%                     model with the minimium flux set as upper bound.</span></p></div></div><p class = "S13"><span class = "S2">Given outp</span><span class = "S2">uts in this function are:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% OUTPUTS:</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% GeneClasses:  Structure with fields for each gene class</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% RxnsClasses:  Structure with fields for each reaction class</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelIrrevFM: Irreversible model used for minimizing flux with</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">%               the minimum flux set as a flux upper bound</span></p></div></div><p class = "S13"><span class = "S21">Calculating ATP energy production under anaerobic conditions using pFBA:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelp = modelalter;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% For Recon3.0 model</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelp = changeRxnBounds (modelp, 'EX_glc_D[e]', -20, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modelp = changeRxnBounds (modelp, 'EX_o2[e]', 0, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelp = changeRxnBounds(modelp, </span><span class = "S18">'EX_glc(e)'</span><span class = "S17">,-20,</span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelp = changeRxnBounds (modelp, </span><span class = "S18">'EX_o2(e)'</span><span class = "S17">, 0, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modelp = changeObjective(modelp, </span><span class = "S18">'DM_atp_c_'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">[GeneClasses RxnClasses modelIrrevFM] = pFBA(modelp,</span><span class = "S23">...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">    </span><span class = "S18">'geneoption'</span><span class = "S17">, 0, </span><span class = "S18">'skipclass'</span><span class = "S12">, 1)</span></p></div></div><p class = "S13"><span class = "S2">Display minimal fluxes of the r</span><span class = "S2">eactions that are required for producing energy only from only glucose media. </span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S19">for </span><span class = "S12">i=1:length(modelIrrevFM.lb)</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">    </span><span class = "S19">if </span><span class = "S12">modelIrrevFM.lb(i)~=0</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">        fprintf(</span><span class = "S18">'%1.3f \t %s\n'</span><span class = "S12">, modelIrrevFM.lb(i), modelIrrevFM.rxns{i})</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">    </span><span class = "S23">end</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S23">end</span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S20">6. Dynamic FBA</span></h2><p class = "S4"><span class = "S2">The dynamic FBA is an extension of standard FBA that accounts for cell culture dynamics, implementing both dynamic (nonlinear programming) and static (LP) optimisation of an objective function and applying constraints to the rates of change of flux in addition to the standard FBA constraints</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAABU0lEQVRIS+3UPyjuURzH8dczKAubwUYZdJVNycZkwaRbt8SgGzKRZJBV+TcalLoDg8WtO5nRTUwSSbEYDBYllL8d/Z76+Xme349M6jnTGc55dz5/vifniyv3xfu+H6Acd3HZH5HwA4MYxhJGcJ2HpAEqMIvf+Il13CdNLwaoxhaq0IzDYmkVApRhG03oxL+0qAsBBrCIZfTj+TOA4PIOGrGBC/RiHzNYxVNaCrU4jQ6sYQLBzHAxpDGHsfirkhLqcIJNtOIxgtXgLNrX47hYjHnAKOYT2qcxjhb8LwYI9KPoyT0Jvd0Istojf14ZSQkh9wPsoismIZz9hZUo3r20Jk5GRgXTzmMypjCEBlymASojE6/QhgfkvenDn48MU4huAR34G5uH4MGblTWN4TWh2je4LdTILEDmj1cCvG9ipmmfjTETWEqhlEKhPzGzOKUmvrfoBQTOOimCthPPAAAAAElFTkSuQmCC" width="8" height="20" /></span><span class = "S2">.</span></p><p class = "S4"><span class = "S2">The dynamic FBA metho</span><span class = "S2">d implemented in this function is essentially the same as the method described by Varma A. and B. O. Palsson</span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAoCAYAAAD32nqhAAABAklEQVRIS+3VsS4FQRTG8d+VkHgDhQcQkdCIQnIbnZ6H0OiUSlQqT6AinuAmoiA6Ci+gEVEoEBIlOcldkcnO7trtZKfds//55vvOnBnouAYd//dPAStYxFeJP5M4xkfxLfVgApdYzZh7jwV85gDLOME6XsdFoSTAI5xi7zc8VbCFMzwnCmbwhCXcVQFybbGJ/VR+FDfpg6g5xzV20x2aAGbxgPDnpg1gAwfjaH/iy6VQtkHIv8VOmUF1RyjcH+KqDSDkH2Ie738FhLroiUds5/KtOkIhfw0XbQAh/whzeGkDmMYU3qqmVl0KtROvBzS7zpVG9ib2JjZ9F/pGqplJnS/TN0myJCkB9o01AAAAAElFTkSuQmCC" width="8" height="20" /></span><span class = "S2">.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modeldinamic = model;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% For Recon3.0 model</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modeldinamic = changeRxnBounds (modeldinamic, 'EX_glc_D[e]', -20, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modeldinamic = changeRxnBounds (modeldinamic, 'EX_o2[e]', -1000, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% modeldinamic = changeRxnBounds (modeldinamic, 'EX_ac[e]', -1000, 'l');</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modeldinamic = changeRxnBounds (modeldinamic, </span><span class = "S18">'EX_glc(e)'</span><span class = "S17">, -20, </span><span class = "S18">'b'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modeldinamic = changeRxnBounds (modeldinamic, </span><span class = "S18">'EX_o2(e)'</span><span class = "S17">, -1000, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">modeldinamic = changeRxnBounds (modeldinamic, </span><span class = "S18">'EX_ac(e)'</span><span class = "S17">, -1000, </span><span class = "S18">'l'</span><span class = "S12">);</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% For Recon3.0 model</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% smi = {'EX_glc_D[e]' 'EX_ac[e]'}; </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">smi = {</span><span class = "S18">'EX_glc(e)' 'EX_ac(e)'</span><span class = "S12">}; </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% exchange reaction for substrate in environment</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">smc = [10.8]; </span><span class = "S15">% Glucose, Acetate concentration (all in mM)</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">Xec = 0.001; </span><span class = "S15">% initial biomass</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">dt = 1.0/1000.0; </span><span class = "S15">% time steps</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">time = 1.0/dt; </span><span class = "S15">% simulation time</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12"></span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">[concentrationMatrix, excRxnNames, timeVec,</span><span class = "S23">...</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">    biomassVec] = dynamicFBA(modeldinamic, smi, smc, Xec, dt, time, smi ); </span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S20">7. Relax FBA</span></h2><p class = "S4"><span class = "S2">Find the minimal set of </span><span class = "S2">relaxations on bounds and steady-state constraint to make the FBA problem feasible.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelrelax = modelalter;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">FBArel = relaxedFBA(modelrelax)</span></p></div></div><p class = "S13"><span class = "S2">The output </span><span class = "S14">FBArel</span><span class = "S2"> contains solution fields, where </span></p><p class = "S4"><span class = "S14">FBArel.v</span><span class = "S2"> is the reaction rate; </span></p><p class = "S4"><span class = "S14">FBArel.r</span><span class = "S2"> is set of reactions that need relaxation on steady state constraints </span><span class = "S14">S*v = b</span><span class = "S2">;</span></p><p class = "S4"><span class = "S14">FBArel.p</span><span class = "S2"> is relaxation on lower bound of reactions;</span></p><p class = "S4"><span class = "S14">FBArel.r</span><span class = "S2"> is relaxation on upper bound of reactions;</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S20">8. Flux enrichment analysis (FEA)</span></h2><p class = "S4"><span class = "S2">The flux enrichment analysis calculates the likelihood th</span><span class = "S2">at a set of fluxes would belong to a subsystem or pathway.</span></p><p class = "S4"><span class = "S21">Timing:</span></p><p class = "S4"><span class = "S2">The time to calcu</span><span class = "S2">late the FEA is </span><span style="vertical-align:-3"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAC8AAAAmCAYAAABUKMJkAAABwklEQVRYR+3XT6gPURTA8c8r5c/OgrKRZKWelSd/Q7F6K2+hFHmrt2FjQVHKRsmG8icUHit7JQtlKflTFlb0bGRBsrAVujVTY5r3vO69w2/qznJmzp3v+Z7TnHvHDPgaGzC7Av+/qlfMF/N/GtiL5Xi0kJhRa5utuIlNOI5rQ4DfjHvY2IA9huujDr8BT3EY47hSAQ8Cvil3HT4MFT5U4V2BX8T/ewV24yqm8GYRMQu98k/Mr8UhXGiQTODlqMIvwTacwIEW5GfsbPRrbA7Zza+qWuJGB9FthPuv8CuWuBGXDT4MjjDljnZYPo2H+JIBuLlEMvwk7mB1C2wWt/AcPzND18slwYe9zt2W7TnswqeegLOaP4PzLdDX1Z/lCb71mESS+ZprDQ7icgfoRdzH2x6SyAJfcy3FDpzFnnmq8RjfMyWSFb7JtB5HcK4D9FLVal8Tk+gNvuYKW4J9OFVVpb6/Hc8ywvd+GAmHh2mcxBa8iIQP0zsMw1DVmWqNj9iP9/jRtW6uY+DKqu87P/KXhJZVsyMc/ea7wj7qQfthLvhI4WlhBT7NX3x0MR/vLi2ymE/zFx9dzMe7S4ss5tP8xUf/BjcJViegBBPmAAAAAElFTkSuQmCC" width="23.5" height="19" /></span><span class = "S2"> second for any size of a model.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">modelfea = model;</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S17">res = optimizeCbModel(modelfea,</span><span class = "S18">'max'</span><span class = "S12">); </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S15">% say you are interested in enriching the active reactions</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">activeReactions = find(res.x) </span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">% You can also look for e.g. positive/negative/zeros flux reactions,</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">% that depends pretty much on the question.</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">% Now you look for the enrichement of reactions per subsystems</span></p></div><div class = 'inlineWrapper'><p class = "S11 lineNode"><span class = "S12">resultCell = FEA(modelfea, activeReactions, 'subSystems')</span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S9"><span class = "S2">REFERENCES </span></h2><p class = "S4"><span class = "S2">[1] Orth, J. D., Thiele I., and Palsson, B. Ø.  What is flux balance analysis? </span><span class = "S25">Nat. Biotechnol., </span><span class = "S2">28(3), 245–248 (2010).</span></p><p class = "S4"><span class = "S2">[2] Thiele, I., et al. </span><span class = "S2">A community-driven global reconstruction of human metabolism. </span><span class = "S25">Nat. Biotechnol., </span><span class = "S2">31(5), 419–425 (2013).</span></p><p class = "S4"><span class = "S2">[3] Benyamini, T, Folger, O., Ruppin, E., Schlomi, T. Flux balance analysis accounting for metabolite dilution.</span><span class = "S25"> Genome Biology</span><span class = "S2">., 11(4):R43 (2010).</span></p><p class = "S4"><span class = "S2">[4] Smallbone, K., and Simeonidis, E. Flux balance analysis: A geometric perspective. </span><span class = "S25">J Theor Biol</span><span class = "S2">., 258: 311-315 (2009).</span></p><p class = "S4"><span class = "S2">[5] Lewis, N.E., et al. Omic data from evolved E. coli are consistent with computed optimal growth from genome-scale models. </span><span class = "S25">Mol Syst Biol</span><span class = "S2">., 6:390 (2010).</span></p><p class = "S4"><span class = "S2">[6] Mahadevan, R., Edwards, J.S., Doyle, F.J. Dynamic Flux Balance Analysis of Diauxic Growth in Escherichia coli. </span><span class = "S25">Biophys J., </span><span class = "S2">83(3):1331-1340 (2002).</span></p><p class = "S4"><span class = "S2">[7] Varma A. and Palsson, B. Ø. Stoichiometric flux balance models quantitatively predict growth and metabolic by-product secretion in wild-type Escherichia coli W3110. </span><span class = "S25">App Environ Microbiol</span><span class = "S2">., 60(10):3724-3731 (1994).</span></p><p class = "S4"><span class = "S2"></span></p></div></div>
<!-- 
##### SOURCE BEGIN #####
%% Flux Balance Analysis (FBA)
%% Author(s): *Vanja Vlasov, *Marouen Ben Guebila, *Systems Biochemistry Group, LCSB, University of Luxembourg, *
%% *Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg* 
%%  Reviewer(s): Ines Thiele,* Catherine Clancy, Systems Biochemistry Group, LCSB, University of Luxembourg*
%% *Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg* 
%% INTRODUCTION
% Flux balance analysis (FBA) evaluates the metabolic flux distribution$$^1$, 
% and is one of the most used modelling approaches for metabolic systems. 
% 
% The applications of FBA for molecular systems biology include prediction 
% of the growth rates, uptake rates, knockout lethality and product secretion. 
% In FBA, the solution space is constrained by the assumption of a steady-state, 
% under which each internal metabolite is consumed at the same rate as it is produced.
% 
% For the quantitative estimation of the metabolic fluxes, linear programming 
% (LP) can be used to solve the stoichiometric matrix for a given objective function 
% under different constraints. The constraints of the problem depict the space 
% of all eligible possibilities from which an optimal solution can be selected; 
% 
% $$\begin{array}{ll}\min\limits _{v} & \ c^{T}v\\\text{s.t.} & Sv=b,\\ & 
% l\leq v\leq u,\end{array}$$
% 
% Equation 1: Formula of standard FBA.
% 
% where $$c\in\Re^{n}$$ is a parameter vector that linearly combines one 
% or more reaction fluxes to form what is termed the objective function,  and 
% where a $$b_{i}<0$$, or  $$b_{i}>0$$, represents some fixed output, or input, 
% of the ith molecular species. $$S\in\Re^{m\times n}$$ is a stoichiometric matrix 
% for $$m$ molecular species and $$n$ reactions, and $$b$ is a vector of known 
% metabolic exchanges. The output of FBA is a particular flux distribution, $$v$ 
% , which maximises or minimises the objective function and stands between upper 
% and lower bounds, $$u$ and $$l$, respectively.
% 
% There are multiple different variants of FBA which will be discussed here:
% 
% # *Standard FBA*
% # *Sparse FBA*
% # *Metabolite dilution FBA (mdFBA)*
% # *Geometric FBA*
% # *Parsimonious enzyme usage Flux Balance Analysis (pFBA)*
% # *Dynamic FBA*
% # *Relax FBA*
% # *Flux enrichment analysis (FEA)*
%% EQUIPMENT SETUP
%% Initialise The Cobra Toolbox and set the solver.
% If necessary, initialise the cobra toolbox:

 initCobraToolbox
%% 
% For solving LP problems in a FBA analysis, certain solvers are required 
% and can be set using the |changeCobraSolver| function:

% solverOK = changeCobraSolver(solverName, solverType, printLevel, unchecked)
%% 
% The present tutorial can run with <https://opencobra.github.io/cobratoolbox/deprecated/docs/cobra/solvers/changeCobraSolver.html 
% glpk package>, which does not require additional installation and configuration. 
% Although, for the analysis of large models is recommended to use the <https://github.com/opencobra/cobratoolbox/blob/master/docs/source/installation/solvers.md 
% GUROBI> package.
% 
% Setup the appropriate solver for the machine you are using by removing 
% the "%" (comment) sign for only the desired solver.

 changeCobraSolver('glpk','all');
% changeCobraSolver('tomlab_cplex','all');
% changeCobraASolver('ibm_cplex','all');
% changeCobraSolver ('gurobi', 'all');
%% Model Setup
% This tutorial will use the generic model of the human cellular metabolism$$ 
% ^2$, Recon 2.0. Other COBRA models, including Recon 3, may also be run with 
% this tutorial. For information on metabolites structures and reactions, and 
% to download the latest COBRA model releases, visit the Virtual Metabolic Human 
% database (VMH, <http://vmh.life). http://vmh.life).>
% 
% Before proceeding with the simulations, load the model into the workspace:

global CBTDIR
modelFileName = 'Recon2.0model.mat';
modelDirectory = getDistributedModelFolder(modelFileName); %Look up the folder for the distributed Models.
modelFileName= [modelDirectory filesep modelFileName]; % Get the full path. Necessary to be sure, that the right model is loaded
model = readCbModel(modelFileName);
%% 
% In this tutorial we assume, that the cellular objectives include energy 
% production or optimisation of uptake rates and by-product secretion for various 
% physiological functions of the human body.
%% PROCEDURE
%% 1. Standard FBA
% Standard FBA predicts an optimal solution for a cellular objective within 
% a given set of constraints on a metabolic network (see Equation 1). Constraints 
% on the network are set by assigning limits on the uptake, consumption or production 
% of metabolites in reactions.
% 
% *Timing:*
% 
% The time to determine a FBA solution depends on the size of the genome-scale 
% model and is commonly less than a second for a medium sized model.
% 
% *Calculating maximal ATP energy production under aerobic conditions:*
% 
% For each new simulation, the original model will be copied to a new variable. 
% This preserves the constraints of the original model to perform further simulations 
% with new constraints. Additionally, this method of renaming the model avoids 
% confusion while performing multiple simulations at the same time.

modelaerobic = model;
%% 
% The ATP demand reaction, i.e., |<http://vmh.life/#human/all/DM_atp_c_ 
% DM_atp_c_> |within the model is a reaction that involves hydrolysis of ATP to 
% ADP, Pi and proton in the cytosol. 

 printRxnFormula(model, 'DM_atp_c_');
%% 
% We will set this reaction as our objective with the |'changeObjective'| 
% command. Maximising the flux through the ATP demand reaction will result in 
% the network producing a maximal amount of ATP (up to the limit of the reaction).

modelaerobic = changeObjective (modelaerobic, 'DM_atp_c_');
%% 
% The glucose and oxygen, in this case, are provided in high amounts for 
% calculating the flux through ATP demand. 
% 
% The |'changeRxnBounds' |function changes the flux constraints of the lower 
% ('|l|'), upper ('|u|'), or both the bounds ('|b|'), of the specified reaction. 
% Here, we will change the maximal uptake of glucose to 20 $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mi>&mu;</mi></mrow></math>$mol/min/gDW and of oxygen 
% to 1000 $<math xmlns="http://www.w3.org/1998/Math/MathML" display="inline"><mrow><mi>&mu;</mi></mrow></math>$mol/min/gDW. 
% The uptake of oxygen is effectively unconstrainted (i.e. infinity). 

% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and 
%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.
modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc(e)', -20, 'l');  
modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2(e)', -1000, 'l'); 
%% 
% The function |optimizeCbModel| calculates one of the optimal solutions 
% for a (maximum or minimum) objective reaction within the defined solution space. 
% In the above example, the maximal flux through the |<http://vmh.life/#human/all/DM_atp_c_ 
% DM_atp_c_>| is desired. 

FBAaerobic = optimizeCbModel (modelaerobic, 'max')
%% 
% * *Anticipated results*
% 
% When oxygen and all carbon sources (internal and external) are provided 
% the flux through ATP demand reaction can reach its maximum rate of 1000 $<math 
% xmlns="http://www.w3.org/1998/Math/MathML" display="inline"><mrow><mi>&mu;</mi></mrow></math>$mol/min/gDW. 
% 
% * *Troubleshooting*
% 
% If there are multiple carbon sources available in the model, it may be 
% necessary to specify more constraints in order to examine the effect of a single 
% carbon source on ATP production.  
% 
% To avoid this issue, all external carbon sources need to be closed with 
% the exception of the single carbon source of interest.

%Closing the uptake of all energy and oxygen sources
[exchBool,uptBool] = findExcRxns(model);
uptakes = model.rxns(uptBool);

% If you use Recon3.0 model, then:
% modelalter = model;
% modelalter = changeRxnBounds(modelalter, uptakes, 0, 'b');
% modelalter = changeRxnBounds(modelalter, 'EX_HC00250[e]', -1000, 'l');

% The alternative way to do that, in case you were using another large model, 
% that does not contain defined Subsystem is
% to find uptake exchange reactions with following codes:
% [selExc, selUpt] = findExcRxns(model);
% uptakes1 = model.rxns(selUpt);

% Selecting from the exchange uptake reactions those 
% which contain at least 1 carbon in the metabolites included in the reaction:
 subuptakeModel = extractSubNetwork(model, uptakes);
 hiCarbonRxns = findCarbonRxns(subuptakeModel,1);
% Closing the uptake of all the carbon sources
 modelalter = model;
 modelalter = changeRxnBounds(modelalter, hiCarbonRxns, 0, 'b');
% Closing other oxygen and energy sources. Use the following lines for recon2, or uncomment the lines below for recon3
 exoxygen = {'EX_adp',   'EX_amp(e)',    'EX_atp(e)',    'EX_co2(e)',    'EX_coa(e)',    'EX_fad(e)',    'EX_fe2(e)',...
     'EX_fe3(e)',    'EX_gdp(e)',   'EX_gmp(e)',    'EX_gtp(e)',    'EX_h(e)',    'EX_h2o(e)',    'EX_h2o2(e)',...
     'EX_nad(e)',    'EX_nadp(e)',    'EX_no(e)',    'EX_no2(e)',    'EX_o2s(e)'};
 %exoxygen = {'EX_adp',   'EX_amp[e]',    'EX_atp[e]',    'EX_co2[e]',    'EX_coa[e]',    'EX_fad[e]',    'EX_fe2[e]',...
 %    'EX_fe3[e]',    'EX_gdp[e]',   'EX_gmp[e]',    'EX_gtp[e]',    'EX_h[e]',    'EX_h2o[e]',    'EX_h2o2[e]',...
 %    'EX_nad[e]',    'EX_nadp[e]',    'EX_no[e]',    'EX_no2[e]',    'EX_o2s[e]'};
modelalter = changeRxnBounds (modelalter, exoxygen, 0, 'l');
%% 
% *Calculating maximum ATP energy production under anaerobic and glucose 
% only conditions:*

modelanaerobic = modelalter;
% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and 
%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.
modelanaerobic = changeRxnBounds(modelanaerobic, 'EX_glc(e)',-20,'l');
modelanaerobic = changeRxnBounds (modelanaerobic, 'EX_o2(e)', 0, 'l');
modelanaerobic = changeObjective(modelanaerobic,'DM_atp_c_');
FBAanaerob = optimizeCbModel(modelanaerobic,'max')
%% 
% * *Anticipated results*
% 
% Compared to the aerobic condition, anaerobic condition with only glucose 
% as an energy source has reduced flux through ATP demand (82 $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mi>&mu;</mi></mrow></math>$mol/min/gDW), signifying 
% the need to oxygen to run the oxidative phosphorylation. The results are dependant 
% on the model you are using. For Recon 3.0, under anaerobic conditions with only 
% glucose as an energy source, the flux for ATP demand is 40 $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mi>&mu;</mi></mrow></math>$mol/min/gDW.
%% 2. Sparse FBA
% Sparse FBA calculates the optimal solution of an objective function and finds 
% the smallest set of reactions that can carry flux to achieve the objective. 
% Sparse FBA minimises the number of reactions by keeping same maximal objective;
% 
% $$\begin{array}{ll}\min\limits _{v} & \ \\\text{s.t.} & Sv=b,\\ & l\leq 
% v\leq u,\\& c^{T}v=\rho^*\end{array}$$
% 
% Equation 2: Formula of Sparse FBA.
% 
% where the last constraint is optional and represents the requirement to 
% satisfy an optimal objective value $\rho^*$ derived from any solution to a FBA 
% problem. This approach is used to check for minimal sets of reactions that either 
% should be active or should not be active in a flux balance model that is representative 
% of a biochemical network.

% [vSparse, sparseRxnBool, essentialRxnBool]  = sparseFBA(model, osenseStr,...
%  checkMinimalSet, checkEssentialSet, zeroNormApprox)
%% 
% As an optional input, there are different appoximation types of zero-norm 
% (only available when |minNorm = 'zero'|). Default is |cappedL1|.

% Other types of zero-norm:
%  * 'cappedL1' : Capped-L1 norm
%  * 'exp'      : Exponential function
%  * 'log'      : Logarithmic function
%  * 'SCAD'     : SCAD function
%  * 'lp-'      : :math:`L_p` norm with :math:`p < 0`
%  * 'lp+'      : :math:`L_p` norm with :math:`0 < p < 1`
%  * 'l1'       : L1 norm
%  * 'all'      : try all approximations and return the best result
%% 
% *Timing:*
% 
% The time to determine a |sparseFBA()| solution depends on the size of the 
% genome-scale model and is taking from $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mo stretchy="false">&lt;</mo><mn>1</mn></mrow></math>$ 
% second for a 1,000 reaction model, to $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mo>&lt;</mo><mn>2</mn></mrow></math>$ seconds for a 
% model with more than 10,000 reactions.
% 
% *Calculating maximal ATP energy production under anaerobic and glucose 
% only conditions:*

modelspar = modelalter;
% For Recon3.0 model
% modelspar = changeRxnBounds (modelspar, 'EX_glc_D[e]', -20, 'l');
% modelspar = changeRxnBounds (modelspar, 'EX_o2[e]', 0, 'l');
modelspar = changeRxnBounds(modelspar, 'EX_glc(e)',-20,'l');
modelspar = changeRxnBounds (modelspar, 'EX_o2(e)', 0, 'l');
modelspar = changeObjective(modelspar, 'DM_atp_c_');
[vSparse, sparseRxnBool, essentialRxnBool] = sparseFBA(modelspar, 'max');
%% 
% * *Anticipated results:*
% 
% Commonly, a sparse FBA solution will have much smaller number of active 
% reactions compared to a standard FBA on the same model with same objective function. 
% The outputs |sparseRxnBool |and |essentialRxnBool| return vectors with 1 and 
% 0's, with sparse and essential reactions respectively.
% 
% Display the sparse flux solution, but only the non-zero fluxes.

for i=1:length(vSparse)
    if abs(vSparse(i)) > 1e-3
        fprintf('%1.3f \t %s\n', vSparse(i), modelspar.rxns{i})
    end
end
%% 3. Metabolite dilution flux balance analysis (mdFBA)
% This is a variant of FBA for predicting metabolic flux distributions by accounting 
% for growth-associated dilution of all metabolites in a context-dependent manner$$^3$.
% 
% A solution from the function |mdFBA| supports that all metabolites used 
% in any reaction of the solution can either be produced by the network or taken 
% up from the surrounding medium.
% 
% *Timing:*
% 
% Since this is a MIXED Integer Problem it can take a long time to solve.
% 
% *Calculating ATP energy production under aerobic condition using mdFBA:*
% 
% In this function, there is an optional output |newActives|, that represent 
% reactions that are only active in this analysis.

% The valid solution can be produced with the Recon 3.0 model
% modelmd = model;
% modelmd = changeRxnBounds(modelmd, 'EX_glc_D[e]',-20,'l');
% modelmd = changeRxnBounds (modelmd, 'EX_o2[e]', -1000, 'l');
% modelmd = changeObjective(modelmd, 'DM_atp_c_');

% [sol, newActives] = mdFBA(modelmd)
%% 
% * *Troubleshooting:*
% 
% When a model does not have a feasible solution, add the input:  |'getInvalidSolution', 
% true|.

% clear modelmd
modelnosol = modelalter;
modelnosol = changeObjective(modelnosol, 'DM_atp_c_');
[sol, newActives] = mdFBA(modelnosol,  'getInvalidSolution', true)
%% 
% Sometimes when an FBA analysis of a model with the same objective function 
% and constraints is run many times, or using different LP logarithm, we may get 
% different set of solutions for individual reactions. In other words, there are 
% different sets of |'FBAsolution.x'| values (fluxes of the reactions) and still 
% get the same objective function value |'f'|. Therefore, the opitmal solution 
% is not unique. This can create difficulty when investigating the changes to 
% fluxes between two different conditions. In this case a unique solution is required 
% to compare the changes to fluxes.
% 
% This issue can be solved with wither of the following the methods| |
% 
% * |geometricFBA|, which provides a standard, central and reproducible solution, 
% or
% * |pFBA|, which provides a solution based on the minimal fluxes through the 
% model, and classify each gene according to how it contributes to the optimal 
% solution.
%% 4. Geometric FBA
% The geometric FBA solves the smallest frame that contains all sets of optimal 
% FBA solutions and posts a set of multiple linear programming problems$$^4$.
% 
% This FBA analysis applies iterations, where by each iteration reduces the 
% permissible solution space. After a finite number of iterations, it resolves 
% one single solution of the flux distribution.

% USAGE:
% flux = geometricFBA(model, varargin)
%% 
% *Timing:*
% 
% The time to determine a geometric FBA solution depends on the size of the 
% genome-scale model and the number of iterations. For a model with more than 
% 10,000 reactions and several iterations takes $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mo>&geq;</mo><mn>30</mn></mrow></math>$minutes.
% 
% *Calculating ATP energy production under anaerobic conditions using geometric 
% FBA:*

modelgeo = modelalter;
% For Recon3.0 model
% modelgeo = changeRxnBounds (modelgeo, 'EX_glc_D[e]', -20, 'l');
%modelgeo = changeRxnBounds (modelgeo, 'EX_o2[e]', 0, 'l');
modelgeo = changeRxnBounds(modelgeo, 'EX_glc(e)',-20,'l');
modelgeo = changeRxnBounds (modelgeo, 'EX_o2(e)', 0, 'l');
modelgeo = changeObjective(modelgeo, 'DM_atp_c_');
% WARNING: Depending on the size of the model running this function might take very long; 
% FBAgeo = geometricFBA (modelgeo, 'flexRel', 1e-3);
%% 
% Display the unique fluxes from reactions, that are non-zero in the geometric 
% FBA solution.

% for i=1:length(FBAgeo)
%     if abs(FBAgeo(i)) > 1e-3
%         fprintf('%1.3f \t %s\n', FBAgeo(i), modelgeo.rxns{i})
%     end
% end
%% 
% *Troubleshooting:*
% 
% When the algorithm has convergence problems, change one of the optional 
% inputs, |flexRel|, into e.g. |1e-3|. The default is 0 when there is flexibility 
% to flux bounds.
% 
% Enter the optional parameters as parameter name followed by parameter value, 
% for example:  
% 
% |flux = geometricFBA(model, 'epsilon', 1e-9)|
%% 5. Parsimonious enzyme usage Flux Balance Analysis (pFBA)
% The pFBA method was developed to achieve higher flux levels when more enzymes 
% are required$$^5$. 
% 
% After performing the FBA to find the optimal value for the objective function, 
% pFBA gets the answer of an another linear program to determine the flux distribution 
% that minimises the total flux through all metabolic reactions in the model.
% 
% *Timing:*
% 
% The time to determine a pFBA solution depends on the size of the genome-scale 
% model and is taking from $<math xmlns="http://www.w3.org/1998/Math/MathML" display="inline"><mrow><mo 
% stretchy="false">&lt;</mo><mn>1</mn></mrow></math>$ minute for a 1,000 reaction 
% model, to 5 minutes for a model with more than 10,000 reactions.
% 
% The function is:

% [GeneClasses RxnClasses modelIrrevFM] = pFBA(model, varargin)
%% 
% Where 'varagin' includes required inputs:

% * 'geneoption' - 0 = minimize the sum of all fluxes in the network,
%                   1 = only minimize the sum of the flux through
%                   gene-associated fluxes (default),
%                   2 = only minimize the sum of the flux through
%                   non-gene-associated fluxes
%
% * 'map' - map structure from readCbMap.m (no map written if empty)
%
% * 'mapoutname' - File Name for map
%
% * 'skipclass' - 0 = classify genes and reactions (default).
%                 1 = Don't classify genes and reactions. Only return
%                     model with the minimium flux set as upper bound.
%% 
% Given outputs in this function are:

% OUTPUTS:
% GeneClasses:  Structure with fields for each gene class
% RxnsClasses:  Structure with fields for each reaction class
% modelIrrevFM: Irreversible model used for minimizing flux with
%               the minimum flux set as a flux upper bound
%% 
% *Calculating ATP energy production under anaerobic conditions using pFBA:*

modelp = modelalter;
% For Recon3.0 model
% modelp = changeRxnBounds (modelp, 'EX_glc_D[e]', -20, 'l');
% modelp = changeRxnBounds (modelp, 'EX_o2[e]', 0, 'l');
modelp = changeRxnBounds(modelp, 'EX_glc(e)',-20,'l');
modelp = changeRxnBounds (modelp, 'EX_o2(e)', 0, 'l');
modelp = changeObjective(modelp, 'DM_atp_c_');
[GeneClasses RxnClasses modelIrrevFM] = pFBA(modelp,...
    'geneoption', 0, 'skipclass', 1)
%% 
% Display minimal fluxes of the reactions that are required for producing 
% energy only from only glucose media. 

for i=1:length(modelIrrevFM.lb)
    if modelIrrevFM.lb(i)~=0
        fprintf('%1.3f \t %s\n', modelIrrevFM.lb(i), modelIrrevFM.rxns{i})
    end
end
%% 6. Dynamic FBA
% The dynamic FBA is an extension of standard FBA that accounts for cell culture 
% dynamics, implementing both dynamic (nonlinear programming) and static (LP) 
% optimisation of an objective function and applying constraints to the rates 
% of change of flux in addition to the standard FBA constraints$$^6$.
% 
% The dynamic FBA method implemented in this function is essentially the 
% same as the method described by Varma A. and B. O. Palsson$$^7$.

modeldinamic = model;
% For Recon3.0 model
% modeldinamic = changeRxnBounds (modeldinamic, 'EX_glc_D[e]', -20, 'l');
% modeldinamic = changeRxnBounds (modeldinamic, 'EX_o2[e]', -1000, 'l');
% modeldinamic = changeRxnBounds (modeldinamic, 'EX_ac[e]', -1000, 'l');

modeldinamic = changeRxnBounds (modeldinamic, 'EX_glc(e)', -20, 'b');
modeldinamic = changeRxnBounds (modeldinamic, 'EX_o2(e)', -1000, 'l');
modeldinamic = changeRxnBounds (modeldinamic, 'EX_ac(e)', -1000, 'l');
% For Recon3.0 model
% smi = {'EX_glc_D[e]' 'EX_ac[e]'}; 
smi = {'EX_glc(e)' 'EX_ac(e)'}; 
% exchange reaction for substrate in environment

smc = [10.8]; % Glucose, Acetate concentration (all in mM)

Xec = 0.001; % initial biomass
dt = 1.0/1000.0; % time steps
time = 1.0/dt; % simulation time

[concentrationMatrix, excRxnNames, timeVec,...
    biomassVec] = dynamicFBA(modeldinamic, smi, smc, Xec, dt, time, smi ); 
%% 7. Relax FBA
% Find the minimal set of relaxations on bounds and steady-state constraint 
% to make the FBA problem feasible.

modelrelax = modelalter;
FBArel = relaxedFBA(modelrelax)
%% 
% The output |FBArel| contains solution fields, where 
% 
% |FBArel.v| is the reaction rate; 
% 
% |FBArel.r| is set of reactions that need relaxation on steady state constraints 
% |S*v = b|;
% 
% |FBArel.p| is relaxation on lower bound of reactions;
% 
% |FBArel.r| is relaxation on upper bound of reactions;
%% 8. Flux enrichment analysis (FEA)
% The flux enrichment analysis calculates the likelihood that a set of fluxes 
% would belong to a subsystem or pathway.
% 
% *Timing:*
% 
% The time to calculate the FEA is $<math xmlns="http://www.w3.org/1998/Math/MathML" 
% display="inline"><mrow><mo stretchy="false">&lt;</mo><mn>1</mn></mrow></math>$ 
% second for any size of a model.

modelfea = model;
res = optimizeCbModel(modelfea,'max'); 
% say you are interested in enriching the active reactions
activeReactions = find(res.x) 
% You can also look for e.g. positive/negative/zeros flux reactions,
% that depends pretty much on the question.
% Now you look for the enrichement of reactions per subsystems
resultCell = FEA(modelfea, activeReactions, 'subSystems')
%% REFERENCES 
% [1] Orth, J. D., Thiele I., and Palsson, B. Ø.  What is flux balance analysis? 
% _Nat. Biotechnol., _28(3), 245–248 (2010).
% 
% [2] Thiele, I., et al. A community-driven global reconstruction of human 
% metabolism. _Nat. Biotechnol., _31(5), 419–425 (2013).
% 
% [3] Benyamini, T, Folger, O., Ruppin, E., Schlomi, T. Flux balance analysis 
% accounting for metabolite dilution._ Genome Biology_., 11(4):R43 (2010).
% 
% [4] Smallbone, K., and Simeonidis, E. Flux balance analysis: A geometric 
% perspective. _J Theor Biol_., 258: 311-315 (2009).
% 
% [5] Lewis, N.E., et al. Omic data from evolved E. coli are consistent with 
% computed optimal growth from genome-scale models. _Mol Syst Biol_., 6:390 (2010).
% 
% [6] Mahadevan, R., Edwards, J.S., Doyle, F.J. Dynamic Flux Balance Analysis 
% of Diauxic Growth in Escherichia coli. _Biophys J., _83(3):1331-1340 (2002).
% 
% [7] Varma A. and Palsson, B. Ø. Stoichiometric flux balance models quantitatively 
% predict growth and metabolic by-product secretion in wild-type Escherichia coli 
% W3110. _App Environ Microbiol_., 60(10):3724-3731 (1994).
% 
%
##### SOURCE END #####
--></body></html>