<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><script type="text/javascript" src="https://cdn.rawgit.com/opencobra/cobratoolbox/gh-pages/latest/_static/js/iframeResizer.contentWindow.min.js"></script><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="generator" content="MATLAB R2016a"><meta http-equiv="X-UA-Compatible" content="IE=edge,IE=9,chrome=1"><title>Extraction of context-specific models</title><style type="text/css">
* {margin: 0; padding: 0;}
body {text-align: start; line-height: 17.234px; min-height: 0px; white-space: normal; color: rgb(0, 0, 0); font-family: Consolas, Inconsolata, Menlo, monospace; font-style: normal; font-size: 14px; font-weight: normal; text-decoration: none; white-space: normal; }
h1, h2 {font-weight: normal;}
.content { padding: 30px; }

.S0 { margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S1 { line-height: 26.4px; min-height: 24px; white-space: normal; color: rgb(213, 80, 0); font-family: Helvetica, Arial, sans-serif; font-size: 22px; white-space: normal; margin-left: 4px; margin-top: 3px; margin-bottom: 15px; margin-right: 10px;  }
.S2 { min-height: 0px; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S3 { line-height: 20.576px; min-height: 20px; white-space: normal; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-size: 16px; font-weight: bold; white-space: normal; margin-left: 4px; margin-top: 15px; margin-bottom: 9px; margin-right: 10px;  }
.S4 { line-height: 21px; min-height: 17px; white-space: normal; font-family: Helvetica, Arial, sans-serif; white-space: normal; margin-left: 4px; margin-top: 2px; margin-bottom: 9px; margin-right: 10px;  }
.S5 { font-family: Helvetica, Arial, sans-serif; margin-left: 0px; margin-top: 10px; margin-bottom: 20px; margin-right: 0px;  }
.S6 { text-align: left; line-height: 21px; white-space: normal; white-space: normal; margin-left: 56px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S7 { min-height: 0px; font-family: monospace; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S8 { margin-left: 3px; margin-top: 10px; margin-bottom: 10px; margin-right: 3px;  }
.S9 { line-height: 15.5927px; min-height: 18px; white-space: nowrap; font-size: 12.6667px; white-space: nowrap; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S10 { min-height: 0px; white-space: normal; color: rgb(0, 0, 255); white-space: normal; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S11 { min-height: 0px; white-space: normal; white-space: normal; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S12 { min-height: 0px; white-space: normal; color: rgb(0, 0, 255); white-space: normal; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S13 { min-height: 0px; white-space: normal; white-space: normal; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S14 { min-height: 0px; white-space: normal; color: rgb(160, 32, 240); white-space: normal; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S15 { color: rgb(64, 64, 64); margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S16 { line-height: 21px; min-height: 17px; white-space: normal; font-family: Helvetica, Arial, sans-serif; white-space: normal; margin-left: 4px; margin-top: 10px; margin-bottom: 9px; margin-right: 10px;  }
.S17 { min-height: 0px; color: rgb(0, 95, 206); margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S18 { min-height: 0px; white-space: normal; color: rgb(34, 139, 34); white-space: normal; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S19 { font-weight: bold; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S20 { margin-left: 3px; margin-top: 10px; margin-bottom: 4px; margin-right: 3px;  }
.S21 { font-style: italic; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S22 { min-height: 0px; font-style: italic; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S23 { min-height: 0px; font-weight: bold; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S24 { line-height: 20.576px; min-height: 20px; white-space: normal; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-size: 16px; font-weight: bold; white-space: normal; margin-left: 4px; margin-top: 3px; margin-bottom: 9px; margin-right: 10px;  }

.LineNodeBlock {margin: 10px 0 10px 0;}
.LineNodeBlock+.paragraphNode {margin-top: 10px;}
.lineNode {padding-left: 10px; background-color: #F7F7F7; border-left: 1px solid #E9E9E9; border-right: 1px solid #E9E9E9;}
.inlineWrapper:first-child .lineNode,.inlineWrapper.outputs+.inlineWrapper .lineNode {padding-top: 5px; border-top: 1px solid #E9E9E9;}
.inlineWrapper:last-child .lineNode,.inlineWrapper.outputs .lineNode {padding-bottom: 5px; border-bottom: 1px solid #E9E9E9;}
.lineNode .textBox {white-space: normal;}
.outputGroup {margin: 2px 0 2px 0; padding: 2px 2px 2px 4px;}
.outputRegion {}
.outputParagraph {color: rgba(64, 64, 64, 1); padding: 10px 0 6px 17px; background: white; overflow-x: hidden;}
.inlineWrapper:last-child .outputParagraph {border-bottom-left-radius: 4px; border-bottom-right-radius: 4px;}
.outputParagraph:empty {margin: 0;}
.inlineElement .symbolicElement {margin-top: 1px; margin-bottom: 1px;}
.embeddedOutputsSymbolicElement .MathEquation {margin-top: 4px; margin-bottom: 4px;}
.embeddedOutputsSymbolicElement .MathEquation.displaySymbolicElement {margin-left: 15px;}
.embeddedOutputsSymbolicElement .MathEquation.inlineSymbolicElement {margin-left: 5px;}
.embeddedOutputsSymbolicElement {overflow-x: auto; overflow-y: hidden;}
.embeddedOutputsSymbolicElement { overflow: initial !important;}
.embeddedOutputsTextElement,.embeddedOutputsVariableStringElement {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px; white-space: normal; word-wrap: initial; min-height: 18px; max-height: 250px; overflow: auto;}
.textElement {padding-top: 3px;}
.embeddedOutputsTextElement.inlineElement,.embeddedOutputsVariableStringElement.inlineElement {}
.inlineElement .textElement {}
.embeddedOutputsTextElement,.embeddedOutputsVariableStringElement { max-height: none !important; overflow: initial !important;}
.veSpecifier {}
.veContainer {}
.veSpecifierBox {height: 400px; width: 500px;}
.veSpecifier .veTable {padding-top: 3px; padding-bottom: 4px;}
.veSpecifierBox .veSpecifier .veContainer {position: relative; width: 100%; height: 370px;}
.veSpecifierBox .dijitDialogPaneContent {width: 97% !important; height: 88% !important;}
.veSpecifier .veTable .rowHeadersWrapper {padding-bottom: 0;}
.veSpecifier .veTable .scroller .variableEditorRenderers {padding-right: 3px; -webkit-user-select: none; -moz-user-select: none; -ms-user-select: none;}
.veVariableValueSummary {display: inline-block;}
.veSpecifier .veTable .topHeaderWrapper,.veSpecifier .veTable .bottomRowHeaderWrapper {visibility: hidden; z-index: 0;}
.veMetaSummary {font-style: italic;}
.veSpecifier .veTable .scroller {overflow: hidden;}
.veSpecifier .veTable:hover .scroller {overflow: auto;}
.veSpecifier .veVariableName,.veSpecifier .veDimensionFont {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px;}
.veSpecifier .veVariableName {padding-top: 2px;}
.veSpecifier .veDimensionFont {font-style: italic; color: #9A9A9A;}
.veSpecifier .scroller::-webkit-scrollbar-track {background-color: white;}
.veSpecifier .scroller::-webkit-scrollbar-corner {background-color: white;}
.veSpecifier .veTable .topRowHeaderWrapper {border: none; background-color: #F8F9FA;}
.veSpecifier .mw_type_ListBase.showCellBorders,.veSpecifier .veTable .topHeaderWrapper,.veSpecifier .veTable .bottomRowHeaderWrapper,.veSpecifier .veTable .verticalScrollSpacer,.veSpecifier .veTable .horizontalScrollSpacer {border: none;}
.veSpecifier .veTable .dataScrollerNode {border: 1px solid #BFBFBF;}
.veSpecifier .veTable .columnHeaderNode,.veSpecifier .veTable .rowHeaderNode,.veSpecifier .veTable .dataBody {font-family: Arial; font-size: 13px;}
.veSpecifier .veTable .columnHeaderNode,.veSpecifier .veTable .rowHeaderNode {color: #7F7F7F;}
.veSpecifier .veTable .dataBody {color: #000000;}
.veSpecifier .veTable .columnHeaderNode .cell .drag,.veSpecifier .veTable .columnHeaderNode .cell .header,.veSpecifier .veTable .topHeaderWrapper,.veSpecifier .veTable .bottomRowHeaderWrapper {background-color: #F8F9FA;}
.veSpecifier .veTable .columnHeaderNode .cell .dragBorder {border-right: 1px solid #F8F9FA;}
.veSpecifier .veTable .rowHeaderNode .cell {padding-top: 3px; text-align: center; border-bottom: 1px solid #F8F9FA;}
.veSpecifier .veTable .dataBody .cell .plainText {text-align: right;}
.veSpecifier .veTable .dataBody .row .cell {border-bottom: 1px solid #E9E9E9; border-right: 1px solid #E9E9E9;}
.embeddedOutputsVariableElement {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px; white-space: normal; word-wrap: break-word; min-height: 18px; max-height: 250px; overflow: auto;}
.variableElement {}
.embeddedOutputsVariableElement.inlineElement {}
.inlineElement .variableElement {}
.variableValue { width: 100% !important; }
.embeddedOutputsMatrixElement {min-height: 18px; box-sizing: border-box; font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px;}
.embeddedOutputsMatrixElement .matrixElement{position: relative;}
.matrixElement .variableValue {white-space: normal; display: inline-block; vertical-align: top; overflow: hidden;}
.embeddedOutputsMatrixElement.inlineElement {}
.embeddedOutputsMatrixElement.inlineElement .topHeaderWrapper {display: none;}
.embeddedOutputsMatrixElement.inlineElement .veTable .body {padding-top: 0 !important; max-height: 100px;}
.embeddedOutputsMatrixElement.inlineElement .veVariableName {padding-top: 0;}
.inlineElement .matrixElement {max-height: 300px;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer {white-space: nowrap; padding-top: 10px; margin-bottom: 3px;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer .horizontalEllipsis.hide,.embeddedOutputsMatrixElement .matrixElement .verticalEllipsis.hide {display: none;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer .horizontalEllipsis {margin-bottom: -3px;}
.matrixElement { max-height: none !important; }
.dijitTooltipDialog .dijitTooltipContainer .dijitTooltipContents .alertPlugin-alertMessage {min-width: 12px; max-width: 400px; max-height: 300px; overflow: auto;}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar {width: 11px; height: 11px;}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-track {background-color: rgba(0, 0, 0, 0);}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-corner {background-color: rgba(0, 0, 0, 0);}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-thumb {border-radius: 7px; background-color: rgba(0, 0, 0, 0.1); border: 2px solid rgba(0, 0, 0, 0); background-clip: padding-box;}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-thumb:hover {background-color: rgba(0, 0, 0, 0.2);}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-thumb {background-color: rgba(0, 0, 0, 0);}
.dijitTooltipDialog .alertPlugin-alertMessage:hover::-webkit-scrollbar-thumb {background-color: rgba(0, 0, 0, 0.1);}
.dijitTooltipDialog .alertPlugin-alertMessage:hover::-webkit-scrollbar-thumb:hover {background-color: rgba(0, 0, 0, 0.2);}
.alertPlugin-alertLine {position: absolute; display: initial; width: 40px; text-align: right; cursor: text;}
.alertPlugin-onTextLine {visibility: hidden;}
.alertPlugin-hasTooltip .alertPlugin-warningImg,.alertPlugin-hasTooltip .alertPlugin-errorImg {cursor: pointer;}
.alertPlugin-alertLine .alertPlugin-warningElement,.alertPlugin-alertLine .alertPlugin-errorElement {display: inline-block; margin-right: 4px;}
.alertPlugin-isStale {-webkit-filter: opacity(0.4) grayscale(80%); filter: opacity(0.4) grayscale(80%);}
.diagnosticMessage-wrapper {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px;}
.diagnosticMessage-wrapper.diagnosticMessage-warningType {color: rgb(255,100,0);}
.diagnosticMessage-wrapper.diagnosticMessage-warningType a {color: rgb(255,100,0); text-decoration: underline;}
.diagnosticMessage-wrapper.diagnosticMessage-errorType {color: rgb(230,0,0);}
.diagnosticMessage-wrapper.diagnosticMessage-errorType a {color: rgb(230,0,0); text-decoration: underline;}
.diagnosticMessage-wrapper .diagnosticMessage-messagePart {white-space: normal;}
.diagnosticMessage-wrapper .diagnosticMessage-stackPart {white-space: normal;}
.embeddedOutputsWarningElement{min-height: 18px; max-height: 250px; overflow: auto;}
.embeddedOutputsWarningElement.inlineElement {}
.embeddedOutputsErrorElement {min-height: 18px; max-height: 250px; overflow: auto;}
.embeddedOutputsErrorElement.inlineElement {}
</style></head><body><div class = "content"><div class = 'SectionBlock containment active'><h1 class = "S1"><span class = "S2">Extraction of context-specific models</span></h1><h2 class = "S3"><span class = "S2">Authors: Thomas Pfau, Systems Biochemistry Group, University of Luxembourg - Anne Richelle, Systems Biology and Cell Engineering, University of California San Diego</span></h2><h2 class = "S3"><span class = "S2">Reviewer(s): </span><span class = "S2">Almut Heinken, Molecular Systems Physiology Group, LCSB, University of Luxembourg</span></h2><h2 class = "S3"><span class = "S2">INTRODUCTION</span></h2><p class = "S4"><span class = "S2">Genome-scale reconstruction of metabolism (GEM) can illuminate the molecular basis of cell phenotypes exhibited by an organism. Since some enzymes are only active in specific cell types or environmental conditions, several algorithms have been developed to extract context-specific models that capture the metabolism of individual tissues or cell types. Therefore, a context-specific model is a subset of the GEM, in which inactive reactions are removed. Reaction removal is determined by the algorithm used, gene expression levels, presence of proteins or metabolites, experimental data availability, literature knowledge, and/or predefined metabolic functions of the cell type that need to be maintained in the extracted model. These decisions on methodology and data processing significantly influence the size, functionality and accuracy of constructed context-specific models. While there is no strong evidence that one model extraction method (MEM) universally gives the most physiologically accurate models, each method has different underlying assumptions that affect the resulting model. Therefore, selection of the MEM and the associated parameters should be done while considering the goals of the study and the available data [1].  	 </span></p><p class = "S4"><span class = "S2">Multiple algorithms have been suggested to automatically derive context specific networks from a generic reconstruction and a set of transcriptomic or proteomic data. The COBRA toolbox offers the following selection of extraction agorithms: </span></p><ul class = "S5"><li class = "S6"><span class = "S0">FASTCORE [2] - Define one set of core reactions that is guaranteed to be active in the extracted model and find the minimum number of reactions possible to support the core;</span></li><li class = "S6"><span class = "S0">GIMME [3] - Minimize usage of low-expression reactions while keeping the objective (e.g., biomass) above a certain value. Does not favor inclusion of reactions not related to the objective; </span></li><li class = "S6"><span class = "S0">iMAT [4] - Find the optimal trade-off between including high-expression reactions and removing low-expression reactions; </span></li><li class = "S6"><span class = "S0">INIT [5] - Find the optimal trade-off between including and removing reactions based on their given weights. If desired, accumulation of certain metabolites can be allowed or even forced; </span></li><li class = "S6"><span class = "S0">MBA [6] - Define high-confidence reactions to ensure activity in the extracted model. Medium confidence reactions are only kept when a certain parsimony trade-off is met. In random order, prune other reactions and remove them if not required to support high- or medium- confidence reactions; </span></li><li class = "S6"><span class = "S0">mCADRE [7] - Define a set of core reactions and prune all other reactions based on their expression, connectivity to core and confidence score. Remove reactions not necessary to support the core or defined functionalities. Core reactions are only removed if supported by a certain number of zero-expression reactions. </span></li></ul><p class = "S4"><span class = "S2">For convenience, there is a common interface to use these algorithms provided in the createTissueSpecificModel method. The method can be called as follow:</span></p><p class = "S4"><span class = "S7">tissueModel = createTissueSpecificModel(model, options)</span></p><h2 class = "S3"><span class = "S2">PROCEDURE</span></h2><p class = "S4"><span class = "S2">If necessary, initialise The COBRA Toolbox</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S10">global </span><span class = "S11">TUTORIAL_INIT_CB;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S10">if </span><span class = "S11">~isempty(TUTORIAL_INIT_CB) &amp;&amp; TUTORIAL_INIT_CB==1</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">    initCobraToolbox</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S12">end</span></p></div><div class = 'inlineWrapper outputs'><p class = "S9 lineNode"><span class = "S13">changeCobraSolver (</span><span class = "S14">'glpk'</span><span class = "S13">, </span><span class = "S14">'all'</span><span class = "S11">);</span></p><div class="outputParagraph"><div class="inlineElement embeddedOutputsTextElement scrollableOutput" uid="56606D35" data-testid="output_0" data-width="498" style="width: 528px;"><div class="textElement"> &gt; glpk is compatible and fully tested with MATLAB R2016b on your operating system.<br> &gt; Solver for LP problems has been set to glpk.<br> &gt; glpk is compatible and fully tested with MATLAB R2016b on your operating system.<br> &gt; Solver for MILP problems has been set to glpk.<br> &gt; Solver glpk not supported for problems of type MIQP. Currently used: gurobi <br> &gt; Solver glpk not supported for problems of type NLP. No solver set for this problemtype <br> &gt; Solver glpk not supported for problems of type QP. Currently used: gurobi </div></div></div></div></div><p class = "S16"><span class = "S2">Load the model that will be used for the extraction. For this tutorial, we have choosen to use E. coli core model as example. Please download the model from </span><a href = 'http://systemsbiology.ucsd.edu/sites/default/files/Attachments/Images/downloads/Ecoli_core/ecoli_core_model.mat'><span class = "S0">http://systemsbiology.ucsd.edu/sites/default/files/Attachments/Images/downloads/Ecoli_core/ecoli_core_model.mat</span></a><span class = "S2"> and save it in your preferred folder.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">modelFileName = </span><span class = "S14">'ecoli_core_model.mat'</span><span class = "S11">;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">modelDirectory = getDistributedModelFolder(modelFileName); </span><span class = "S18">%Look up the folder for the distributed Models.</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">modelFileName= [modelDirectory filesep modelFileName]; </span><span class = "S18">% Get the full path. Necessary to be sure, that the right model is loaded</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">model = readCbModel(modelFileName);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11"></span></p></div></div><p class = "S16"><span class = "S2">Load the expression data that will be used for the extraction. For this tutorial, we have choosen to use E. coli Microarray-based gene expression data (downloaded from http://systemsbiology.ucsd.edu/InSilicoOrganisms/Ecoli/EcoliExpression2) related to a anaerobic growth on glucose of a wild-type E. coli strain.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load(</span><span class = "S14">'dataEcoli'</span><span class = "S11">);</span></p></div></div><p class = "S16"><span class = "S2">Depending on the method that will be used for the extraction, different options need to be provided that could depend on a preprocessing step of the gene expression data. Since preprocessing is data dependent and the data used can involve multiple sources, we provide  preprocessed data for all the methods. The section "CRITICAL STEP" below briefly explains how the preprocessing has been done for this tutorial and provides some guidelines for the users on how to preprocess their data.</span></p><p class = "S4"><span class = "S2">Load the preprocessed data associated with the extraction method. Select one of the following sections to select the algorithm and data you want to use.</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">For the iMAT method, the available parameter options are (with all options marked with * being optional):</span></p><ul class = "S5"><li class = "S6"><span class = "S19">options.solver</span><span class = "S0"> : 'iMAT'</span></li><li class = "S6"><span class = "S19">options.expressionRxns</span><span class = "S0"> : gene expression data corresponding to model.rxns. Note : If no gene expression data are available for a reaction, set the value to -1</span></li><li class = "S6"><span class = "S19">options.threshold_lb</span><span class = "S0"> : lower bound of gene expression threshold, reactions with expression below this value are "non-expressed"</span></li><li class = "S6"><span class = "S19">options.threshold_ub</span><span class = "S0"> : upper bound of gene expression threshold, reactions with  expression above this value are "expressed"</span></li><li class = "S6"><span class = "S19">options.tol*</span><span class = "S0"> : minimum flux value for "expressed" reactions (default - 1e-8)</span></li><li class = "S6"><span class = "S19">options.core*</span><span class = "S0"> : list of reaction names that are associated with a high confidence (default - no core reactions)                           </span></li><li class = "S6"><span class = "S19">options.logfile*</span><span class = "S0"> : name of the file to save the MILP log (defaut - 'MILPlog')</span></li><li class = "S6"><span class = "S19">options.runtime*</span><span class = "S0"> : maximum solve time for the MILP (default - 7200s)</span></li></ul><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">options = </span><span class = "S14">'options_iMAT'</span><span class = "S11">;</span></p></div></div><p class = "S16"><span class = "S2">Now, load the preprocessed data </span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load([</span><span class = "S14">'options_methods' </span><span class = "S11">filesep options]);</span></p></div></div><p class = "S16"><span class = "S2">Call createTissueSpecificModel to extract your model</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper outputs'><p class = "S9 lineNode"><span class = "S11">iMAT_model = createTissueSpecificModel(model, options);</span></p><div class="outputParagraph"><div class="inlineElement embeddedOutputsTextElement scrollableOutput" uid="B3F95801" data-testid="output_1" data-width="498" style="width: 528px; max-height: 392px;"><div class="textElement">GLPK Integer Optimizer, v4.42<br>170 rows, 175 columns, 547 non-zeros<br>80 integer variables, all of which are binary<br>Preprocessing...<br>6 constraint coefficient(s) were reduced<br>134 rows, 125 columns, 445 non-zeros<br>57 integer variables, all of which are binary<br>Scaling...<br> A: min|aij| =&nbsp;&nbsp;7.090e-02&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;1.001e+03&nbsp;&nbsp;ratio =&nbsp;&nbsp;1.412e+04<br>GM: min|aij| =&nbsp;&nbsp;1.836e-01&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;5.448e+00&nbsp;&nbsp;ratio =&nbsp;&nbsp;2.968e+01<br>EQ: min|aij| =&nbsp;&nbsp;3.406e-02&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;1.000e+00&nbsp;&nbsp;ratio =&nbsp;&nbsp;2.936e+01<br>2N: min|aij| =&nbsp;&nbsp;3.125e-02&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;1.000e+00&nbsp;&nbsp;ratio =&nbsp;&nbsp;3.200e+01<br>Constructing initial basis...<br>Size of triangular part = 126<br>Solving LP relaxation...<br>GLPK Simplex Optimizer, v4.42<br>134 rows, 125 columns, 445 non-zeros<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0: obj =&nbsp;&nbsp;-8.732017982e+02&nbsp;&nbsp;infeas =&nbsp;&nbsp;8.531e+04 (8)<br>*&nbsp;&nbsp;&nbsp;&nbsp;47: obj =&nbsp;&nbsp; 4.019237671e+01&nbsp;&nbsp;infeas =&nbsp;&nbsp;8.235e-16 (5)<br>*&nbsp;&nbsp;&nbsp;&nbsp;82: obj =&nbsp;&nbsp; 5.894902997e+01&nbsp;&nbsp;infeas =&nbsp;&nbsp;4.996e-16 (5)<br>OPTIMAL SOLUTION FOUND<br>Integer optimization begins...<br>+&nbsp;&nbsp;&nbsp;&nbsp;82: mip =&nbsp;&nbsp;&nbsp;&nbsp; not found yet &lt;=&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+inf&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(1; 0)<br>+&nbsp;&nbsp; 502: &gt;&gt;&gt;&gt;&gt;&nbsp;&nbsp; 4.100000000e+01 &lt;=&nbsp;&nbsp; 4.800000000e+01&nbsp;&nbsp;17.1% (54; 13)<br>+&nbsp;&nbsp;2446: &gt;&gt;&gt;&gt;&gt;&nbsp;&nbsp; 4.300000000e+01 &lt;=&nbsp;&nbsp; 4.300000000e+01&nbsp;&nbsp; 0.0% (138; 169)<br>+&nbsp;&nbsp;2446: mip =&nbsp;&nbsp; 4.300000000e+01 &lt;=&nbsp;&nbsp;&nbsp;&nbsp; tree is empty&nbsp;&nbsp; 0.0% (0; 611)<br>INTEGER OPTIMAL SOLUTION FOUND</div></div></div></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">For GIMME method, the available parameter options are (with all options marked with * being optional):</span></p><ul class = "S5"><li class = "S6"><span class = "S19">options.solver</span><span class = "S0"> : 'GIMME'</span></li><li class = "S6"><span class = "S19">options.expressionRxns</span><span class = "S0"> : gene expression data corresponding to model.rxns. Note : If no gene expression data are available for a reaction, set the value to -1</span></li><li class = "S6"><span class = "S19">options.threshold</span><span class = "S0"> : gene expression threshold, reactions below this are minimized</span></li><li class = "S6"><span class = "S19">options.obj_frac*</span><span class = "S0"> : minimum fraction of the model objective function that need to be maintained in the extracted model (default - 0.9)</span></li></ul><p class = "S4"><span class = "S2">The code to create the tissue-specific model is as for iMAT.</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">options = </span><span class = "S14">'options_GIMME'</span><span class = "S11">;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load([</span><span class = "S14">'options_methods' </span><span class = "S11">filesep options]);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">GIMME_model = createTissueSpecificModel(model, options);</span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">For MBA  method, the available parameter options are (with all options marked with * being optional):</span></p><ul class = "S5"><li class = "S6"><span class = "S19">options.solver</span><span class = "S0"> : 'MBA'</span></li><li class = "S6"><span class = "S19">options.medium_set</span><span class = "S0"> : list of reaction names with medium confidence</span></li><li class = "S6"><span class = "S19">options.high_set</span><span class = "S0"> : list of reaction names with high confidence</span></li><li class = "S6"><span class = "S19">options.tol*</span><span class = "S0"> : minimum flux value for "expressed" reactions (default - 1e-8)</span></li></ul><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">options = </span><span class = "S14">'options_MBA'</span><span class = "S11">;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load([</span><span class = "S14">'options_methods' </span><span class = "S11">filesep options]);</span></p></div><div class = 'inlineWrapper outputs'><p class = "S9 lineNode"><span class = "S11">MBA_model = createTissueSpecificModel(model, options);</span></p><div class="outputParagraph"><div class="inlineElement embeddedOutputsTextElement" uid="3144FD22" data-testid="output_2" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n<br>0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="F35CA6BE" data-testid="output_3" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="AB2E7C76" data-testid="output_4" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="51FBE175" data-testid="output_5" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="95873107" data-testid="output_6" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="45608333" data-testid="output_7" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="0A1BECCD" data-testid="output_8" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="63231B03" data-testid="output_9" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="3772C5B3" data-testid="output_10" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="C2C4832B" data-testid="output_11" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="B3999C55" data-testid="output_12" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="DE18F800" data-testid="output_13" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="9A0FD1AB" data-testid="output_14" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="567EA8A1" data-testid="output_15" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="51DF8D0D" data-testid="output_16" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="CDFBA7FC" data-testid="output_17" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8EA06EA4" data-testid="output_18" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="F5CD6CCF" data-testid="output_19" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="073F1B65" data-testid="output_20" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="79E65578" data-testid="output_21" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="C3006878" data-testid="output_22" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="0A43EA23" data-testid="output_23" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="51702E97" data-testid="output_24" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6BF89FED" data-testid="output_25" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="48DD6AF1" data-testid="output_26" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="2FAC0F89" data-testid="output_27" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="63BCAAB9" data-testid="output_28" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="2D070506" data-testid="output_29" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="6F76A246" data-testid="output_30" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5E2D0E23" data-testid="output_31" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div></div></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">For FASTCORE  method, the available parameter options are (with all options marked with * being optional):</span></p><ul class = "S5"><li class = "S6"><span class = "S19">options.solver</span><span class = "S0"> : 'fastCore'</span></li><li class = "S6"><span class = "S19">options.core</span><span class = "S0"> : indices of reactions in  the model that are part of the core set of reactions</span></li><li class = "S6"><span class = "S19">options.epsilon*</span><span class = "S0"> : smallest flux value that is considered nonzero (default - 1e-4)</span></li><li class = "S6"><span class = "S19">options.printLevel*</span><span class = "S0"> : 0 = silent, 1 = summary, 2 = debug (default - 0)</span></li></ul><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">options = </span><span class = "S14">'options_fastCore'</span><span class = "S11">;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load([</span><span class = "S14">'options_methods' </span><span class = "S11">filesep options]);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">FastCore_model = createTissueSpecificModel(model, options);</span></p></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">For mCADRE method, the available parameter options are (with all options marked with * being optional):</span></p><ul class = "S5"><li class = "S6"><span class = "S19">options.solver</span><span class = "S0"> : 'mCADRE'</span></li><li class = "S6"><span class = "S19">options.ubiquityScore</span><span class = "S0"> : ubiquity scores, vector of the size of 'model.rxns' quantifying how often a gene is expressed accross samples</span></li><li class = "S6"><span class = "S19">options.confidenceScores</span><span class = "S0"> : literature-based evidence for generic model, vector of the size of 'model.rxns'</span></li><li class = "S6"><span class = "S19">options.protectedRxns*</span><span class = "S0"> : cell array with reactions names that are manually added to the core reaction set (default - no reactions)</span></li><li class = "S6"><span class = "S19">options.checkFunctionality*</span><span class = "S0"> : Boolean variable that determines if the model should be able to produce the metabolites associated with the '</span><span class = "S21">protectedRxns</span><span class = "S0">' (0: don't use functionality check (default value) - 1: include functionality check)</span></li><li class = "S6"><span class = "S19">options.eta*</span><span class = "S0"> : tradeoff between removing core and zero-expression reactions (default - 1/3)</span></li><li class = "S6"><span class = "S19">options.tol*</span><span class = "S0"> : minimum flux value for "expressed" reactions (default - 1e-8)</span></li></ul><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">options = </span><span class = "S14">'options_mCADRE'</span><span class = "S11">;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load([</span><span class = "S14">'options_methods' </span><span class = "S11">filesep options]);</span></p></div><div class = 'inlineWrapper outputs'><p class = "S9 lineNode"><span class = "S11">mCADRE_model = createTissueSpecificModel(model, options);</span></p><div class="outputParagraph"><div class="inlineElement embeddedOutputsTextElement" uid="96B34276" data-testid="output_32" data-width="498" style="width: 528px;"><div class="textElement">Processing inputs and ranking reactions...</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1630E284" data-testid="output_33" data-width="498" style="width: 528px;"><div class="textElement">Generic model passed precursor metabolites test</div></div><div class="inlineElement embeddedOutputsTextElement" uid="65F501B3" data-testid="output_34" data-width="498" style="width: 528px;"><div class="textElement">Pruning reactions...</div></div><div class="inlineElement embeddedOutputsTextElement" uid="65D4E41C" data-testid="output_35" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 1</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E5EB58A1" data-testid="output_36" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction O2t...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="71E096EE" data-testid="output_37" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="FCDAA849" data-testid="output_38" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="AB61856B" data-testid="output_39" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7B6065BB" data-testid="output_40" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 11 (0 core, 11 non-core); Num. remaining: 75</div></div><div class="inlineElement embeddedOutputsTextElement" uid="FB0CBE04" data-testid="output_41" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 2</div></div><div class="inlineElement embeddedOutputsTextElement" uid="22365A47" data-testid="output_42" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ACALDt...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="CA7852AD" data-testid="output_43" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="3E19E57D" data-testid="output_44" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="73F17421" data-testid="output_45" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="F993141F" data-testid="output_46" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 13 (0 core, 13 non-core); Num. remaining: 73</div></div><div class="inlineElement embeddedOutputsTextElement" uid="83B1684C" data-testid="output_47" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 3</div></div><div class="inlineElement embeddedOutputsTextElement" uid="9239D504" data-testid="output_48" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_co2(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="291B360D" data-testid="output_49" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9E7BD54A" data-testid="output_50" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8C0B3EE7" data-testid="output_51" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="82619526" data-testid="output_52" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 15 (0 core, 15 non-core); Num. remaining: 71</div></div><div class="inlineElement embeddedOutputsTextElement" uid="902BD74B" data-testid="output_53" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 4</div></div><div class="inlineElement embeddedOutputsTextElement" uid="0B9ED7F7" data-testid="output_54" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_h2o(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="61619B81" data-testid="output_55" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="AA843AF1" data-testid="output_56" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="41005202" data-testid="output_57" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A0CEED37" data-testid="output_58" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 17 (0 core, 17 non-core); Num. remaining: 69</div></div><div class="inlineElement embeddedOutputsTextElement" uid="AEFDD5FA" data-testid="output_59" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 5</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7491C571" data-testid="output_60" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_etoh(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="E77787A8" data-testid="output_61" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="758E7264" data-testid="output_62" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="C6F6F05F" data-testid="output_63" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="B234E7EB" data-testid="output_64" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 17 (0 core, 17 non-core); Num. remaining: 68</div></div><div class="inlineElement embeddedOutputsTextElement" uid="63A4DA40" data-testid="output_65" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 6</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C23D9ACE" data-testid="output_66" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_ac(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="54567FC2" data-testid="output_67" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="590B3601" data-testid="output_68" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8D956F9F" data-testid="output_69" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8B69160D" data-testid="output_70" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 21 (0 core, 21 non-core); Num. remaining: 64</div></div><div class="inlineElement embeddedOutputsTextElement" uid="83F39607" data-testid="output_71" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 7</div></div><div class="inlineElement embeddedOutputsTextElement" uid="97A9ADD3" data-testid="output_72" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_pyr(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="7BC1ACD5" data-testid="output_73" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="25A95F29" data-testid="output_74" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="2CA80E9E" data-testid="output_75" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="009279BB" data-testid="output_76" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 23 (0 core, 23 non-core); Num. remaining: 62</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E38D6545" data-testid="output_77" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 8</div></div><div class="inlineElement embeddedOutputsTextElement" uid="FC044FE1" data-testid="output_78" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_succ(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="240F8D12" data-testid="output_79" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5DAFA2B9" data-testid="output_80" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="091B1559" data-testid="output_81" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="82631CCB" data-testid="output_82" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 30 (0 core, 30 non-core); Num. remaining: 55</div></div><div class="inlineElement embeddedOutputsTextElement" uid="622D23E7" data-testid="output_83" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 9</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1F134AF7" data-testid="output_84" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_lac_D(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6EE63504" data-testid="output_85" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="32256FFC" data-testid="output_86" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="61039655" data-testid="output_87" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="2297305A" data-testid="output_88" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 52</div></div><div class="inlineElement embeddedOutputsTextElement" uid="2F2D4D43" data-testid="output_89" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 10</div></div><div class="inlineElement embeddedOutputsTextElement" uid="778550DE" data-testid="output_90" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_glu_L(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="07D71804" data-testid="output_91" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="5CED4758" data-testid="output_92" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="608DB5ED" data-testid="output_93" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="4759F23D" data-testid="output_94" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="A1D3C639" data-testid="output_95" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="B11000E6" data-testid="output_96" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="F6E8C331" data-testid="output_97" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="959B7503" data-testid="output_98" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="FD546716" data-testid="output_99" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7AB984EF" data-testid="output_100" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 51</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EB9CBE08" data-testid="output_101" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 11</div></div><div class="inlineElement embeddedOutputsTextElement" uid="40EB670E" data-testid="output_102" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_akg(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9A640235" data-testid="output_103" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="3AD2C569" data-testid="output_104" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="967B2B04" data-testid="output_105" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="25486A15" data-testid="output_106" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 50</div></div><div class="inlineElement embeddedOutputsTextElement" uid="3851BCCA" data-testid="output_107" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 12</div></div><div class="inlineElement embeddedOutputsTextElement" uid="6CF1522C" data-testid="output_108" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_glc(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="64643D33" data-testid="output_109" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="A24405B0" data-testid="output_110" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="28F4DCEA" data-testid="output_111" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="7C536A22" data-testid="output_112" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="D2574732" data-testid="output_113" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="0CEB52AB" data-testid="output_114" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="FB7CFB5C" data-testid="output_115" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="7BC6828F" data-testid="output_116" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="5D8882FD" data-testid="output_117" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="52F99241" data-testid="output_118" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 49</div></div><div class="inlineElement embeddedOutputsTextElement" uid="131B9D2C" data-testid="output_119" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 13</div></div><div class="inlineElement embeddedOutputsTextElement" uid="0259EF88" data-testid="output_120" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_pi(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="BCA8A71B" data-testid="output_121" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="BA4BF6D0" data-testid="output_122" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="5833F344" data-testid="output_123" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EC94B98F" data-testid="output_124" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 48</div></div><div class="inlineElement embeddedOutputsTextElement" uid="CC123DDC" data-testid="output_125" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 14</div></div><div class="inlineElement embeddedOutputsTextElement" uid="126A7353" data-testid="output_126" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_nh4(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="07DFB624" data-testid="output_127" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="800AB70D" data-testid="output_128" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="D9AE5F6F" data-testid="output_129" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="A7EBBED5" data-testid="output_130" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="EBC1C52B" data-testid="output_131" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="604ADC9F" data-testid="output_132" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="0EECA186" data-testid="output_133" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6F4FF07C" data-testid="output_134" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8A725732" data-testid="output_135" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="DE9E11EC" data-testid="output_136" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 47</div></div><div class="inlineElement embeddedOutputsTextElement" uid="12784239" data-testid="output_137" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 15</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C23D75D0" data-testid="output_138" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_for(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="56687CC7" data-testid="output_139" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="017E93E8" data-testid="output_140" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="8AEE4922" data-testid="output_141" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="BCD54E87" data-testid="output_142" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="B7536CDE" data-testid="output_143" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="7B3719C5" data-testid="output_144" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="86EAAF9D" data-testid="output_145" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="A135338A" data-testid="output_146" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="61F4A45A" data-testid="output_147" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="587A8B51" data-testid="output_148" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 46</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C790CEA4" data-testid="output_149" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 16</div></div><div class="inlineElement embeddedOutputsTextElement" uid="AED65F44" data-testid="output_150" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction EX_h(e)...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="043B1CBD" data-testid="output_151" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="795B0F0F" data-testid="output_152" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="C81A91D4" data-testid="output_153" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="E2859E70" data-testid="output_154" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6E86241D" data-testid="output_155" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="D88DA844" data-testid="output_156" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5E62E7ED" data-testid="output_157" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="7F45E48F" data-testid="output_158" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="667D080E" data-testid="output_159" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E1C02806" data-testid="output_160" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 45</div></div><div class="inlineElement embeddedOutputsTextElement" uid="2BDA7FAB" data-testid="output_161" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 17</div></div><div class="inlineElement embeddedOutputsTextElement" uid="D40D35A5" data-testid="output_162" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ETOHt2r...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9B1C50CA" data-testid="output_163" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8C60188F" data-testid="output_164" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="1971CC84" data-testid="output_165" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="2B62F35B" data-testid="output_166" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="85216B93" data-testid="output_167" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="1963520D" data-testid="output_168" data-width="498" style="width: 528px;"><div class="textElement">0 = solution.stat<br>110 = solution.origStat</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="AF06234E" data-testid="output_169" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: LP solution may not be optimal</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="4805A9CF" data-testid="output_170" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="556B61DE" data-testid="output_171" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="50B9ADCE" data-testid="output_172" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 33 (0 core, 33 non-core); Num. remaining: 44</div></div><div class="inlineElement embeddedOutputsTextElement" uid="B42F871D" data-testid="output_173" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 18</div></div><div class="inlineElement embeddedOutputsTextElement" uid="438786B5" data-testid="output_174" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction SUCCt3...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="4119FDBD" data-testid="output_175" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="A0C8D018" data-testid="output_176" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="E30241BA" data-testid="output_177" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C9F7B288" data-testid="output_178" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 35 (0 core, 35 non-core); Num. remaining: 42</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A05D6268" data-testid="output_179" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 19</div></div><div class="inlineElement embeddedOutputsTextElement" uid="57DB5988" data-testid="output_180" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ATPM...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="1A835230" data-testid="output_181" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="C234F9D5" data-testid="output_182" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="D1C039EC" data-testid="output_183" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="CE7D4E1A" data-testid="output_184" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="BF3640C9" data-testid="output_185" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 36 (0 core, 36 non-core); Num. remaining: 41</div></div><div class="inlineElement embeddedOutputsTextElement" uid="5002DA5A" data-testid="output_186" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 20</div></div><div class="inlineElement embeddedOutputsTextElement" uid="0903C1D1" data-testid="output_187" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction Biomass_Ecoli_core_w_GAM...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="3D9895EA" data-testid="output_188" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="50469817" data-testid="output_189" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="CD901974" data-testid="output_190" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8C92D9ED" data-testid="output_191" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 36 (0 core, 36 non-core); Num. remaining: 40</div></div><div class="inlineElement embeddedOutputsTextElement" uid="189CE57F" data-testid="output_192" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 21</div></div><div class="inlineElement embeddedOutputsTextElement" uid="DCF1A505" data-testid="output_193" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction RPE...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="4292A16F" data-testid="output_194" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="DCAB6AF0" data-testid="output_195" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="DC36B849" data-testid="output_196" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="098AD5FE" data-testid="output_197" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="341C9367" data-testid="output_198" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 37 (0 core, 37 non-core); Num. remaining: 39</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E898BDB8" data-testid="output_199" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 22</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C542AB58" data-testid="output_200" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction GLNS...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="644480A6" data-testid="output_201" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="DAA14BE9" data-testid="output_202" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="F4AF223C" data-testid="output_203" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="2E968A64" data-testid="output_204" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 37 (0 core, 37 non-core); Num. remaining: 38</div></div><div class="inlineElement embeddedOutputsTextElement" uid="DE3A0632" data-testid="output_205" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 23</div></div><div class="inlineElement embeddedOutputsTextElement" uid="5E09BD66" data-testid="output_206" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction SUCDi...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="514D5C52" data-testid="output_207" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="72E94554" data-testid="output_208" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="4524DE7F" data-testid="output_209" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="04207DA9" data-testid="output_210" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 39 (0 core, 39 non-core); Num. remaining: 36</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7BB4982E" data-testid="output_211" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 24</div></div><div class="inlineElement embeddedOutputsTextElement" uid="CA15ABF7" data-testid="output_212" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction FORti...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5CD13885" data-testid="output_213" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="E675472A" data-testid="output_214" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="3B448EF2" data-testid="output_215" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="612627EF" data-testid="output_216" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 39 (0 core, 39 non-core); Num. remaining: 35</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A91B891C" data-testid="output_217" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 25</div></div><div class="inlineElement embeddedOutputsTextElement" uid="6338B92E" data-testid="output_218" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction FORt2...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="E3B47ABC" data-testid="output_219" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="9869AA92" data-testid="output_220" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="E8007EA8" data-testid="output_221" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="E2BDBA27" data-testid="output_222" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="9D43DBB2" data-testid="output_223" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 40 (0 core, 40 non-core); Num. remaining: 34</div></div><div class="inlineElement embeddedOutputsTextElement" uid="6C404D27" data-testid="output_224" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 26</div></div><div class="inlineElement embeddedOutputsTextElement" uid="9B5C1D80" data-testid="output_225" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction NH4t...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="B190F316" data-testid="output_226" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="8AFB8612" data-testid="output_227" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="2860CF06" data-testid="output_228" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="261DC24D" data-testid="output_229" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 40 (0 core, 40 non-core); Num. remaining: 33</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8BE109C7" data-testid="output_230" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 27</div></div><div class="inlineElement embeddedOutputsTextElement" uid="085F1B3B" data-testid="output_231" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PFL...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="D119A378" data-testid="output_232" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="51962E1B" data-testid="output_233" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="CE410CB4" data-testid="output_234" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="69CCAAFA" data-testid="output_235" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 40 (0 core, 40 non-core); Num. remaining: 32</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EBC664DD" data-testid="output_236" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 28</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E30FA250" data-testid="output_237" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction GLCpts...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="C14F2FCB" data-testid="output_238" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="2D71FD46" data-testid="output_239" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="5128D555" data-testid="output_240" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1D836358" data-testid="output_241" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 40 (0 core, 40 non-core); Num. remaining: 31</div></div><div class="inlineElement embeddedOutputsTextElement" uid="66D55AA7" data-testid="output_242" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 29</div></div><div class="inlineElement embeddedOutputsTextElement" uid="56B675C3" data-testid="output_243" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ACONTa...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="EF4CA764" data-testid="output_244" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="C863362A" data-testid="output_245" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="96189549" data-testid="output_246" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="5B8ED1BC" data-testid="output_247" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 40 (0 core, 40 non-core); Num. remaining: 30</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A5FA13C9" data-testid="output_248" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 30</div></div><div class="inlineElement embeddedOutputsTextElement" uid="916884DB" data-testid="output_249" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ACONTb...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="FBB55A77" data-testid="output_250" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="BDF4B715" data-testid="output_251" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="06F0BB4B" data-testid="output_252" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="AE0807F8" data-testid="output_253" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 40 (0 core, 40 non-core); Num. remaining: 29</div></div><div class="inlineElement embeddedOutputsTextElement" uid="F34785D0" data-testid="output_254" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 31</div></div><div class="inlineElement embeddedOutputsTextElement" uid="D79AECF8" data-testid="output_255" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction GLUN...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5568A88C" data-testid="output_256" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="62814AFA" data-testid="output_257" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="1C44F1C7" data-testid="output_258" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="E25F7ABA" data-testid="output_259" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="D3BC1508" data-testid="output_260" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 41 (0 core, 41 non-core); Num. remaining: 28</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EEF07B41" data-testid="output_261" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 32</div></div><div class="inlineElement embeddedOutputsTextElement" uid="0A537B64" data-testid="output_262" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ME2...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="C51E16C5" data-testid="output_263" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="5FF9118F" data-testid="output_264" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="8ED42081" data-testid="output_265" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8FB85592" data-testid="output_266" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8C35BAE9" data-testid="output_267" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 42 (0 core, 42 non-core); Num. remaining: 27</div></div><div class="inlineElement embeddedOutputsTextElement" uid="597DBBBE" data-testid="output_268" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 33</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EFB62F35" data-testid="output_269" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction GLUt2r...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="DEA947CB" data-testid="output_270" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9BE00CEF" data-testid="output_271" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="08092B50" data-testid="output_272" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="24DD0592" data-testid="output_273" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 42 (0 core, 42 non-core); Num. remaining: 26</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7EADE1D7" data-testid="output_274" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 34</div></div><div class="inlineElement embeddedOutputsTextElement" uid="DF3D6D80" data-testid="output_275" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PPCK...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="72C42148" data-testid="output_276" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="7B008151" data-testid="output_277" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="491E6C19" data-testid="output_278" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="21570D2A" data-testid="output_279" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="D7AC29C6" data-testid="output_280" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 43 (0 core, 43 non-core); Num. remaining: 25</div></div><div class="inlineElement embeddedOutputsTextElement" uid="FD430647" data-testid="output_281" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 35</div></div><div class="inlineElement embeddedOutputsTextElement" uid="89358DBC" data-testid="output_282" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PPS...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="626D3B73" data-testid="output_283" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="B3E7A024" data-testid="output_284" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="DBDED53A" data-testid="output_285" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7EB38926" data-testid="output_286" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 43 (0 core, 43 non-core); Num. remaining: 24</div></div><div class="inlineElement embeddedOutputsTextElement" uid="3EFBA8F0" data-testid="output_287" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 36</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EDCF6ED0" data-testid="output_288" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction RPI...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="A2812CAB" data-testid="output_289" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6632F444" data-testid="output_290" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="7518B4F4" data-testid="output_291" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="F4F2D409" data-testid="output_292" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 43 (0 core, 43 non-core); Num. remaining: 23</div></div><div class="inlineElement embeddedOutputsTextElement" uid="46491293" data-testid="output_293" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 37</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C92CF762" data-testid="output_294" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction TKT1...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="EC505BB4" data-testid="output_295" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="4B1E58C0" data-testid="output_296" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="C577EF0B" data-testid="output_297" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="2236EE61" data-testid="output_298" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 43 (0 core, 43 non-core); Num. remaining: 22</div></div><div class="inlineElement embeddedOutputsTextElement" uid="77B274C7" data-testid="output_299" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 38</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A8ECF535" data-testid="output_300" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction TKT2...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="CCF5EE2D" data-testid="output_301" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9581AA2C" data-testid="output_302" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="1247FA90" data-testid="output_303" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A1E0BF76" data-testid="output_304" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 43 (0 core, 43 non-core); Num. remaining: 21</div></div><div class="inlineElement embeddedOutputsTextElement" uid="21BD858F" data-testid="output_305" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 39</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C2AC7CF0" data-testid="output_306" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction AKGt2r...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5863B1F0" data-testid="output_307" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="41E991EC" data-testid="output_308" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="D94818BC" data-testid="output_309" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="3CDA3B51" data-testid="output_310" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 43 (0 core, 43 non-core); Num. remaining: 20</div></div><div class="inlineElement embeddedOutputsTextElement" uid="EAF1F0B3" data-testid="output_311" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 40</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1583DEB1" data-testid="output_312" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ME1...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="C8F0D461" data-testid="output_313" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="3B91CC22" data-testid="output_314" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="62F7A360" data-testid="output_315" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="3DDEC168" data-testid="output_316" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 45 (0 core, 45 non-core); Num. remaining: 18</div></div><div class="inlineElement embeddedOutputsTextElement" uid="B4C2D32A" data-testid="output_317" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 41</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8ED017B6" data-testid="output_318" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction FBP...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="5A6CB55B" data-testid="output_319" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="E2DCD5A5" data-testid="output_320" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="CA9B79D3" data-testid="output_321" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="DABB7BDE" data-testid="output_322" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="03DB39F2" data-testid="output_323" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 46 (0 core, 46 non-core); Num. remaining: 17</div></div><div class="inlineElement embeddedOutputsTextElement" uid="AE561148" data-testid="output_324" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 42</div></div><div class="inlineElement embeddedOutputsTextElement" uid="080AAFBE" data-testid="output_325" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PGL...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="B21D1B9D" data-testid="output_326" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9F3B70E5" data-testid="output_327" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="F785CEC2" data-testid="output_328" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E0ECB963" data-testid="output_329" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 46 (0 core, 46 non-core); Num. remaining: 16</div></div><div class="inlineElement embeddedOutputsTextElement" uid="BDF1F5F8" data-testid="output_330" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 43</div></div><div class="inlineElement embeddedOutputsTextElement" uid="115AF2C1" data-testid="output_331" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PIt2r...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="0F4010AD" data-testid="output_332" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="449EF666" data-testid="output_333" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="235FC2F9" data-testid="output_334" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1F24DF3B" data-testid="output_335" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 46 (0 core, 46 non-core); Num. remaining: 15</div></div><div class="inlineElement embeddedOutputsTextElement" uid="F1E6E531" data-testid="output_336" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 44</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C3FB962C" data-testid="output_337" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PGM...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="93C93FB0" data-testid="output_338" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="1A6A7CFC" data-testid="output_339" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="410C2E9B" data-testid="output_340" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C7D2B745" data-testid="output_341" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 46 (0 core, 46 non-core); Num. remaining: 14</div></div><div class="inlineElement embeddedOutputsTextElement" uid="22028D00" data-testid="output_342" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 45</div></div><div class="inlineElement embeddedOutputsTextElement" uid="AAD02421" data-testid="output_343" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction NADTRHD...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9CE85B15" data-testid="output_344" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="B754CF48" data-testid="output_345" data-width="498" style="width: 528px;"><div class="textElement">fastcc.m: The input model is entirely flux consistent.\n</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="FEA18E45" data-testid="output_346" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="A99B08FB" data-testid="output_347" data-width="498" style="width: 528px;"><div class="textElement">Removed non-core inactive reactions</div></div><div class="inlineElement embeddedOutputsTextElement" uid="CB104902" data-testid="output_348" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 13</div></div><div class="inlineElement embeddedOutputsTextElement" uid="B6BCD94C" data-testid="output_349" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 46</div></div><div class="inlineElement embeddedOutputsTextElement" uid="98396163" data-testid="output_350" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction THD2...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="D2C612BC" data-testid="output_351" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="97F0B510" data-testid="output_352" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="65EB44CF" data-testid="output_353" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1FE21C68" data-testid="output_354" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 12</div></div><div class="inlineElement embeddedOutputsTextElement" uid="6C316C7A" data-testid="output_355" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 47</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8C7CE147" data-testid="output_356" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ATPS4r...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="10397106" data-testid="output_357" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="489D75AB" data-testid="output_358" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="2A75F4AC" data-testid="output_359" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="29897048" data-testid="output_360" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 11</div></div><div class="inlineElement embeddedOutputsTextElement" uid="65B93E7F" data-testid="output_361" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 48</div></div><div class="inlineElement embeddedOutputsTextElement" uid="89B06EFA" data-testid="output_362" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction CS...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="47439C44" data-testid="output_363" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="75DC6823" data-testid="output_364" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="AEE58D3A" data-testid="output_365" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="9B6400DC" data-testid="output_366" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 10</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E7C20FE0" data-testid="output_367" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 49</div></div><div class="inlineElement embeddedOutputsTextElement" uid="25A1EAE4" data-testid="output_368" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PDH...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="09A1046F" data-testid="output_369" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="439905BB" data-testid="output_370" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="40012774" data-testid="output_371" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="243B253C" data-testid="output_372" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 9</div></div><div class="inlineElement embeddedOutputsTextElement" uid="58BED129" data-testid="output_373" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 50</div></div><div class="inlineElement embeddedOutputsTextElement" uid="503A1864" data-testid="output_374" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction G6PDH2r...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="1D9E1FB4" data-testid="output_375" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="342814F1" data-testid="output_376" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="FF139E34" data-testid="output_377" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="8E838E12" data-testid="output_378" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 8</div></div><div class="inlineElement embeddedOutputsTextElement" uid="348DF53A" data-testid="output_379" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 51</div></div><div class="inlineElement embeddedOutputsTextElement" uid="60483ECE" data-testid="output_380" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PFK...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="2F3E546E" data-testid="output_381" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="1FCACF43" data-testid="output_382" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="3A65DCB9" data-testid="output_383" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="E4372B93" data-testid="output_384" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 7</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A1E6962A" data-testid="output_385" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 52</div></div><div class="inlineElement embeddedOutputsTextElement" uid="7FA7CF2E" data-testid="output_386" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PPC...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="A5B8A202" data-testid="output_387" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="F9EC1D92" data-testid="output_388" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="8331D910" data-testid="output_389" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C232E8D9" data-testid="output_390" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 6</div></div><div class="inlineElement embeddedOutputsTextElement" uid="709FCB22" data-testid="output_391" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 53</div></div><div class="inlineElement embeddedOutputsTextElement" uid="3C75F38B" data-testid="output_392" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PGI...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="939DF54B" data-testid="output_393" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="E9F653B2" data-testid="output_394" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="43EA850D" data-testid="output_395" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="855276E7" data-testid="output_396" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 5</div></div><div class="inlineElement embeddedOutputsTextElement" uid="76C9515D" data-testid="output_397" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 54</div></div><div class="inlineElement embeddedOutputsTextElement" uid="35043BFF" data-testid="output_398" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction PYK...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="A7002B01" data-testid="output_399" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6A206B58" data-testid="output_400" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="13350B82" data-testid="output_401" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="1CCE44A3" data-testid="output_402" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 4</div></div><div class="inlineElement embeddedOutputsTextElement" uid="6CD8C718" data-testid="output_403" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 55</div></div><div class="inlineElement embeddedOutputsTextElement" uid="66BA79CE" data-testid="output_404" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction ICDHyr...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="9243ED3C" data-testid="output_405" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="7701B619" data-testid="output_406" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="5CDE21AE" data-testid="output_407" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="4127F5F4" data-testid="output_408" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 3</div></div><div class="inlineElement embeddedOutputsTextElement" uid="5619993D" data-testid="output_409" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 56</div></div><div class="inlineElement embeddedOutputsTextElement" uid="A701D904" data-testid="output_410" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction TALA...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="4A6DBAE9" data-testid="output_411" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="49F9479F" data-testid="output_412" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="9284D4D3" data-testid="output_413" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="89497E4A" data-testid="output_414" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 2</div></div><div class="inlineElement embeddedOutputsTextElement" uid="CC67BC0C" data-testid="output_415" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 57</div></div><div class="inlineElement embeddedOutputsTextElement" uid="F08DD8BD" data-testid="output_416" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction FBA...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="B5C1BA65" data-testid="output_417" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="D112F512" data-testid="output_418" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="73336D55" data-testid="output_419" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="13865D05" data-testid="output_420" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 1</div></div><div class="inlineElement embeddedOutputsTextElement" uid="6CA4ADEE" data-testid="output_421" data-width="498" style="width: 528px;"><div class="textElement">Reaction no. 58</div></div><div class="inlineElement embeddedOutputsTextElement" uid="C9D559B1" data-testid="output_422" data-width="498" style="width: 528px;"><div class="textElement">Attempting to remove reaction TPI...</div></div><div class="inlineElement embeddedOutputsWarningElement" uid="442107C2" data-testid="output_423" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsWarningElement" uid="6D4289D1" data-testid="output_424" data-width="498" style="width: 528px;"><div class="diagnosticMessage-wrapper diagnosticMessage-warningType"><div class="diagnosticMessage-messagePart">Warning: No metabolites defined to check the model function</div><div class="diagnosticMessage-stackPart"></div></div></div><div class="inlineElement embeddedOutputsTextElement" uid="9074FE6B" data-testid="output_425" data-width="498" style="width: 528px;"><div class="textElement">No reactions removed</div></div><div class="inlineElement embeddedOutputsTextElement" uid="B67A3376" data-testid="output_426" data-width="498" style="width: 528px;"><div class="textElement">Num. removed: 47 (0 core, 47 non-core); Num. remaining: 0</div></div></div></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">For INIT method, the available parameter options are (with all options marked with * being optional):</span></p><ul class = "S5"><li class = "S6"><span class = "S19">options.solver</span><span class = "S0"> : 'INIT'</span></li><li class = "S6"><span class = "S19">options.weights</span><span class = "S0"> : column with positive (high expression) and negative (low expression) weights for each reaction</span></li><li class = "S6"><span class = "S19">options.tol*</span><span class = "S0"> : minimum flux value for "expressed" reactions (default - 1e-8)</span></li><li class = "S6"><span class = "S19">options.logfile*</span><span class = "S0"> : name of the file to save the MILP log (defaut - 'MILPlog')</span></li><li class = "S6"><span class = "S19">options.runtime*</span><span class = "S0"> : maximum solve time for the MILP (default - 7200s)</span></li></ul><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">options = </span><span class = "S14">'options_INIT'</span><span class = "S11">;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load([</span><span class = "S14">'options_methods' </span><span class = "S11">filesep options]);</span></p></div><div class = 'inlineWrapper outputs'><p class = "S9 lineNode"><span class = "S11">INIT_model = createTissueSpecificModel(model, options);</span></p><div class="outputParagraph"><div class="inlineElement embeddedOutputsTextElement scrollableOutput" uid="69EA8BC7" data-testid="output_427" data-width="498" style="width: 528px; max-height: 420px;"><div class="textElement">GLPK Integer Optimizer, v4.42<br>198 rows, 181 columns, 590 non-zeros<br>86 integer variables, all of which are binary<br>Preprocessing...<br>6 constraint coefficient(s) were reduced<br>144 rows, 136 columns, 471 non-zeros<br>69 integer variables, all of which are binary<br>Scaling...<br> A: min|aij| =&nbsp;&nbsp;7.090e-02&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;1.001e+03&nbsp;&nbsp;ratio =&nbsp;&nbsp;1.412e+04<br>GM: min|aij| =&nbsp;&nbsp;1.836e-01&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;5.448e+00&nbsp;&nbsp;ratio =&nbsp;&nbsp;2.968e+01<br>EQ: min|aij| =&nbsp;&nbsp;3.406e-02&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;1.000e+00&nbsp;&nbsp;ratio =&nbsp;&nbsp;2.936e+01<br>2N: min|aij| =&nbsp;&nbsp;3.125e-02&nbsp;&nbsp;max|aij| =&nbsp;&nbsp;1.250e+00&nbsp;&nbsp;ratio =&nbsp;&nbsp;4.000e+01<br>Constructing initial basis...<br>Size of triangular part = 134<br>Solving LP relaxation...<br>GLPK Simplex Optimizer, v4.42<br>144 rows, 136 columns, 471 non-zeros<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0: obj =&nbsp;&nbsp; 5.648220556e+02&nbsp;&nbsp;infeas =&nbsp;&nbsp;1.019e+05 (10)<br>*&nbsp;&nbsp;&nbsp;&nbsp;60: obj =&nbsp;&nbsp; 5.614599110e+01&nbsp;&nbsp;infeas =&nbsp;&nbsp;1.029e-14 (5)<br>*&nbsp;&nbsp; 100: obj =&nbsp;&nbsp; 7.458402071e+01&nbsp;&nbsp;infeas =&nbsp;&nbsp;5.551e-16 (5)<br>OPTIMAL SOLUTION FOUND<br>Integer optimization begins...<br>+&nbsp;&nbsp; 100: mip =&nbsp;&nbsp;&nbsp;&nbsp; not found yet &lt;=&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+inf&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(1; 0)<br>+&nbsp;&nbsp; 765: &gt;&gt;&gt;&gt;&gt;&nbsp;&nbsp; 4.926985032e+01 &lt;=&nbsp;&nbsp; 6.489465815e+01&nbsp;&nbsp;31.7% (88; 17)<br>+&nbsp;&nbsp;3373: &gt;&gt;&gt;&gt;&gt;&nbsp;&nbsp; 4.987940265e+01 &lt;=&nbsp;&nbsp; 5.581258238e+01&nbsp;&nbsp;11.9% (217; 297)<br>+&nbsp;&nbsp;3780: &gt;&gt;&gt;&gt;&gt;&nbsp;&nbsp; 5.106322395e+01 &lt;=&nbsp;&nbsp; 5.531164139e+01&nbsp;&nbsp; 8.3% (256; 353)<br>+&nbsp;&nbsp;3940: &gt;&gt;&gt;&gt;&gt;&nbsp;&nbsp; 5.331563786e+01 &lt;=&nbsp;&nbsp; 5.493999008e+01&nbsp;&nbsp; 3.0% (238; 428)<br>+&nbsp;&nbsp;4001: mip =&nbsp;&nbsp; 5.331563786e+01 &lt;=&nbsp;&nbsp;&nbsp;&nbsp; tree is empty&nbsp;&nbsp; 0.0% (0; 1257)<br>INTEGER OPTIMAL SOLUTION FOUND</div></div></div></div></div></div><p class = "S0"></p><div class = 'SectionBlock containment'><p class = "S4"><span class = "S2">Note that additional options are available when extracting your model (</span><span class = "S7">funcModel</span><span class = "S2"> and </span><span class = "S7">exRxnRemove</span><span class = "S2">). </span><span class = "S7">funcModel</span><span class = "S2"> allows to define whether the extracted model will be/ or not flux consistent (i.e., an extracted model that contains only reactions that can carry fluxes). </span><span class = "S7">exRxnRemove</span><span class = "S22"> </span><span class = "S2">allows to predefine a list of reactions that will be automatically removed in the extracted model. Example, if you want to extract a consistent model but you do not have any predefined list of reactions to remove, your call will be:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">funcModel=1;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">exRxnRemove={};</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">tissueModel = createTissueSpecificModel(model, options,funcModel,exRxnRemove);</span></p></div></div><h2 class = "S3"><span class = "S2">CRITICAL STEPS</span></h2><p class = "S4"><span class = "S23">Gene Expression Preprocessing</span></p><p class = "S4"><span class = "S2">When integrating transcriptomic data, the selection of options related to each method is critical and algorithmic performance often strongly depends on these choices. In GIMME [3], a fixed threshold value was used to distinguish between unexpressed and expressed reactions (threshold = 12 of log2 expression). For iMAT [4], the authors used a data dependent cutoff of half a standard deviation above the mean for active reactions, and half a standard deviation below for inactive reactions. In contrast to the selection by automated processing of expression data, the authors of FASTCORE [2] and MBA [5] used manually assembled sets of high confidence reactions along with data derived information as input. In INIT [5], the selection is based on the data in the Human Protein Atlas [8] using the indication from the proteomics data. For mCADRE [7], proprietary software was used to obtain presence/absence calls. </span></p><p class = "S4"><span class = "S2">In the context of this tutorial, the options have been defined following the original paper for iMAT and GIMME and arbitrary preprocessing as used in [1] has been done for the four other methods.</span></p><p class = "S4"><span class = "S2">The quality of the results also strongly depends on data preprocessing, and the COBRA toolbox does not provide an automatic preprocessing pipeline to derive expression values from raw measurements as multipe methods can be used. We therefore strongly suggest, that all preprocessing and discretization is performed prior to the call to </span><span class = "S7">createTissueSpecificModel</span><span class = "S2">. </span></p><p class = "S4"><span class = "S23">Gene to Reaction Mapping</span></p><p class = "S4"><span class = "S2">As with Preprocessing, we intentionally left the gene to reaction mapping separate from the tissue specific model extraction. The most common approach to obtain reaction expression values from geen expression values is to evaluate the Gene-Protein-Reaction rules associated with each reaction as follows:</span></p><p class = "S4"><span class = "S2">Replace 'and' bin 'min' and 'or' by 'max'. E.g. for a reaction R1 with GPR 'A and (B or C)' with expression of Genes A = 2, B = 7 and C = 0, 'B or C' would evaluate to 7 and the whole would evaluate to 2.</span></p><p class = "S4"><span class = "S2">This type of mapping</span><span class = "S2"> can be applied to</span><span class = "S2"> iMAT a</span><span class = "S2">nd </span><span class = "S2">GIMME, and is provided as a function in the toolbox:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">model = getDistributedModel(</span><span class = "S14">'ecoli_core_model.mat'</span><span class = "S11">)</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">load(</span><span class = "S14">'dataEcoli'</span><span class = "S11">);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">[expressionRxns parsedGPR] = mapExpressionToReactions(model, expression);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11"></span></p></div></div><p class = "S16"><span class = "S2">INIT and mCADRE use similar approaches, but require the expression to be protein expression, or ubiquityscores respectively, i.e. the function can be used for them, but the inputs need to be adjusted according to the required data for INIT and mCADRE. </span></p><p class = "S4"><span class = "S2">FastCore and MBA again, rely on clearly defined core sets of reactions, the quality of which strongly influences the resulting models.</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S24"><span class = "S2">TIMING</span></h2><p class = "S4"><span class = "S2">TIMING: 15 minutes to hours (computation) - days (interpretation)</span></p></div><p class = "S0"></p><div class = 'SectionBlock containment'><h2 class = "S24"><span class = "S2">ANTICIPATED RESULTS</span></h2><p class = "S4"><span class = "S2">The size of extracted models varies depending of the MEM used</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">expected_results = {</span><span class = "S14">'ecoli_core_model'</span><span class = "S13">,95,72;</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                    </span><span class = "S14">'FastCore_model'</span><span class = "S13">,45,48;</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                    </span><span class = "S14">'GIMME_model'</span><span class = "S13">,74,69;</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                    </span><span class = "S14">'INIT_model'</span><span class = "S13">,25,32;</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                    </span><span class = "S14">'MBA_model'</span><span class = "S13">,24,41;</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                    </span><span class = "S14">'iMAT_model'</span><span class = "S13">,59,58;</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                    </span><span class = "S14">'mCADRE_model'</span><span class = "S11">,48,49};</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">common_reactions_obtained = model.rxns;</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">obtained_results = cell(size(expected_results,1),2);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">obtained_results(1,:) = {numel(model.rxns),numel(model.mets)};</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S10">for </span><span class = "S11">i = 2:size(expected_results,1)    </span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    obtained_results(i,:) = {eval([</span><span class = "S14">'numel(' </span><span class = "S13">expected_results{i,1} </span><span class = "S14">'.rxns)'</span><span class = "S13">]),eval([</span><span class = "S14">'numel(' </span><span class = "S13">expected_results{i,1} </span><span class = "S14">'.mets)'</span><span class = "S11">])};</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    common_reactions_obtained = eval([</span><span class = "S14">'intersect(common_reactions_obtained,' </span><span class = "S13">expected_results{i,1} </span><span class = "S14">'.rxns)'</span><span class = "S11">]);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S12">end</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11"></span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S18">%Set the result table</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">result_table = table(expected_results(:,1),expected_results(:,2),obtained_results(:,1), expected_results(:,3),obtained_results(:,2),</span><span class = "S12">...</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">                     </span><span class = "S14">'VariableNames'</span><span class = "S13">,{</span><span class = "S14">'ModelName'</span><span class = "S13">,</span><span class = "S14">'Reactions_expected'</span><span class = "S13">,</span><span class = "S14">'Reactions_obtained'</span><span class = "S13">,</span><span class = "S14">'Metabolited_expected'</span><span class = "S13">,</span><span class = "S14">'Metabolited_obtained'</span><span class = "S11">});</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">                 </span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S18">%And display it, any values that do not match indicate some form of problem.                 </span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">disp(result_table)</span></p></div></div><p class = "S16"><span class = "S2">14 reactions are common to the 6 extracted models:</span></p><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S18">%These are the expected reactions</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">common_reacs = {</span><span class = "S14">'ACALD'</span><span class = "S13">;</span><span class = "S14">'ALCD2x'</span><span class = "S13">;</span><span class = "S14">'ATPS4r'</span><span class = "S13">;</span><span class = "S14">'ENO'</span><span class = "S13">;</span><span class = "S14">'FBA'</span><span class = "S13">;</span><span class = "S14">'GAPD'</span><span class = "S13">;</span><span class = "S14">'GLUDy'</span><span class = "S13">;</span><span class = "S14">'GLUSy'</span><span class = "S13">;</span><span class = "S14">'PDH'</span><span class = "S13">;</span><span class = "S14">'PFK'</span><span class = "S13">;</span><span class = "S14">'PGI'</span><span class = "S13">;</span><span class = "S14">'PGK'</span><span class = "S13">;</span><span class = "S14">'PYK'</span><span class = "S13">;</span><span class = "S14">'TPI'</span><span class = "S11">}</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11"></span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S18">%We will test if those are present and inform if not.</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S10">if </span><span class = "S11">isempty(setxor(common_reacs,common_reactions_obtained))</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    fprintf(</span><span class = "S14">'The expected common reacs where obtained:\n'</span><span class = "S11">);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">    printRxnFormula(model,common_reacs);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S12">else</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">    not_in_all = setdiff(common_reacs,common_reactions_obtained);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">    not_in_expected = setdiff(common_reactions_obtained,common_reacs);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    </span><span class = "S10">if </span><span class = "S11">~isempty(not_in_all)</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">        fprintf(</span><span class = "S14">'The following reactions were not in all models but should be contained in them:\n'</span><span class = "S11">);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">        printRxnFormula(model,not_in_all);    </span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    </span><span class = "S12">end</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    </span><span class = "S10">if </span><span class = "S11">~isempty(not_in_expected)</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">        fprintf(</span><span class = "S14">'The following reactions were unexpectedly present in all tissue models:\n'</span><span class = "S11">);</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S11">        printRxnFormula(model,not_in_expected);    </span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S13">    </span><span class = "S12">end</span></p></div><div class = 'inlineWrapper'><p class = "S9 lineNode"><span class = "S12">end</span></p></div></div><h2 class = "S3"><span class = "S2">REFERENCES</span></h2><p class = "S4"><span class = "S22">1. Opdam, S,, Richelle, A., Kellman, B., Li, S., Zielinski, D.C., Lewis, N.E. A systematic evaluation of methods for tailoring genome-scale models. Cell Syetms, 4:1-12 (2017).</span></p><p class = "S4"><span class = "S22">2. Vlassis, N., Pacheco, M.P., and Sauter, T. Fast reconstruction of compact context-specific metabolic network models. PLoS Comput. Biol. 10, e1003424 (2014).</span></p><p class = "S4"><span class = "S22">3. Becker, S.A., and Palsson, B.O. Context-specific metabolic networks are consistent with experiments. PLoS Comput. Biol. 4, e1000082 (2008).</span></p><p class = "S4"><span class = "S22">4. Zur, H., Ruppin, E., and Shlomi, T. iMAT: an integrative metabolic anal- ysis tool. Bioinformatics 26, 3140–3142 (2010).</span></p><p class = "S4"><span class = "S22">5.Agren, R. et al. Reconstruction of genome-scale active metabolic networks for 69 human cell types and 16 cancer types using INIT. PLoS Comput. Biol. 8, e1002518 (2012).</span></p><p class = "S4"><span class = "S22">6. Jerby, L., Shlomi, T., and Ruppin, E. Computational reconstruction of tissue-specific metabolic models: application to human liver metabolism. Mol. Syst. Biol. 6, 401 (2010). </span></p><p class = "S4"><span class = "S22">7. Wang, Y., Eddy, J.A., and Price, N.D. Reconstruction of genome-scale metabolic models for 126 human tissues using mCADRE. BMC Syst. Biol. 6, 153 (2012). </span></p><p class = "S4"><span class = "S22">8. Uhlén M, et al. Tissue-based map of the human proteome. Science, 347(6220):1260419  (2015). </span></p><p class = "S4"><span class = "S2"></span></p><p class = "S4"><span class = "S2"></span></p><p class = "S4"><span class = "S2"></span></p></div></div>
<!-- 
##### SOURCE BEGIN #####
%% Extraction of context-specific models
%% Authors: Thomas Pfau, Systems Biochemistry Group, University of Luxembourg - Anne Richelle, Systems Biology and Cell Engineering, University of California San Diego
%% Reviewer(s): *Almut Heinken, Molecular Systems Physiology Group, LCSB, University of Luxembourg*
%% INTRODUCTION
% Genome-scale reconstruction of metabolism (GEM) can illuminate the molecular 
% basis of cell phenotypes exhibited by an organism. Since some enzymes are only 
% active in specific cell types or environmental conditions, several algorithms 
% have been developed to extract context-specific models that capture the metabolism 
% of individual tissues or cell types. Therefore, a context-specific model is 
% a subset of the GEM, in which inactive reactions are removed. Reaction removal 
% is determined by the algorithm used, gene expression levels, presence of proteins 
% or metabolites, experimental data availability, literature knowledge, and/or 
% predefined metabolic functions of the cell type that need to be maintained in 
% the extracted model. These decisions on methodology and data processing significantly 
% influence the size, functionality and accuracy of constructed context-specific 
% models. While there is no strong evidence that one model extraction method (MEM) 
% universally gives the most physiologically accurate models, each method has 
% different underlying assumptions that affect the resulting model. Therefore, 
% selection of the MEM and the associated parameters should be done while considering 
% the goals of the study and the available data [1].  	 
% 
% Multiple algorithms have been suggested to automatically derive context 
% specific networks from a generic reconstruction and a set of transcriptomic 
% or proteomic data. The COBRA toolbox offers the following selection of extraction 
% agorithms: 
% 
% * FASTCORE [2] - Define one set of core reactions that is guaranteed to be 
% active in the extracted model and find the minimum number of reactions possible 
% to support the core;
% * GIMME [3] - Minimize usage of low-expression reactions while keeping the 
% objective (e.g., biomass) above a certain value. Does not favor inclusion of 
% reactions not related to the objective; 
% * iMAT [4] - Find the optimal trade-off between including high-expression 
% reactions and removing low-expression reactions; 
% * INIT [5] - Find the optimal trade-off between including and removing reactions 
% based on their given weights. If desired, accumulation of certain metabolites 
% can be allowed or even forced; 
% * MBA [6] - Define high-confidence reactions to ensure activity in the extracted 
% model. Medium confidence reactions are only kept when a certain parsimony trade-off 
% is met. In random order, prune other reactions and remove them if not required 
% to support high- or medium- confidence reactions; 
% * mCADRE [7] - Define a set of core reactions and prune all other reactions 
% based on their expression, connectivity to core and confidence score. Remove 
% reactions not necessary to support the core or defined functionalities. Core 
% reactions are only removed if supported by a certain number of zero-expression 
% reactions. 
% 
% For convenience, there is a common interface to use these algorithms provided 
% in the createTissueSpecificModel method. The method can be called as follow:
% 
% |tissueModel = createTissueSpecificModel(model, options)|
%% PROCEDURE
% If necessary, initialise The COBRA Toolbox

global TUTORIAL_INIT_CB;
if ~isempty(TUTORIAL_INIT_CB) && TUTORIAL_INIT_CB==1
    initCobraToolbox
end
changeCobraSolver ('glpk', 'all');
%% 
% Load the model that will be used for the extraction. For this tutorial, 
% we have choosen to use E. coli core model as example. Please download the model 
% from <http://systemsbiology.ucsd.edu/sites/default/files/Attachments/Images/downloads/Ecoli_core/ecoli_core_model.mat 
% http://systemsbiology.ucsd.edu/sites/default/files/Attachments/Images/downloads/Ecoli_core/ecoli_core_model.mat> 
% and save it in your preferred folder.

modelFileName = 'ecoli_core_model.mat';
modelDirectory = getDistributedModelFolder(modelFileName); %Look up the folder for the distributed Models.
modelFileName= [modelDirectory filesep modelFileName]; % Get the full path. Necessary to be sure, that the right model is loaded
model = readCbModel(modelFileName);

%% 
% Load the expression data that will be used for the extraction. For this 
% tutorial, we have choosen to use E. coli Microarray-based gene expression data 
% (downloaded from http://systemsbiology.ucsd.edu/InSilicoOrganisms/Ecoli/EcoliExpression2) 
% related to a anaerobic growth on glucose of a wild-type E. coli strain.

load('dataEcoli');
%% 
% Depending on the method that will be used for the extraction, different 
% options need to be provided that could depend on a preprocessing step of the 
% gene expression data. Since preprocessing is data dependent and the data used 
% can involve multiple sources, we provide  preprocessed data for all the methods. 
% The section "CRITICAL STEP" below briefly explains how the preprocessing has 
% been done for this tutorial and provides some guidelines for the users on how 
% to preprocess their data.
% 
% Load the preprocessed data associated with the extraction method. Select 
% one of the following sections to select the algorithm and data you want to use.
% 
% For the iMAT method, the available parameter options are (with all options 
% marked with * being optional):
% 
% * *options.solver* : 'iMAT'
% * *options.expressionRxns* : gene expression data corresponding to model.rxns. 
% Note : If no gene expression data are available for a reaction, set the value 
% to -1
% * *options.threshold_lb* : lower bound of gene expression threshold, reactions 
% with expression below this value are "non-expressed"
% * *options.threshold_ub* : upper bound of gene expression threshold, reactions 
% with  expression above this value are "expressed"
% * *options.tol** : minimum flux value for "expressed" reactions (default - 
% 1e-8)
% * *options.core** : list of reaction names that are associated with a high 
% confidence (default - no core reactions)                           
% * *options.logfile** : name of the file to save the MILP log (defaut - 'MILPlog')
% * *options.runtime** : maximum solve time for the MILP (default - 7200s)

options = 'options_iMAT';
%% 
% Now, load the preprocessed data 

load(['options_methods' filesep options]);
%% 
% Call createTissueSpecificModel to extract your model

iMAT_model = createTissueSpecificModel(model, options);
%% 
% For GIMME method, the available parameter options are (with all options 
% marked with * being optional):
% 
% * *options.solver* : 'GIMME'
% * *options.expressionRxns* : gene expression data corresponding to model.rxns. 
% Note : If no gene expression data are available for a reaction, set the value 
% to -1
% * *options.threshold* : gene expression threshold, reactions below this are 
% minimized
% * *options.obj_frac** : minimum fraction of the model objective function that 
% need to be maintained in the extracted model (default - 0.9)
% 
% The code to create the tissue-specific model is as for iMAT.

options = 'options_GIMME';
load(['options_methods' filesep options]);
GIMME_model = createTissueSpecificModel(model, options);
%% 
% For MBA  method, the available parameter options are (with all options 
% marked with * being optional):
% 
% * *options.solver* : 'MBA'
% * *options.medium_set* : list of reaction names with medium confidence
% * *options.high_set* : list of reaction names with high confidence
% * *options.tol** : minimum flux value for "expressed" reactions (default - 
% 1e-8)

options = 'options_MBA';
load(['options_methods' filesep options]);
MBA_model = createTissueSpecificModel(model, options);
%% 
% For FASTCORE  method, the available parameter options are (with all options 
% marked with * being optional):
% 
% * *options.solver* : 'fastCore'
% * *options.core* : indices of reactions in  the model that are part of the 
% core set of reactions
% * *options.epsilon** : smallest flux value that is considered nonzero (default 
% - 1e-4)
% * *options.printLevel** : 0 = silent, 1 = summary, 2 = debug (default - 0)

options = 'options_fastCore';
load(['options_methods' filesep options]);
FastCore_model = createTissueSpecificModel(model, options);
%% 
% For mCADRE method, the available parameter options are (with all options 
% marked with * being optional):
% 
% * *options.solver* : 'mCADRE'
% * *options.ubiquityScore* : ubiquity scores, vector of the size of 'model.rxns' 
% quantifying how often a gene is expressed accross samples
% * *options.confidenceScores* : literature-based evidence for generic model, 
% vector of the size of 'model.rxns'
% * *options.protectedRxns** : cell array with reactions names that are manually 
% added to the core reaction set (default - no reactions)
% * *options.checkFunctionality** : Boolean variable that determines if the 
% model should be able to produce the metabolites associated with the '_protectedRxns_' 
% (0: don't use functionality check (default value) - 1: include functionality 
% check)
% * *options.eta** : tradeoff between removing core and zero-expression reactions 
% (default - 1/3)
% * *options.tol** : minimum flux value for "expressed" reactions (default - 
% 1e-8)

options = 'options_mCADRE';
load(['options_methods' filesep options]);
mCADRE_model = createTissueSpecificModel(model, options);
%% 
% For INIT method, the available parameter options are (with all options 
% marked with * being optional):
% 
% * *options.solver* : 'INIT'
% * *options.weights* : column with positive (high expression) and negative 
% (low expression) weights for each reaction
% * *options.tol** : minimum flux value for "expressed" reactions (default - 
% 1e-8)
% * *options.logfile** : name of the file to save the MILP log (defaut - 'MILPlog')
% * *options.runtime** : maximum solve time for the MILP (default - 7200s)

options = 'options_INIT';
load(['options_methods' filesep options]);
INIT_model = createTissueSpecificModel(model, options);
%% 
% Note that additional options are available when extracting your model 
% (|funcModel| and |exRxnRemove|). |funcModel| allows to define whether the extracted 
% model will be/ or not flux consistent (i.e., an extracted model that contains 
% only reactions that can carry fluxes). |exRxnRemove|_ _allows to predefine a 
% list of reactions that will be automatically removed in the extracted model. 
% Example, if you want to extract a consistent model but you do not have any predefined 
% list of reactions to remove, your call will be:

funcModel=1;
exRxnRemove={};
tissueModel = createTissueSpecificModel(model, options,funcModel,exRxnRemove);
%% *CRITICAL STEPS*
% *Gene Expression Preprocessing*
% 
% When integrating transcriptomic data, the selection of options related 
% to each method is critical and algorithmic performance often strongly depends 
% on these choices. In GIMME [3], a fixed threshold value was used to distinguish 
% between unexpressed and expressed reactions (threshold = 12 of log2 expression). 
% For iMAT [4], the authors used a data dependent cutoff of half a standard deviation 
% above the mean for active reactions, and half a standard deviation below for 
% inactive reactions. In contrast to the selection by automated processing of 
% expression data, the authors of FASTCORE [2] and MBA [5] used manually assembled 
% sets of high confidence reactions along with data derived information as input. 
% In INIT [5], the selection is based on the data in the Human Protein Atlas [8] 
% using the indication from the proteomics data. For mCADRE [7], proprietary software 
% was used to obtain presence/absence calls. 
% 
% In the context of this tutorial, the options have been defined following 
% the original paper for iMAT and GIMME and arbitrary preprocessing as used in 
% [1] has been done for the four other methods.
% 
% The quality of the results also strongly depends on data preprocessing, 
% and the COBRA toolbox does not provide an automatic preprocessing pipeline to 
% derive expression values from raw measurements as multipe methods can be used. 
% We therefore strongly suggest, that all preprocessing and discretization is 
% performed prior to the call to |createTissueSpecificModel|. 
% 
% *Gene to Reaction Mapping*
% 
% As with Preprocessing, we intentionally left the gene to reaction mapping 
% separate from the tissue specific model extraction. The most common approach 
% to obtain reaction expression values from geen expression values is to evaluate 
% the Gene-Protein-Reaction rules associated with each reaction as follows:
% 
% Replace 'and' bin 'min' and 'or' by 'max'. E.g. for a reaction R1 with 
% GPR 'A and (B or C)' with expression of Genes A = 2, B = 7 and C = 0, 'B or 
% C' would evaluate to 7 and the whole would evaluate to 2.
% 
% This type of mapping can be applied to iMAT and GIMME, and is provided 
% as a function in the toolbox:

model = getDistributedModel('ecoli_core_model.mat')
load('dataEcoli');
[expressionRxns parsedGPR] = mapExpressionToReactions(model, expression);

%% 
% INIT and mCADRE use similar approaches, but require the expression to 
% be protein expression, or ubiquityscores respectively, i.e. the function can 
% be used for them, but the inputs need to be adjusted according to the required 
% data for INIT and mCADRE. 
% 
% FastCore and MBA again, rely on clearly defined core sets of reactions, 
% the quality of which strongly influences the resulting models.
%% TIMING
% TIMING: 15 minutes to hours (computation) - days (interpretation)
%% ANTICIPATED RESULTS
% The size of extracted models varies depending of the MEM used

expected_results = {'ecoli_core_model',95,72;...
                    'FastCore_model',45,48;...
                    'GIMME_model',74,69;...
                    'INIT_model',25,32;...
                    'MBA_model',24,41;...
                    'iMAT_model',59,58;...
                    'mCADRE_model',48,49};
common_reactions_obtained = model.rxns;
obtained_results = cell(size(expected_results,1),2);
obtained_results(1,:) = {numel(model.rxns),numel(model.mets)};
for i = 2:size(expected_results,1)    
    obtained_results(i,:) = {eval(['numel(' expected_results{i,1} '.rxns)']),eval(['numel(' expected_results{i,1} '.mets)'])};
    common_reactions_obtained = eval(['intersect(common_reactions_obtained,' expected_results{i,1} '.rxns)']);
end

%Set the result table
result_table = table(expected_results(:,1),expected_results(:,2),obtained_results(:,1), expected_results(:,3),obtained_results(:,2),...
                     'VariableNames',{'ModelName','Reactions_expected','Reactions_obtained','Metabolited_expected','Metabolited_obtained'});
                 
%And display it, any values that do not match indicate some form of problem.                 
disp(result_table)
%% 
% 14 reactions are common to the 6 extracted models:

%These are the expected reactions
common_reacs = {'ACALD';'ALCD2x';'ATPS4r';'ENO';'FBA';'GAPD';'GLUDy';'GLUSy';'PDH';'PFK';'PGI';'PGK';'PYK';'TPI'}

%We will test if those are present and inform if not.
if isempty(setxor(common_reacs,common_reactions_obtained))
    fprintf('The expected common reacs where obtained:\n');
    printRxnFormula(model,common_reacs);
else
    not_in_all = setdiff(common_reacs,common_reactions_obtained);
    not_in_expected = setdiff(common_reactions_obtained,common_reacs);
    if ~isempty(not_in_all)
        fprintf('The following reactions were not in all models but should be contained in them:\n');
        printRxnFormula(model,not_in_all);    
    end
    if ~isempty(not_in_expected)
        fprintf('The following reactions were unexpectedly present in all tissue models:\n');
        printRxnFormula(model,not_in_expected);    
    end
end
%% REFERENCES
% _1. Opdam, S,, Richelle, A., Kellman, B., Li, S., Zielinski, D.C., Lewis, 
% N.E. A systematic evaluation of methods for tailoring genome-scale models. Cell 
% Syetms, 4:1-12 (2017)._
% 
% _2. Vlassis, N., Pacheco, M.P., and Sauter, T. Fast reconstruction of compact 
% context-specific metabolic network models. PLoS Comput. Biol. 10, e1003424 (2014)._
% 
% _3. Becker, S.A., and Palsson, B.O. Context-specific metabolic networks 
% are consistent with experiments. PLoS Comput. Biol. 4, e1000082 (2008)._
% 
% _4. Zur, H., Ruppin, E., and Shlomi, T. iMAT: an integrative metabolic 
% anal- ysis tool. Bioinformatics 26, 3140–3142 (2010)._
% 
% _5.Agren, R. et al. Reconstruction of genome-scale active metabolic networks 
% for 69 human cell types and 16 cancer types using INIT. PLoS Comput. Biol. 8, 
% e1002518 (2012)._
% 
% _6. Jerby, L., Shlomi, T., and Ruppin, E. Computational reconstruction 
% of tissue-specific metabolic models: application to human liver metabolism. 
% Mol. Syst. Biol. 6, 401 (2010). _
% 
% _7. Wang, Y., Eddy, J.A., and Price, N.D. Reconstruction of genome-scale 
% metabolic models for 126 human tissues using mCADRE. BMC Syst. Biol. 6, 153 
% (2012). _
% 
% _8. Uhlén M, et al. Tissue-based map of the human proteome. Science, 347(6220):1260419  
% (2015). _
% 
% 
% 
% 
% 
%
##### SOURCE END #####
--></body></html>
